---
title: "CH variants analysis"
author: "Christelle Colin-Leitzinger"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: no
    theme: cerulean
    highlight: pygments
    df_print: paged
editor_options:
  chunk_output_type: console
---

<style type="text/css">
h1.title {
  font-size: 25px;
}
.figure {
   margin-top: 25px;
   margin-bottom: 25px;
}

table {
    margin-top: 25px;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.align='center'
                      )
```


<style>
div.blue { background-color:#0099CC; border-radius: 5px; padding: 20px; font-size: 38px}
</style>
<div class = "blue">

<span style="color: white;">Potential CH variants analysis in GnomAD database</span>

</div>
<br>

```{r library, include=FALSE}
library(tidyverse)
library(gtsummary)

theme_gtsummary_compact()
theme_set(theme_classic())
```

```{r load}
# path <- fs::path("", "Volumes", "Lab_Gillis", "Christelle")
# 
# gnomad_variants <-
#   read.delim(paste0(path, "/gnomAD_skweness/potential_CH_variants/chr2_CH_variants.vcf.gz"), 
#              colClasses = c("ALT"="character"),
#               sep = " ")

gnomad_variants <-
  read.delim(paste0(here::here(), "/chr2_CH_variants.vcf.gz"), 
             colClasses = c("ALT"="character"),
              sep = " ")
```
The number of variants in chromosome `r as.character(gnomad_variants %>% distinct(X.CHROM))` is `r nrow(gnomad_variants %>% distinct(IDs))`.
```{r BH}
pval_selection <- 0.05
mannw_variants <- gnomad_variants %>% 
  filter(adjusted_pval <= pval_selection)
```
In chromosome `r as.character(mannw_variants %>% distinct(X.CHROM))`, the number of variants with BH â‰¤ `r pval_selection` is `r nrow(mannw_variants %>% distinct(IDs))`.
```{r Hedges g}
hedges_selection <- "negligible|small"
CH_variants <- mannw_variants %>%
  # filter(effect_size_cat != "negligible")
  filter(!str_detect(effect_size_cat, hedges_selection))
```
In chromosome `r as.character(CH_variants %>% distinct(X.CHROM))`, the number of variants with Hedges g effect size better than `r hedges_selection %>% str_replace(., "\\|", " or ")` is `r nrow(CH_variants %>% distinct(IDs))`.
<br>
<br>

# Number of overall variants per gene
Not averaged
```{r fig number of variant per gene}
known_CH_genes <- c("DNMT3A", "AC093899.2")
known_CH_gene_list <- paste0(known_CH_genes, collapse = "|")

CH_variants %>% 
  filter(!is.na(Gene.name)) %>% 
  # mutate(known_CH_gene = case_when(
  #   str_detect(Gene.name, known_CH_gene_list)       ~ "known CH gene",
  #   TRUE                                            ~ Gene.name
  # )) %>%
  # mutate(known_CH_gene = fct_relevel(known_CH_gene, "known CH gene")) %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  ggplot(aes(x= Gene.name, fill= Gene.name))+
  geom_bar(show.legend = FALSE)+
  ggtitle("Number of overall variants per gene")+
  labs(x= NULL)+
  coord_flip()#+ 
  # coord_cartesian(ylim = c(0, 1000))
```

# Number of overall variants per gene
Averaged
```{r fig number of variant per gene}

CH_variants1 <- CH_variants %>% filter(str_detect(Gene.name, "DNMT3|AAK1")) %>% 
  select(IDs, Gene.name)

b <- CH_variants1 %>% 
  filter(!is.na(Gene.name)) %>% 
  mutate(known_CH_gene = case_when(
    str_detect(Gene.name, known_CH_gene_list)       ~ "known CH gene",
    TRUE                                            ~ Gene.name
  )) %>%
  group_by(Gene.name) %>% 
  arrange(Gene.name) %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  mutate(number_of_variants_per_gene = n()) %>% 
  
  select(-IDs) %>% 
  group_by(Gene.name, number_of_variants_per_gene) %>% 
  distinct(Gene.name, .keep_all = TRUE) %>% 
  
  mutate(average_of_variants_per_gene = case_when(
    str_detect(Gene.name, known_CH_gene_list)          ~ number_of_variants_per_gene / length(known_CH_genes), #wrong here need to calculate how many ch gene are effectively in the data. not just the number in the string
    TRUE                                               ~ as.numeric(number_of_variants_per_gene)
  )) 
  
  # mutate(known_CH_gene = fct_relevel(known_CH_gene, "known CH gene")) %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  ggplot(aes(x= Gene.name, fill= Gene.name))+
  geom_bar(show.legend = FALSE)+
  ggtitle("Number of overall variants per gene")+
  labs(x= NULL)+
  coord_flip()#+ 
  # coord_cartesian(ylim = c(0, 1000))
```

# Cosmic Analysis for each variants
here 1 time / variant counted
```{r plot variant in cosmic}
CH_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  ggplot(aes(x=variant_in_cosmic))+
  geom_bar()+
  ggtitle("Number of unique variants in Cosmic")
```

```{r}
variants_in_cosmic <- CH_variants %>% 
  filter(variant_in_cosmic == "Yes")
```
In chromosome `r as.character(variants_in_cosmic %>% distinct(X.CHROM))`, the number of variants present in Cosmic is `r nrow(variants_in_cosmic %>% distinct(IDs))`.

### Restricted analysis to few variants
2-25234373-C-T is the CH known variant in DNMT3A gene + 2-25234374-G-A   
2-206304912-G-A is in ZDBF2 gene
```{r variants_restricted}
variants_restricted <- variants_in_cosmic %>% 
  filter(str_detect(IDs, "2-25234373-C-T|2-25234374-G-A|2-206304912-G-A"))
rm(mannw_variants)
```

The variable available in Cosmic are:
`"gene_name_cosmic", "Accession.Number", "Gene.CDS.length", "HGNC.ID", "Sample.name", `
                         `  "ID_sample", "ID_tumour", "Primary.site", "Site.subtype.1", "Site.subtype.2", `
                         `  "Site.subtype.3", "Primary.histology", "Histology.subtype.1", "Histology.subtype.2", `
                        `   "Histology.subtype.3", "Genome.wide.screen", "GENOMIC_MUTATION_ID", `
                        `   "LEGACY_MUTATION_ID", "MUTATION_ID", "Mutation.CDS", "Mutation.AA", `
                         `  "Mutation.Description", "Mutation.zygosity", "LOH", "GRCh", `
                        `   "Mutation.genome.position", "Mutation.strand", "Resistance.Mutation", `
                        `   "FATHMM.prediction", "FATHMM.score", "Mutation.somatic.status", "Pubmed_PMID", `
                        `   "ID_STUDY", "Sample.Type", "Tumour.origin", "Age", "HGVSP", "HGVSC", "HGVSG"`
                        
`r emo::ji("warning")` 1 variant can have multiple samples and primary site, etc
```{r}
variants_restricted %>% 
  ggplot(aes(x=Gene.name))+
  geom_bar()+
  ggtitle("Number of overall samples per gene")

variants_restricted %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  ggplot(aes(x=Gene.name))+
  geom_bar()+
  ggtitle("Number of overall variants per gene")
```





## Cosmic Patients/Cell Line Characteristics

```{r Cosmic Patients/Cell Line Characteristics}
# show_header_names(tbl)

variants_restricted %>% 
  select(Age,
         Primary.site, Primary.histology, Tumour.origin,
         Sample.Type,
         Mutation.Description,
         FATHMM.prediction, Mutation.somatic.status, 
         IDs) %>% 
  tbl_summary(by= IDs) %>% 
  bold_labels() %>% 
  modify_header(
    label = '**Cosmic Patients/Cell Line Characteristics**'#,
    # stat_0 = '**Variants number**, N = {n}',
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Variants**")
```

### each sample of a variant is counted.
```{r sample plot}
cosmic_var1 <- c("Age",
         "Primary.site", "Primary.histology", "Tumour.origin",
         "Sample.Type",
         "Mutation.Description",
         "FATHMM.prediction", "Mutation.somatic.status") 

for (i in cosmic_var1){
  print(
    variants_restricted %>%
      ggplot(aes(x= .data[[i]] ))+
      geom_bar()+
      facet_wrap(.~ IDs)
  )
}
```

### 1 variant is counted once
```{r variant plot}
for (i in cosmic_var1){
  print(
    variants_restricted %>%
      distinct(IDs, .data[[i]]) %>% 
      ggplot(aes(x= .data[[i]] ))+
      geom_bar()
  )
}
# # bar_plot <- function(data){
#   for (i in cosmic_var1){
#     # if(class(markers_ROI[[i]]) == "factor" | class(markers_ROI[[i]]) == "character") {
#       variants_restricted %>%
#         ggplot()+
#         geom_bar(aes(x=.data[[i]]))
#       # p2 <- markers_ROI %>%
#       #   ggplot()+
#       #   geom_violin(aes(x=.data[[i]], y= .data[[paste0(type, ".p")]]), scale = "count")+
#       #   geom_boxplot(aes(x=.data[[i]], y= .data[[paste0(type, ".p")]]), width=.1) +
#       #   theme_classic2()+
#       #   labs(x=NULL, y=NULL, title="Peripheral \nROIs")
#       # main <- str_match(type, "percent_(.*)_total")[,2]
#       # gridExtra::grid.arrange(p1, p2, ncol = 2,
#       #                         top = "% of Cells Present in",
#       #                         left =  paste("Overall", main, "immune cells (%) per patients"),
#       #                         bottom = "Data on Race-Matched Patients")
#     # }
#   }
# # }
# bar_plot(data = variants_restricted)
```

## Cosmic Gene Characteristics
```{r Cosmic Gene Characteristics}
variants_restricted %>% 
  select(gene_name_cosmic,
         Gene.CDS.length, HGNC.ID,
         Genome.wide.screen, GENOMIC_MUTATION_ID, 
         LEGACY_MUTATION_ID, MUTATION_ID, Mutation.CDS, Mutation.AA, 
         Mutation.Description, Mutation.zygosity,
         Mutation.genome.position, Mutation.strand, Resistance.Mutation, 
         FATHMM.prediction, FATHMM.score, Mutation.somatic.status,
         HGVSP, HGVSC, HGVSG,
         IDs) %>% 
  tbl_summary(by= IDs) %>% 
  bold_labels() %>% 
  modify_header(
    label = '**Cosmic Gene Characteristics**'
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Variants**")
```

### each sample of a variant is counted.
```{r cosmic plot}
cosmic_var2 <- c("gene_name_cosmic",
         "Gene.CDS.length", "HGNC.ID",
         "Genome.wide.screen", "GENOMIC_MUTATION_ID", 
         "LEGACY_MUTATION_ID", "MUTATION_ID", "Mutation.CDS", "Mutation.AA", 
         "Mutation.Description", "Mutation.zygosity",
         "Mutation.genome.position", "Mutation.strand", "Resistance.Mutation", 
         "FATHMM.prediction", "FATHMM.score", "Mutation.somatic.status",
         "HGVSP", "HGVSC", "HGVSG") 

for (i in cosmic_var2){
  print(
    variants_restricted %>%
      ggplot(aes(x= .data[[i]] ))+
      geom_bar()+
      facet_wrap(.~ IDs)
  )
}
```

### 1 variant is counted once
```{r}
for (i in cosmic_var2){
  print(
    variants_restricted %>%
      distinct(IDs, .data[[i]]) %>% 
      ggplot(aes(x= .data[[i]] ))+
      geom_bar()
  )
}

```














# GnomAD Analysis for each variants

## Extract Consequence
I extracted the VEP annotations data from `INFO`.
VEP contains these information :
`"Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type", "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp", "`
  `cDNA_position", "CDS_position", "Protein_position", "Amino_acids", "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "`
  `STRAND", "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID", "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "`
  `TREMBL", "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET", "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "`
  `SAS_MAF", "AA_MAF", "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", "ExAC_EAS_MAF", "ExAC_FIN_MAF", "`
  `ExAC_NFE_MAF", "ExAC_OTH_MAF", "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME", "MOTIF_POS", "`
  `HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter", "LoF_flags", "LoF_info"`

```{r GnomAD Analysis}
variants_in_gnomAD <- CH_variants %>%
  distinct(IDs, .keep_all = TRUE)
```

```{r plot bar charts number of candidates known and unknown in gnomAD}
var_restricted <- variants_in_gnomAD %>%
  filter(str_detect(IDs, "2-25234373-C-T|2-206304912-G-A"))

var_restricted %>%
  ggplot(aes(x=Gene.name))+
  geom_bar()+
  ggtitle("Number of overall variants per gene")
```

```{r}
variants_in_gnomAD <- variants_in_gnomAD %>%
  filter(str_detect(IDs, "2-25234373-C-T|2-25234321-G-A|2-25234374-G-A|2-25328714-C-G|2-25328736-G-A|2-25244379-G-C|2-25244494-C-T|2-25248052-G-A|2-25274935-C-T|2-25282744-G-A"))
```

## Plots alleles count
```{r, fig.width=12, fig.height=12}
variants_in_gnomAD %>%
  select(IDs,
         nc_freq_allele_count, nc_freq_allele_count_female, nc_freq_allele_count_male,
         nc_freq_allele_count_afr, nc_freq_allele_count_afr_female, nc_freq_allele_count_afr_male,
         nc_freq_allele_count_amr, nc_freq_allele_count_amr_female, nc_freq_allele_count_amr_male) %>%
  pivot_longer(cols = -IDs) %>%
  mutate(value = as.numeric(value)) %>%
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>%
  mutate(Ethnicity = str_extract(name, "afr|amr"), Ethnicity = case_when(
    is.na(Ethnicity)       ~ "overall",
    TRUE                   ~ Ethnicity
  )) %>%
  ggplot(aes(x=name, y= value, color= Sex, shape= Ethnicity))+
  geom_point(size= 3)+
  labs(title = "Frequency of the allele in Non-Cancer population")+
  theme_classic(base_size = 16)+
  theme(legend.position = "none")+
  facet_wrap(. ~ IDs, scales = "free", ncol = 2)+
  coord_flip()
```

## Extract Consequence
I extracted the VEP annotations data from `INFO`.
VEP contains these information :
`"Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type", "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp", "`
  `cDNA_position", "CDS_position", "Protein_position", "Amino_acids", "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "`
  `STRAND", "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID", "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "`
  `TREMBL", "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET", "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "`
  `SAS_MAF", "AA_MAF", "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", "ExAC_EAS_MAF", "ExAC_FIN_MAF", "`
  `ExAC_NFE_MAF", "ExAC_OTH_MAF", "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME", "MOTIF_POS", "`
  `HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter", "LoF_flags", "LoF_info"`
```{r}
vep_in_gnomAD <- variants_in_gnomAD %>%
  mutate(ens_vep = str_match(INFO, ";vep=(.*?)$")[,2])
vep_in_gnomAD <- vep_in_gnomAD %>%
  separate(col = ens_vep, paste("ens_vep", 1:25, sep=""),
           sep = ",", remove = T, extra = "warn", fill = "right") %>%
  keep(~!all(is.na(.))) %>%
  pivot_longer(cols = starts_with("ens_vep"), names_to = NULL, values_to = "ens_vep") %>%
  drop_na(ens_vep)

ens_vep_var_names <- c("Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type",
                       "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp",
                       "cDNA_position", "CDS_position", "Protein_position", "Amino_acids",
                       "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "STRAND",
                       "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID",
                       "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "TREMBL",
                       "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET",
                       "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "SAS_MAF", "AA_MAF",
                       "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF",
                       "ExAC_EAS_MAF", "ExAC_FIN_MAF", "ExAC_NFE_MAF", "ExAC_OTH_MAF",
                       "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME",
                       "MOTIF_POS", "HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter",
                       "LoF_flags", "LoF_info")

vep_in_gnomAD <- vep_in_gnomAD %>%
  separate(col = ens_vep, into = ens_vep_var_names,
           sep = "\\|", remove = F, extra = "warn", fill = "right")
```

## VEP in gmomAD
multiple VEP per variants
```{r}
vep_in_gnomAD %>%
  select(IDs, SYMBOL, Consequence, HGVSp,
         "Feature_type", "BIOTYPE", "HGVSc", "HGVSp", "Protein_position", "Amino_acids",
         "VARIANT_CLASS", "SIFT", "PolyPhen",
         "GMAF",
         "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "SAS_MAF", "AA_MAF", "EA_MAF", "ExAC_MAF",
         "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", "ExAC_EAS_MAF", "ExAC_FIN_MAF",
         "ExAC_NFE_MAF", "ExAC_OTH_MAF", "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO",
         "MOTIF_NAME", "MOTIF_POS", "HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF",
         "LoF_filter", "LoF_flags", "LoF_info")
```

```{r}
vep_in_gnomAD %>%
  select(IDs, VARIANT_CLASS, GMAF) %>%
  tbl_summary(by= IDs) %>%
  bold_labels() %>%
  modify_header(
    label = '**VEP in gmomAD Characteristic**'#,
    # stat_0 = '**Variants number**, N = {n}',
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Variants**")
```







# Bick Analysis for each variants
<!-- **CH R882H mutation is depicted in red**   -->
<!-- bh adjusted_selected in yellow    -->
<!-- effect size_selected in orange -->
```{r}
R882H <- "2-25234373-C-T"
bhadjusted_selected <- paste0(c("2-25244379-G-C",
                                "2-25244494-C-T", "2-25248052-G-A", "2-25274935-C-T",
                                "2-25282744-G-A"),
                              collapse = "|")
effsize_selected <- paste0(c("2-25234321-G-A", "2-25234374-G-A",
                                 "2-25328714-C-G", "2-25328736-G-A"),
                              collapse = "|")
```

# Definition

Mutation somatic status :  
Information on whether the sample was reported to be Confirmed Somatic, Previously Reported or Variant of unknown origin  
- Previously observed = when the mutation has been reported as somatic previously but not in current paper.  
- Confirmed Somatic = if the mutation has been confimed to be somatic in the experiment by sequencing both the tumour and a matched normal from the same patient.  
- Variant of unknown origin = when the mutation is known to be somatic but the tumour was sequenced without a matched normal.


INFO=<ID=AC,Number=A,Type=Integer,Description="Alternate allele count for samples  
afr, African-American/African ancestry
amr, Latino ancestry
asj, Ashkenazi Jewish ancestry
eas, East Asian ancestry  
fin, Finnish ancestry  
nfe, Non-Finnish European ancestry  
nfe_bgr, Bulgarian (Eastern European) ancestry  
nfe_est, Estonian ancestry  
nfe_nwe, North-Western European ancestry  
nfe_onf, Other Non-Finnish European ancestry  
nfe_seu, Southern European ancestry  
nfe_swe, Swedish ancestry  
oth, Other ancestry  




