---
title: "CH variants analysis"
author: "Christelle Colin-Leitzinger"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: no
    theme: cerulean
    highlight: pygments
    df_print: paged
editor_options:
  chunk_output_type: console
---

<style type="text/css">
h1.title {
  font-size: 25px;
}
.figure {
   margin-top: 25px;
   margin-bottom: 25px;
}

table {
    margin-top: 25px;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.align='center'
                      )
```


<style>
div.blue { background-color:#0099CC; border-radius: 5px; padding: 20px; font-size: 38px}
</style>
<div class = "blue">

<span style="color: white;">Potential CH variants analysis in GnomAD database</span>

</div>
<br>

```{r library, include=FALSE}
library(tidyverse)
library(gtsummary)

theme_gtsummary_compact()
theme_set(theme_classic())
```

```{r load}
path <- fs::path("", "Volumes", "Lab_Gillis", "Christelle")

gnomad_variants <-
  read.delim(paste0(path, "/gnomAD_skweness/potential_CH_variants/chr2_CH_variants.vcf.gz"), 
             colClasses = c("ALT"="character"),
              sep = " ")
```
The number of variants in chromosome `r gnomad_variants %>% distinct(X.CHROM)` is `r nrow(gnomad_variants %>% distinct(IDs))`.
```{r}
mannw_variants <- gnomad_variants %>% 
  filter(adjusted_pval <= 0.1)
```
In chromosome `r mannw_variants %>% distinct(X.CHROM)`, the number of variants with BH â‰¤ 0.1 is `r nrow(mannw_variants %>% distinct(IDs))`.
```{r}
# CH_variants <- mannw_variants %>% 
#   filter(effect_size_cat != "negligible")
```
In chromosome `r CH_variants %>% distinct(X.CHROM)`, the number of variants with Hedges g effect size better than negligible is `r nrow(CH_variants %>% distinct(IDs))`.
FIX THIS
<br>
<br>

# GnomAD Analysis for each variants
```{r}
variants_in_gnomAD <- CH_variants %>% 
  distinct(IDs, .keep_all = TRUE)
```

## Plots alleles count
```{r, fig.width=12, fig.height=12}
variants_in_gnomAD %>% 
  select(IDs,
         nc_freq_allele_count, nc_freq_allele_count_female, nc_freq_allele_count_male, 
         nc_freq_allele_count_afr, nc_freq_allele_count_afr_female, nc_freq_allele_count_afr_male, 
         nc_freq_allele_count_amr, nc_freq_allele_count_amr_female, nc_freq_allele_count_amr_male) %>% 
  pivot_longer(cols = -IDs) %>% 
  mutate(value = as.numeric(value)) %>% 
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>% 
  mutate(Ethnicity = str_extract(name, "afr|amr"), Ethnicity = case_when(
    is.na(Ethnicity)       ~ "overall",
    TRUE                   ~ Ethnicity
  )) %>% 
  ggplot(aes(x=name, y= value, color= Sex, shape= Ethnicity))+
  geom_point(size= 3)+
  labs(title = "Frequency of the allele in Non-Cancer population")+
  theme_classic(base_size = 16)+
  theme(legend.position = "none")+
  facet_wrap(. ~ IDs, scales = "free", ncol = 2)+
  coord_flip()
```

## Extract Consequence
I extracted the VEP annotations data from `INFO`.
VEP contains these information :
`"Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type", "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp", "`
  `cDNA_position", "CDS_position", "Protein_position", "Amino_acids", "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "`
  `STRAND", "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID", "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "`
  `TREMBL", "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET", "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "`
  `SAS_MAF", "AA_MAF", "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", "ExAC_EAS_MAF", "ExAC_FIN_MAF", "`
  `ExAC_NFE_MAF", "ExAC_OTH_MAF", "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME", "MOTIF_POS", "`
  `HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter", "LoF_flags", "LoF_info"`
```{r}
vep_in_gnomAD <- variants_in_gnomAD %>%
  mutate(ens_vep = str_match(INFO, ";vep=(.*?)$")[,2])
vep_in_gnomAD <- vep_in_gnomAD %>%
  separate(col = ens_vep, paste("ens_vep", 1:25, sep=""),
           sep = ",", remove = T, extra = "warn", fill = "right") %>%
  keep(~!all(is.na(.))) %>%
  pivot_longer(cols = starts_with("ens_vep"), names_to = NULL, values_to = "ens_vep") %>%
  drop_na(ens_vep)

ens_vep_var_names <- c("Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type", 
                       "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp", 
                       "cDNA_position", "CDS_position", "Protein_position", "Amino_acids",
                       "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "STRAND", 
                       "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID", 
                       "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "TREMBL",
                       "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET", 
                       "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "SAS_MAF", "AA_MAF",
                       "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", 
                       "ExAC_EAS_MAF", "ExAC_FIN_MAF", "ExAC_NFE_MAF", "ExAC_OTH_MAF", 
                       "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME",
                       "MOTIF_POS", "HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter", 
                       "LoF_flags", "LoF_info")

vep_in_gnomAD <- vep_in_gnomAD %>%
  separate(col = ens_vep, into = ens_vep_var_names,
           sep = "\\|", remove = F, extra = "warn", fill = "right")
```

## VEP in gmomAD
multiple VEP per variants
```{r}
vep_in_gnomAD %>% 
  select(IDs, SYMBOL, Consequence, HGVSp, 
         "Feature_type", "BIOTYPE", "HGVSc", "HGVSp", "Protein_position", "Amino_acids", 
         "VARIANT_CLASS", "SIFT", "PolyPhen", 
         "GMAF", 
         "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "SAS_MAF", "AA_MAF", "EA_MAF", "ExAC_MAF", 
         "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", "ExAC_EAS_MAF", "ExAC_FIN_MAF", 
         "ExAC_NFE_MAF", "ExAC_OTH_MAF", "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", 
         "MOTIF_NAME", "MOTIF_POS", "HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", 
         "LoF_filter", "LoF_flags", "LoF_info")
```

```{r}
vep_in_gnomAD %>% 
  select(IDs, VARIANT_CLASS, GMAF) %>% 
  tbl_summary(by= IDs) %>% 
  bold_labels() %>% 
  modify_header(
    label = '**VEP in gmomAD Characteristic**'#,
    # stat_0 = '**Variants number**, N = {n}',
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Variants**")
```




# Cosmic Analysis for each variants
```{r}
variants_in_cosmic <- CH_variants %>% 
  filter(variant_in_cosmic == "Yes")
```
In chromosome `r variants_in_cosmic %>% distinct(X.CHROM)`, the number of variants present in Cosmic is `r nrow(variants_in_cosmic %>% distinct(IDs))`.


```{r}
# tbl <- variants_in_cosmic %>% 
#   distinct(IDs, .keep_all = TRUE) %>% 
#   select(Primary.site, Primary.histology, Mutation.Description,
#          FATHMM.prediction, Mutation.somatic.status) %>% 
#   tbl_summary() %>% 
#   bold_labels()
# 
# show_header_names(tbl)

variants_in_cosmic %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, Primary.site, Primary.histology, Mutation.Description,
         FATHMM.prediction, Mutation.somatic.status) %>% 
  tbl_summary(by= IDs) %>% 
  bold_labels() %>% 
  modify_header(
    label = '**Cosmic Characteristic**'#,
    # stat_0 = '**Variants number**, N = {n}',
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Variants**")


variants_in_cosmic %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, gene_name_cosmic) %>% 
  tbl_summary(by= IDs) %>% 
  bold_labels()
```


```{r}
variants_in_cosmic %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, occurrence_in_cosmic : controls_popmax) %>% 
  tbl_summary(by= IDs) %>% 
  bold_labels() %>% 
  modify_header(
    label = '**Cosmic Characteristic**'
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Variants**")
```

# by sample
```{r}
variants_in_cosmic %>% 
  select(IDs, Age) %>% 
  tbl_summary(by= IDs,
              type = list(Age ~ "continuous")) %>% 
  bold_labels() %>% 
  modify_header(
    label = '**Cosmic Sample**'
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Variants**")
```



# Bick Analysis for each variants
<!-- **CH R882H mutation is depicted in red**   -->
<!-- bh adjusted_selected in yellow    -->
<!-- effect size_selected in orange -->
```{r}
R882H <- "2-25234373-C-T"
bhadjusted_selected <- paste0(c("2-25244379-G-C", 
                                "2-25244494-C-T", "2-25248052-G-A", "2-25274935-C-T", 
                                "2-25282744-G-A"),
                              collapse = "|")
effsize_selected <- paste0(c("2-25234321-G-A", "2-25234374-G-A",
                                 "2-25328714-C-G", "2-25328736-G-A"),
                              collapse = "|")
```

