---
title: "CH variants analysis"
author: "Christelle Colin-Leitzinger"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: no
    theme: cerulean
    highlight: pygments
    df_print: paged
editor_options:
  chunk_output_type: console
---

<style type="text/css">
h1.title {
  font-size: 25px;
}
.figure {
   margin-top: 25px;
   margin-bottom: 25px;
}

table {
    margin-top: 25px;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.align='center'
                      )
```


<style>
div.blue { background-color:#0099CC; border-radius: 5px; padding: 20px; font-size: 38px}
</style>
<div class = "blue">

<span style="color: white;">Potential CH variants analysis in GnomAD database</span>

</div>
<br>

```{r library, include=FALSE}
library(tidyverse)
library(gtsummary)

theme_gtsummary_compact()
theme_set(theme_classic())
```

```{r load}
path <- fs::path("", "Volumes", "Lab_Gillis", "Christelle")

CH_variants <-
  read.delim(paste0(path, "/gnomAD_skweness/potential_CH_variants/chr2_CH_variants.vcf.gz"),
              sep = " ")
```
The number of variants in chromosome `r CH_variants %>% distinct(X.CHROM)` is `r nrow(CH_variants %>% distinct(IDs))`.
```{r}
CH_variants <- CH_variants %>% 
  filter(adjusted_pval <= 0.1)
```
In chromosome `r CH_variants %>% distinct(X.CHROM)`, the number of variants with BH â‰¤ 0.1 is `r nrow(CH_variants %>% distinct(IDs))`.
```{r}
# CH_variants <- CH_variants %>% 
#   filter(effect_size_cat != "negligible")
```
In chromosome `r CH_variants %>% distinct(X.CHROM)`, the number of variants with Hedges g effect size better than negligible is `r nrow(CH_variants %>% distinct(IDs))`.
FIX THIS
<br>
<br>

# Analysis for each variants
```{r}
CH_variants <- CH_variants %>% 
  filter(variant_in_cosmic == "Yes")
```
In chromosome `r CH_variants %>% distinct(X.CHROM)`, the number of variants present in Cosmic is `r nrow(CH_variants %>% distinct(IDs))`.


```{r}
# tbl <- CH_variants %>% 
#   distinct(IDs, .keep_all = TRUE) %>% 
#   select(Primary.site, Primary.histology, Mutation.Description,
#          FATHMM.prediction, Mutation.somatic.status) %>% 
#   tbl_summary() %>% 
#   bold_labels()
# 
# show_header_names(tbl)

CH_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, Primary.site, Primary.histology, Mutation.Description,
         FATHMM.prediction, Mutation.somatic.status) %>% 
  tbl_summary(by= IDs) %>% 
  bold_labels() %>% 
  modify_header(
    label = '**Cosmic Characteristic**'#,
    # stat_0 = '**Variants number**, N = {n}',
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Variants**")


CH_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, gene_name_cosmic) %>% 
  tbl_summary(by= IDs) %>% 
  bold_labels()
```


```{r}
CH_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, occurrence_in_cosmic : controls_popmax) %>% 
  tbl_summary(by= IDs) %>% 
  bold_labels() %>% 
  modify_header(
    label = '**Cosmic Characteristic**'
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Variants**")
```

# by sample
```{r}
CH_variants %>% 
  select(IDs, Age) %>% 
  tbl_summary(by= IDs,
              type = list(Age ~ "continuous")) %>% 
  bold_labels() %>% 
  modify_header(
    label = '**Cosmic Sample**'
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Variants**")
```




# Bick list
<!-- **CH R882H mutation is depicted in red**   -->
<!-- bh adjusted_selected in yellow    -->
<!-- effect size_selected in orange -->
```{r}
R882H <- "2-25234373-C-T"
bhadjusted_selected <- paste0(c("2-25244379-G-C", 
                                "2-25244494-C-T", "2-25248052-G-A", "2-25274935-C-T", 
                                "2-25282744-G-A"),
                              collapse = "|")
effsize_selected <- paste0(c("2-25234321-G-A", "2-25234374-G-A",
                                 "2-25328714-C-G", "2-25328736-G-A"),
                              collapse = "|")
```

