---
title: "gnomAD cleaning"
author: "Christelle Colin-Leitzinger"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: no
    theme: cerulean
    highlight: pygments
    df_print: paged
editor_options:
  chunk_output_type: console
---

<style type="text/css">
h1.title {
  font-size: 25px;
}
.figure {
   margin-top: 25px;
   margin-bottom: 25px;
}

table {
    margin-top: 25px;
    <!-- margin-bottom: 100px !important; -->
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.align='bin_age'#, fig.show='hold'
                      )
options(gtsummary.print_engine = "gt")
options(gtsummary.as_gt.addl_cmds = "gt::tab_options(table.font.size = 12, data_row.padding = gt::px(1))")
```


<style>
div.blue { background-color:#0099CC; border-radius: 5px; padding: 20px; font-size: 38px}
</style>
<div class = "blue">

<span style="color: white;">GnomAD data</span>

</div>
<br>

```{r library, include=FALSE}
library(tidyverse)
library(e1071)
library(purrr)
library(gtsummary)
```

```{r load}
path <- fs::path("", "Volumes", "Lab_Gillis", "Christelle")
gnomad <-
  read.delim(paste0(path, "/gnomAD_raw_data/dnmt3a_data.vcf.gz"), 
             header = FALSE, 
             col.names = c("X.CHROM", "POS", "ID", "REF", "ALT", "QUAL", "FILTER", "INFO"))
```

As an example I selected all the variant DNMT3A. Here is a example of the raw data.
```{r}
head(gnomad)
```
<br>

***

# I. GnomAD data extraction for variants selection
## 1. Extract age to look for skewness


I extracted the age bin and sample counts for each bin from the `INFO` variable.
```{r}
gnomad_decoded <- gnomad %>% 
  mutate(X.CHROM = str_remove(X.CHROM, "chr")) %>% 
  unite(IDs, c(X.CHROM, POS, REF, ALT), sep = "-", remove = FALSE) %>% 
  # mutate(IDs = factor(row_number())) %>% 
  
  # ab_hist_alt_bin_freq,Number=A,Type=String,Description="Histogram for AB in heterozygous individuals; 
  # bin edges are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00">
  
  # 1. Extract count per age bin for heterozygous ind--
  # age_hist_het_bin_freq,Number=A,Type=String,Description="Histogram of ages of heterozygous individuals; 
  # bin edges are: 30.0|35.0|40.0|45.0|50.0|55.0|60.0|65.0|70.0|75.0|80.0; 
  # total number of individuals of any genotype bin: 2547|3423|4546|8487|10355|12693|11933|10534|8882|5991|4136|1935">
  mutate(age_hist_het_bin_freq = str_match(INFO, "age_hist_het_bin_freq=(.*?);")[,2]) %>%
  # "Count of age values falling below lowest histogram bin edge for heterozygous individuals">
  mutate("<30_het" = as.numeric(str_match(INFO, "age_hist_het_n_smaller=(.*?);")[,2])) %>%
  # "Count of age values falling above highest histogram bin edge for heterozygous individuals">
  mutate(">80_het" = as.numeric(str_match(INFO, "age_hist_het_n_larger=(.*?);")[,2])) %>% 
  separate(col = age_hist_het_bin_freq,
           into = c("30-35_het","35-40_het","40-45_het","45-50_het",
                    "50-55_het","55-60_het","60-65_het","65-70_het",
                    "70-75_het","75-80_het"), 
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>% 
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>% 

  # 2. Extract count per age bin for heterozygous ind--
  mutate(age_hist_hom_bin_freq = str_match(INFO, "age_hist_hom_bin_freq=(.*?);")[,2]) %>%
  # "Count of age values falling below lowest histogram bin edge for homozygous individuals">
  mutate("<30_hom" = as.numeric(str_match(INFO, "age_hist_hom_n_smaller=(.*?);")[,2])) %>%
  # "Count of age values falling above highest histogram bin edge for homozygous individuals">
  mutate(">80_hom" = as.numeric(str_match(INFO, "age_hist_hom_n_larger=(.*?);")[,2])) %>% 
  separate(col = age_hist_hom_bin_freq,
           into = c("30-35_hom","35-40_hom","40-45_hom","45-50_hom",
                    "50-55_hom","55-60_hom","60-65_hom","65-70_hom",
                    "70-75_hom","75-80_hom"), 
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>% 
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>% 
  
  # 3. Combine count for Het and hom individuals
  mutate("<30" = rowSums(select(.,`30-35_het`,`30-35_hom`), na.rm = TRUE),
         "30-35" = rowSums(select(.,`30-35_het`,`30-35_hom`), na.rm = TRUE),
         "35-40" = rowSums(select(.,`35-40_het`,`35-40_hom`), na.rm = TRUE),
         "40-45" = rowSums(select(.,`40-45_het`,`40-45_hom`), na.rm = TRUE),
         "45-50" = rowSums(select(.,`45-50_het`,`45-50_hom`), na.rm = TRUE),
         "50-55" = rowSums(select(.,`50-55_het`,`50-55_hom`), na.rm = TRUE),
         "55-60" = rowSums(select(.,`55-60_het`,`55-60_hom`), na.rm = TRUE),
         "60-65" = rowSums(select(.,`60-65_het`,`60-65_hom`), na.rm = TRUE),
         "65-70" = rowSums(select(.,`65-70_het`,`65-70_hom`), na.rm = TRUE),
         "70-75" = rowSums(select(.,`70-75_het`,`70-75_hom`), na.rm = TRUE),
         "75-80" = rowSums(select(.,`75-80_het`,`75-80_hom`), na.rm = TRUE),
         ">80" = rowSums(select(.,`>80_het`,`>80_hom`), na.rm = TRUE)
         ) %>%
  add_row("IDs"="All individuals", "ID"="All individuals of any genotype bin",
          "<30" = 2547, "30-35" = 3423,
          "35-40" = 4546,"40-45" = 8487,
          "45-50" = 10355,"50-55" = 12693,
          "55-60" = 11933,"60-65" = 10534,
          "65-70" = 8882,"70-75" = 5991,
          "75-80" = 4136,
          ">80" = 1935) %>%
  mutate(nbr_individuals = rowSums(select(.,`<30`:`>80`), na.rm = TRUE)) %>%
  
  select(-c(ends_with("_het"), ends_with("_hom"))) %>% 
  # Filter variants found in more than 10 individuals
  filter(nbr_individuals > 10) %>%
  # Generate AN variable from the INFO var
  mutate(AN = str_match(INFO, "AN=(.*?);")[,2]) %>% 
  mutate(AN_samples_count = (as.numeric(AN) / 2)
    ) %>% 
  select(IDs, everything())
```

`r  emo::ji("question_mark")`
<span style="color:red">**I filtered out variants that are present in less than or equal to 10 samples but should we do more?**

<span style="color:red">**Should we remove variants tested (AN) in less than 500 samples?**</span>
<span style="color:green">I plotted AN against number of individuals and pval, it looks good, so we decided to not filter on the AN number.</span>  

```{r, max.print = 30}
head(gnomad_decoded %>% select(IDs, "<30", "30-35","35-40","40-45","45-50","50-55","55-60","60-65","65-70","70-75","75-80", ">80", everything() ))
```

```{r}
gnomad_decoded <- gnomad_decoded %>% 
  pivot_longer(cols = c(grep("[[:digit:]]", colnames(.))), 
               names_to = "age_bin", values_to = "sample_count") %>% 
  mutate(age_bin = factor(age_bin, 
                          levels = c("<30", "30-35","35-40","40-45","45-50","50-55",
                                     "55-60","60-65","65-70","70-75","75-80", ">80"#,
                                     # "<30_het", "30-35_het","35-40_het","40-45_het","45-50_het","50-55_het",
                                     # "55-60_het","60-65_het","65-70_het","70-75_het","75-80_het", ">80_het",
                                     # "<30_hom", "30-35_hom","35-40_hom","40-45_hom","45-50_hom","50-55_hom",
                                     # "55-60_hom","60-65_hom","65-70_hom","70-75_hom","75-80_hom", ">80_hom"
                                     ))) %>% 
  # mutate(age_bin_het = factor(age_bin, 
  #                         levels = c("<30_het", "30-35_het","35-40_het","40-45_het","45-50_het","50-55_het",
  #                                    "55-60_het","60-65_het","65-70_het","70-75_het","75-80_het", ">80_het"))) %>% 
  # mutate(age_bin_hom = factor(age_bin, 
  #                         levels = c("<30_hom", "30-35_hom","35-40_hom","40-45_hom","45-50_hom","50-55_hom",
  #                                    "55-60_hom","60-65_hom","65-70_hom","70-75_hom","75-80_hom", ">80_hom"))) %>% 
  mutate(bin_age = case_when(
    # The average age at diagnosis is 8 overall (ages 0 to 19), 
    # 5 years old for children (aged 0 to 14), and 17 years old 
    # for adolescents (aged 15 to 19), while adults' average 
    # age for cancer diagnosis is 65
    str_detect(age_bin, "<30")              ~ 15, 
    str_detect(age_bin, "30-35")            ~ 32.5,
    str_detect(age_bin, "35-40")            ~ 37.5,
    str_detect(age_bin, "40-45")            ~ 42.5,
    str_detect(age_bin, "45-50")            ~ 47.5,
    str_detect(age_bin, "50-55")            ~ 52.5,
    str_detect(age_bin, "55-60")            ~ 57.5,
    str_detect(age_bin, "60-65")            ~ 62.5,
    str_detect(age_bin, "65-70")            ~ 67.5,
    str_detect(age_bin, "70-75")            ~ 72.5,
    str_detect(age_bin, "75-80")            ~ 77.5,
    TRUE ~ 90
  )) %>% 
  # mutate(sample_density = sample_count / nbr_individuals) %>%
  mutate(sample_frequency = sample_count / nbr_individuals) %>% 
  mutate(variant_frequency = sample_count / AN_samples_count)
```
We have data on `r gnomad_decoded %>% distinct(IDs) %>% nrow` DNMT3A variants.  
<br>

The dictionary gives the "reference" distribution =  overall distribution of sample in gnomAD per bins: 2547|3423|4546|8487|10355|12693|11933|10534|8882|5991|4136|1935.  
<br>

### Comparing distribution of each variant with the reference:  
I tried to compare the bins distribution with a chi-square test but it is not the best test as we want to look at skewed distribution.  
To realize a t-test and/or Mann-Whitney test: I used a geom_smooth to get a continuous data reflecting the pattern of the bins distribution. I used a one sided t-test/Mann-Whitney test to look at variants which have greater mean than the reference (shifted to the right - older age)

Detailed explanation:
I found a way to create a continuous data from the binned data without assuming for a type of distribution.  
I first plotted a `geom_smooth()` plot which fit the best the binned data and used the `ggplot_build()` function to extract the continuous data set that ggplot used to plot it.  
 
With this continuous data, I was able to do a t-test/Mann-Whitney test.  

```{r chisq skewness}
# goodness_fit1 <- data.frame(matrix(nrow=1, ncol=0)) 
# goodness_fit2 <- data.frame(matrix(nrow=1, ncol=0)) 
# skew <- data.frame(matrix(nrow=1, ncol=0)) 
# 
# for(i in unique(gnomad_decoded$IDs)) {
#   
#   All <- gnomad_decoded %>% filter(IDs == "All individuals") %>% select(sample_count)
#   
#   a <- bind_cols(gnomad_decoded %>% filter(IDs == i) %>% select(sample_count) , All) %>% 
#     `colnames<-`(c(i, "All"))
#   result <- chisq.test(a[,i], p=a$All, rescale.p=TRUE)$p.value
#   goodness_fit1 <- rbind(goodness_fit1, result)
#   
#   result <- ks.test(a[,i], a$All)$p.value
#   goodness_fit2 <- rbind(goodness_fit2, result)
#   
#   result <- skewness(a[[i]], type = 2)
#   skew <- rbind(skew, result)
#   
# }
# df <- bind_cols(goodness_fit1, goodness_fit2, skew) %>% `colnames<-`(c("chisq", "kolmogorov", "skewness"))
# df$IDs <- c(unique(gnomad_decoded$IDs))
# 
# # Bind the distribution characteristic values
# gnomad_decoded1 <- full_join(gnomad_decoded, df, by = "IDs") #%>% 
#   # filter the significatant different
#   # filter((chisq > 0.05 | kolmogorov > 0.05) )
```


```{r loop to calculate stats on the age distribution}
test_mean <- data.frame(matrix(nrow=1, ncol=0)) 
mann_w <-  data.frame(matrix(nrow=1, ncol=0)) 
set.seed(1234)
for(i in unique(gnomad_decoded$IDs)) {
  
  All <- gnomad_decoded %>% 
    filter(IDs == "All individuals"
  ) %>% 
    mutate(IDs = "Reference")
  data_plot <- gnomad_decoded %>% 
    filter(IDs == i
    ) %>% 
    bind_rows(., All) %>% 
    select("IDs", "age_bin", sample_frequency, bin_age)
  
  p <- data_plot %>% 
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    # ggplot(aes(x = age_bin, y= sample_frequency, color= IDs))+
    # geom_bar(stat = "identity", aes(alpha= IDs))
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                # method.args=list(family=quasibinomial)#,
                span = 0.6
    )+
    ylim(0, max(data_plot$sample_frequency))
  p_ <- layer_data(p, 1) %>% 
  mutate(colour =  as.factor(colour)) %>% 
  mutate_if(is.numeric, ~replace(., is.na(.), 0))

  variant <- p_ %>% filter(group == 1)
  variant <- sample(variant$x, 180, prob = variant$y, replace = TRUE)
  ref <- p_ %>% filter(group == 2)
  ref <- sample(ref$x, 180, prob = ref$y, replace = TRUE)
  
  result <- wilcox.test(variant, ref, alternative = "greater")$p.value
  mann_w <- rbind(mann_w, result)

  result <- t.test(variant, ref, alternative = "greater")$p.value
  test_mean <- rbind(test_mean, result)

}
test_mean <- test_mean %>% `colnames<-`(c("t_test")) %>% 
  mutate(IDs = c(unique(gnomad_decoded$IDs)))
mann_w <- mann_w %>% `colnames<-`(c("mann_w")) %>% 
  mutate(IDs = c(unique(gnomad_decoded$IDs)))

# Bind the distribution characteristic values
gnomad_decoded1 <- full_join(gnomad_decoded, test_mean, by = "IDs") %>%
  full_join(., mann_w, by = "IDs") %>%
  # filter the significatant different
  filter(t_test > 0)

```

<!-- After selecting the samples with a skewed distribution, the number of DNMT3A variants is `r gnomad_decoded1 %>% distinct(IDs) %>% nrow`. For now, I kept all variants (see results in part II).    -->

<!-- ## 2. Filter variants with low sample size? -->
<!-- I extracted AN which is the “Total number of alleles in samples” (from data dictionary). -->

<!-- (From GnomAD) "It is the denominator for allele frequency calculations (AN is allele number), which is 2x the number of individuals in gnomAD with either reference or alternate alleles at the variant site." -->
<!-- ```{r} -->

<!-- ``` -->

<!-- ```{r, fig.width=12} -->
<!-- gnomad_decoded1 %>%  -->
<!--   distinct(X.CHROM, POS, .keep_all = TRUE) %>%  -->
<!--   ggplot(aes(x= AN_samples_count, y= t_test, color = t_test <= 0.05))+ -->
<!--   geom_point()+ -->
<!--   ylim(c(0,1))+ -->
<!--   scale_y_log10()+ -->
<!--   theme_minimal() -->
<!-- gnomad_decoded1 %>%  -->
<!--   distinct(X.CHROM, POS, .keep_all = TRUE) %>%  -->
<!--   ggplot(aes(x= AN_samples_count, y= mann_w, color = mann_w <= 0.05))+ -->
<!--   geom_point()+ -->
<!--   ylim(c(0,1))+ -->
<!--   scale_y_log10()+ -->
<!--   theme_minimal() -->

<!-- gnomad_decoded1 %>%  -->
<!--   distinct(X.CHROM, POS, .keep_all = TRUE) %>%  -->
<!--   ggplot(aes(x= nbr_individuals, y= t_test, color = t_test <= 0.05))+ -->
<!--   geom_point()+ -->
<!--   ylim(c(0,1))+ -->
<!--   labs(x= "nbr of individuals for each variants")+ -->
<!--   scale_y_log10()+ -->
<!--   scale_x_continuous(trans = "log2")+ -->
<!--   theme_minimal() -->
<!-- gnomad_decoded1 %>%  -->
<!--   distinct(X.CHROM, POS, .keep_all = TRUE) %>%  -->
<!--   ggplot(aes(x= nbr_individuals, y= mann_w, color = mann_w <= 0.05))+ -->
<!--   geom_point()+ -->
<!--   ylim(c(0,1))+ -->
<!--   labs(x= "nbr of individuals for each variants")+ -->
<!--   scale_y_log10()+ -->
<!--   scale_x_continuous(trans = "log2")+ -->
<!--   theme_minimal() -->

<!-- gnomad_decoded1 %>%  -->
<!--   ggplot(aes(x= nbr_individuals, y= AN_samples_count, color = t_test <= 0.05))+ -->
<!--   geom_point()+ -->
<!--   labs(x= "nbr of individuals for each variants", -->
<!--        y= "total number of samples")+ -->
<!--   # scale_y_sqrt()+ -->
<!--   scale_x_continuous(trans = "log2")+ -->
<!--   theme_minimal() -->
<!-- gnomad_decoded1 %>%  -->
<!--   ggplot(aes(x= nbr_individuals, y= AN_samples_count, color = mann_w <= 0.05))+ -->
<!--   geom_point()+ -->
<!--   labs(x= "nbr of individuals for each variants", -->
<!--        y= "total number of samples")+ -->
<!--   # scale_y_sqrt()+ -->
<!--   scale_x_continuous(trans = "log2")+ -->
<!--   theme_minimal() -->

<!-- gnomad_decoded1 %>%  -->
<!--   distinct(X.CHROM, POS, .keep_all = TRUE) %>%  -->
<!--   select(AN_samples_count) %>%  -->
<!--   tbl_summary() -->

<!-- gnomad_decoded1 %>%  -->
<!--   distinct(X.CHROM, POS, .keep_all = TRUE) %>%  -->
<!--   select(AN_samples_count) %>%  -->
<!--   tbl_summary(statistic = list(all_continuous() ~ "{mean} ({min} - {max})") -->
<!--   ) -->

<!-- gnomad_decoded1 %>%  -->
<!--   distinct(X.CHROM, POS, .keep_all = TRUE) %>%  -->
<!--   select(nbr_individuals) %>%  -->
<!--   tbl_summary() -->

<!-- gnomad_decoded1 %>%  -->
<!--   distinct(X.CHROM, POS, .keep_all = TRUE) %>%  -->
<!--   select(nbr_individuals) %>%  -->
<!--   tbl_summary(statistic = list(all_continuous() ~ "{mean} ({min} - {max})") -->
<!--   ) -->
<!-- ``` -->

```{r}
# gnomad_decoded2a <- gnomad_decoded1 %>% 
#   mutate(ens_vep = str_match(INFO, ";vep=(.*?)$")[,2])
# gnomad_decoded2b <- gnomad_decoded2a %>% 
#   separate(col = ens_vep, paste("ens_vep", 1:25, sep=""),
#            sep = ",", remove = T, extra = "warn", fill = "right") %>% 
#   keep(~!all(is.na(.))) %>% 
#   pivot_longer(cols = starts_with("ens_vep"), names_to = NULL, values_to = "ens_vep") %>% 
#   drop_na(ens_vep)
# 
# ens_vep_var_names <- c("Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type", "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp", "
#   cDNA_position", "CDS_position", "Protein_position", "Amino_acids", "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "
#   STRAND", "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID", "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "
#   TREMBL", "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET", "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "
#   SAS_MAF", "AA_MAF", "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", "ExAC_EAS_MAF", "ExAC_FIN_MAF", "
#   ExAC_NFE_MAF", "ExAC_OTH_MAF", "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME", "MOTIF_POS", "
#   HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter", "LoF_flags", "LoF_info")
# 
# gnomad_decoded3 <- gnomad_decoded2b %>% 
#   separate(col = ens_vep, into = ens_vep_var_names,
#            # paste("ens_vep", 1:100, sep=""),
#            sep = "\\|", remove = F, extra = "warn", fill = "right")
```

<!-- <br> -->
<!-- <br> -->

<!-- *** -->

<!-- <br> -->

<!-- ## 3. Filtering out Consequence  -->
<!-- We could filtered out `Consequence` in the 3_prime_UTR_variant or 5_prime_UTR_variant.   -->
<!-- I extracted the VEP annotations data from `INFO`.   -->
<!-- But you can see that in VEP, we have multiple Variant Effect Predictor separated by a comma. -->
```{r}
# # gnomad_decoded2 <- gnomad_decoded1 %>% 
# #   mutate(ens_vep = str_match(INFO, ";vep=(.*?)$")[,2])
# 
# head(gnomad_decoded2a %>% select(ens_vep))[1:3,]
```
<!-- <br> -->

<!-- Here an example of `Consequence` on gnomAD browser -->
```{r, out.width='100%'}
# knitr::include_graphics('/Users/colinccm/Documents/GitHub/Gillis/CH_gnomAD/variant effect predictor.jpg')
```
<!-- <br> -->


<!-- I made 1 row for each Variant Effect Predictor so multiple rows per variants. -->
```{r}
# gnomad_decoded2 <- gnomad_decoded2 %>% 
#   separate(col = ens_vep, paste("ens_vep", 1:25, sep=""),
#            sep = ",", remove = T, extra = "warn", fill = "right") %>% 
#   keep(~!all(is.na(.))) %>% 
#   pivot_longer(cols = starts_with("ens_vep"), names_to = NULL, values_to = "ens_vep") %>% 
#   drop_na(ens_vep)

# head(gnomad_decoded2 %>% select(IDs, ens_vep))
```

<!-- VEP contains these information :   -->
<!-- `"Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type", "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp", "` -->
<!--   `cDNA_position", "CDS_position", "Protein_position", "Amino_acids", "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "` -->
<!--   `STRAND", "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID", "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "` -->
<!--   `TREMBL", "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET", "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "` -->
<!--   `SAS_MAF", "AA_MAF", "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", "ExAC_EAS_MAF", "ExAC_FIN_MAF", "` -->
<!--   `ExAC_NFE_MAF", "ExAC_OTH_MAF", "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME", "MOTIF_POS", "` -->
<!--   `HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter", "LoF_flags", "LoF_info"`   -->
```{r}
# Consequence annotations from Ensembl VEP
# Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|
# cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|ALLELE_NUM|DISTANCE|
# STRAND|FLAGS|VARIANT_CLASS|MINIMISED|SYMBOL_SOURCE|HGNC_ID|CANONICAL|TSL|APPRIS|CCDS|ENSP|SWISSPROT|
# TREMBL|UNIPARC|GENE_PHENO|SIFT|PolyPhen|DOMAINS|HGVS_OFFSET|GMAF|AFR_MAF|AMR_MAF|EAS_MAF|EUR_MAF|
# SAS_MAF|AA_MAF|EA_MAF|ExAC_MAF|ExAC_Adj_MAF|ExAC_AFR_MAF|ExAC_AMR_MAF|ExAC_EAS_MAF|ExAC_FIN_MAF|
# ExAC_NFE_MAF|ExAC_OTH_MAF|ExAC_SAS_MAF|CLIN_SIG|SOMATIC|PHENO|PUBMED|MOTIF_NAME|MOTIF_POS|
# HIGH_INF_POS|MOTIF_SCORE_CHANGE|LoF|LoF_filter|LoF_flags|LoF_info
# ens_vep_var_names <- c("Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type", "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp", "
#   cDNA_position", "CDS_position", "Protein_position", "Amino_acids", "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "
#   STRAND", "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID", "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "
#   TREMBL", "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET", "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "
#   SAS_MAF", "AA_MAF", "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", "ExAC_EAS_MAF", "ExAC_FIN_MAF", "
#   ExAC_NFE_MAF", "ExAC_OTH_MAF", "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME", "MOTIF_POS", "
#   HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter", "LoF_flags", "LoF_info")
# 
# gnomad_decoded3 <- gnomad_decoded2 %>% 
#   separate(col = ens_vep, into = ens_vep_var_names,
#            # paste("ens_vep", 1:100, sep=""),
#            sep = "\\|", remove = F, extra = "warn", fill = "right")

# head(gnomad_decoded3 %>% select(IDs, "Allele": "LoF_info"))
```

<!-- What are the ones you are interested in (I can keep them all of course) and how can I filter the variant we want.   -->
<!-- There are multiple types of `Consequence`. -->
```{r}
# table(gnomad_decoded3$Consequence)

# table(gnomad_decoded3$SYMBOL)

gnomad_decoded4 <- gnomad_decoded1 # %>% 
  # filter(!str_detect(Consequence, "3_prime_UTR_variant|5_prime_UTR_variant") #| IDs == "All individuals"#, SYMBOL == "DNMT3A"
  #        )
```
<!-- We could select the variants not in 3 and 5' for example -->
<!-- Let me know if I am correct and if I can be more selective.   -->
<!-- But we have the same thing for HGVSc, HGVSp (I think we only keep the HGVSp...) -->

<br>
<br>

***

<br>

# II. Distribution plots

In here, I show the distribution of all the variants in DNMT3A|TET2|TP53|ASXL1|PPM1D genes

In red is the distribution of each variant. In teal is the reference distribution. The Mann-Whitney and T test p values of comparing each variant age distribution to the reference population age distribution are indiacated on the plot. The number of individuals who express this variant is also indicated on the plot.
```{r}
gnomad_decoded4 <- gnomad_decoded4 %>% 
  mutate(gene = str_match(gnomad_decoded4$INFO, "(DNMT3A|TET2|TP53|ASXL1|PPM1D)(.*?)$")[,2])
```

```{r, fig.height= 3, fig.width=7}
data_p <- gnomad_decoded4 %>% 
  bind_rows(., gnomad_decoded1 %>% filter(IDs == "All individuals"))
```

```{r, fig.height= 3, fig.width=7}
n <- 0
for(i in unique(data_p$IDs)) {
  
  All <- data_p %>% 
    filter(IDs == "All individuals"
           ) %>% 
    mutate(IDs = "Reference")
  data_plot <- data_p %>% 
    filter(IDs == i
           )
  n <- n + 1
  p <- data_plot %>% 
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                  # method.args=list(family=quasibinomial)#, 
                span = 0.6
                )+
    labs(title = n)+
    ylim(0, max(data_plot$sample_frequency))+
    geom_smooth(data = All, aes(x = bin_age, y= sample_frequency, color= IDs), 
                alpha= 0.5, se = FALSE, method = "loess",
                  # method.args=list(family=quasibinomial)#,
                span = 0.6
                )+
    geom_col(data = data_plot ,aes(x= bin_age, y= sample_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("t test = ", round(t_test, 3))),
      hjust   = 0,
      vjust   = -9
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("MannW=", round(mann_w, 3))),
      hjust   = 0,
      vjust   = -5
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals=", nbr_individuals)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$sample_frequency)
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("gene=", gene)),
      colour = "darkgreen",
      hjust   = -3,
      vjust   = 0
    )+
    theme_minimal()
  print(p)

}
```
<br>
<br>

***

# III.Selected variants with Mann-Whitney ≤ 0.1
## Data
Our selected variants are:
```{r}
selected_variants <- gnomad_decoded4 %>%
  filter(mann_w <= 0.1)

selected_variants
```
`r length(unique(selected_variants$IDs))` variants are selected with a skewed age distribution (Mann-Whitney p-val ≤ 0.1).  

<br>

## Plots
```{r, fig.height= 3, fig.width=7}

n <- 0
for(i in unique(selected_variants$IDs)) {

  All <- gnomad_decoded4 %>% 
    filter(IDs == "All individuals"
           ) %>% 
    mutate(IDs = "Reference")
  data_plot <- selected_variants %>%
    filter(IDs == i
           )
  n <- n + 1
  p <- data_plot %>%
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
                )+
    labs(title = n)+
    ylim(0, max(data_plot$sample_frequency))+
    geom_smooth(data = All, aes(x = bin_age, y= sample_frequency, color= IDs),
                alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
                )+
    geom_col(data = data_plot ,aes(x= bin_age, y= sample_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, 
                    label = paste0("t test = ", round(t_test, 3))),
      hjust   = 0,
      vjust   = -9
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, 
                    label = paste0("MannW=", round(mann_w, 3))),
      hjust   = 0,
      vjust   = -5
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals=", nbr_individuals)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$sample_frequency)
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("gene=", gene)),
      colour = "darkgreen",
      hjust   = -3,
      vjust   = 0
    )+
    theme_minimal()
  print(p)

}

selected_variants <- selected_variants %>% 
  distinct(IDs, .keep_all = TRUE)
```
<br>
<br>

***

<br>

# IV. Cosmic data

```{r load cosmic}
# From Annovar
# annovar_cosmic_data <-
#   read.delim(paste0(path, "/cosmic_raw_data/hg38_cosmic70.txt"))

# cosmic1000 <-
#   read.delim(paste0(path, "/cosmic_raw_data/cosmic_head1000_data.txt"))

cosmic <-
  read.delim(paste0(path, "/cosmic_raw_data/cosmic_DNMT3_data.vcf.gz"), 
             header = FALSE, 
             col.names = c("Gene.name", "Accession.Number", "Gene.CDS.length", "HGNC.ID", "Sample.name", 
                           "ID_sample", "ID_tumour", "Primary.site", "Site.subtype.1", "Site.subtype.2", 
                           "Site.subtype.3", "Primary.histology", "Histology.subtype.1", "Histology.subtype.2", 
                           "Histology.subtype.3", "Genome.wide.screen", "GENOMIC_MUTATION_ID", 
                           "LEGACY_MUTATION_ID", "MUTATION_ID", "Mutation.CDS", "Mutation.AA", 
                           "Mutation.Description", "Mutation.zygosity", "LOH", "GRCh", 
                           "Mutation.genome.position", "Mutation.strand", "Resistance.Mutation", 
                           "FATHMM.prediction", "FATHMM.score", "Mutation.somatic.status", "Pubmed_PMID", 
                           "ID_STUDY", "Sample.Type", "Tumour.origin", "Age", "HGVSP", "HGVSC", "HGVSG"))

classification <-
  read_csv(paste0(path, "/cosmic_raw_data/classification.csv"))
```

As an example I selected all the variant DNMT3A. Here is a example of the raw data.
```{r}
head(cosmic)
```
As you can see we have multiple rows for 1 mutation (the data shows 1 sample per row) so I calculated the occurrence of each mutation and pivot the data wider to have only 1 mutation per row before to bind with gnomAD.  

<br>

```{r}
cosmic <-
  cosmic %>% 
  separate(HGVSG, into = c("chr", "chr_arm", "POS", "ALT"), sep = ":|\\.|>|delins|dup|del", remove = FALSE) %>% 
  mutate(REF = str_match(POS, "([0-9]*)([A-Z]*)")[,3]) %>% 
  mutate(POS = str_match(POS, "([0-9]*)([A-Z]*)")[,2],
         POS = as.numeric(POS)) %>% 
  # separate(Mutation.genome.position, into = c("chr", "start", "end"), sep = ":|-", remove = FALSE) %>% 
  unite(IDs, c(chr, POS, REF, ALT), sep = "-", remove = FALSE) %>% 
  # separate(HGVSG, into = c("ENSP", "p_prot_modif"), sep = ":", remove = FALSE) %>% 
  mutate(IDs = case_when(
    IDs == "-NA-NA-NA"                 ~ NA_character_,
    TRUE                               ~ IDs
  )) %>% 
  separate(`Gene.name`, into = c("gene_name_cosmic"), sep = "_", remove = FALSE) %>% 
  arrange(IDs) %>% 
  select(IDs, "Gene.name", gene_name_cosmic, Gene.name, "Accession.Number", "chr", "POS", REF, "ALT", HGVSG, everything())
```

```{r}
# library(data.table)
# final_cosmic <-
#   dcast(setDT(cosmic1), IDs+chr+POS+REF+ALT ~ rowid(IDs),
#         value.var = c(
#           "Mutation.CDS",
#           "Mutation.AA",
#           "Primary.site",
#           "Primary.histology",
#           "Histology.subtype.1", "Histology.subtype.2", "Histology.subtype.3",
#           "Mutation.Description",
#           "FATHMM.prediction", "FATHMM.score",
#           "Mutation.somatic.status", "Pubmed_PMID",
#           "Sample.Type", "Tumour.origin", "Age"
#         )
#   )

final_cosmic <- cosmic %>% 
  group_by(IDs, gene_name_cosmic, chr, POS, REF, ALT) %>% 
  mutate(occurrence_in_cosmic = n())

max_occurence = max(final_cosmic$occurrence_in_cosmic)
final_cosmic <- final_cosmic %>% 
  group_by(IDs, gene_name_cosmic, chr, POS, REF, ALT, HGVSP, occurrence_in_cosmic) %>% 
  summarise_at(vars(Mutation.CDS,
          Mutation.AA,
          Primary.site,
          Primary.histology,
          Histology.subtype.1, Histology.subtype.2, Histology.subtype.3,
          Mutation.Description,
          FATHMM.prediction, FATHMM.score,
          Mutation.somatic.status, Pubmed_PMID,
          Sample.Type, Tumour.origin, Age), str_c, collapse = "; ") %>% 
  ungroup() %>% 
  separate(Age, paste("Age", 1:max(max_occurence), sep = "_"), 
           sep = "; ", remove = TRUE, 
           extra = "warn", fill = "right") %>% 
  purrr::keep(~!all(is.na(.))) %>% 
  mutate(across(starts_with("Age_"), ~ as.numeric(.))) %>% 
  
  mutate(average_age = rowMeans(select(., starts_with("Age")), na.rm = TRUE))
```

`r emo::ji("warning")`But we also have multiple `HGVSP` for a single `chr, POS, REF, ALT`. It can be a problem if we bind by IDs...  
```{r}
(final_cosmic %>% select(IDs, HGVSP, occurrence_in_cosmic))[61:67,]
```

<span style="color:red">**What are the variable you want in cosmic?**</span>
```{r}
colnames(cosmic)
```

What are the variable you want in cosmic classification?
```{r}
colnames(classification)
head(classification)
```

# V.Bind with cosmic by : chr, POS, REF, ALT
## Data
```{r bind cosmic Ids}
CH_variants_cpra <- inner_join(selected_variants %>% 
                  select(-c(AN:t_test)), 
                final_cosmic, 
          by = c("IDs", "X.CHROM" = "chr", "POS", "REF", "ALT"))
CH_variants_cpra %>% select(IDs, HGVSP, everything())
```
We have `r length(unique(CH_variants_cpra$IDs))` but with at total of `r length(unique(CH_variants_cpra$HGVSP))` HGVSP.

## Plots
```{r}
ids_plot <- paste0(CH_variants_cpra$IDs, collapse = "|")

ids_plot <- gnomad_decoded4 %>%
  filter(str_detect(IDs, ids_plot))

n <- 0
for(i in unique(ids_plot$IDs)) {

  All <- gnomad_decoded4 %>% 
    filter(IDs == "All individuals"
           ) %>% 
    mutate(IDs = "Reference")
  data_plot <- ids_plot %>%
    filter(IDs == i
           )
  n <- n + 1
  p <- data_plot %>%
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
                )+
    labs(title = n)+
    ylim(0, max(data_plot$sample_frequency))+
    geom_smooth(data = All, aes(x = bin_age, y= sample_frequency, color= IDs),
                alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
                )+
    geom_col(data = data_plot ,aes(x= bin_age, y= sample_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, 
                    label = paste0("t test = ", round(t_test, 3))),
      hjust   = 0,
      vjust   = -9
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, 
                    label = paste0("MannW=", round(mann_w, 3))),
      hjust   = 0,
      vjust   = -5
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals=", nbr_individuals)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$sample_frequency)
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("gene=", gene)),
      colour = "darkgreen",
      hjust   = -3,
      vjust   = 0
    )+
    theme_minimal()
  print(p)

}
```

# VI.Bind with cosmic by : protein change unavailable ~ p.
Protein change is available in the VEP info of gnomAD.

## 3. Extract VEP Consequence
I extracted the VEP annotations data from gnomAD `INFO`.
But you can see that in VEP, we have multiple Variant Effect Predictor separated by a comma.
```{r}
selected_variants1 <- selected_variants %>%
  mutate(ens_vep = str_match(INFO, ";vep=(.*?)$")[,2])

head(selected_variants1 %>% select(ens_vep))[1,]
```
<!-- <br> -->

<!-- Here an example of `Consequence` on gnomAD browser -->
```{r, out.width='100%'}
# knitr::include_graphics('/Users/colinccm/Documents/GitHub/Gillis/CH_gnomAD/variant effect predictor.jpg')
```
<!-- <br> -->


<!-- I made 1 row for each Variant Effect Predictor so multiple rows per variants. -->
```{r}
selected_variants1 <- selected_variants1 %>%
  separate(col = ens_vep, paste("ens_vep", 1:25, sep=""),
           sep = ",", remove = T, extra = "warn", fill = "right") %>%
  keep(~!all(is.na(.))) %>%
  pivot_longer(cols = starts_with("ens_vep"), names_to = NULL, values_to = "ens_vep") %>%
  drop_na(ens_vep)

head(selected_variants1 %>% select(IDs, ens_vep))
```

VEP contains these information :
`"Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type", "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp", "`
  `cDNA_position", "CDS_position", "Protein_position", "Amino_acids", "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "`
  `STRAND", "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID", "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "`
  `TREMBL", "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET", "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "`
  `SAS_MAF", "AA_MAF", "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", "ExAC_EAS_MAF", "ExAC_FIN_MAF", "`
  `ExAC_NFE_MAF", "ExAC_OTH_MAF", "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME", "MOTIF_POS", "`
  `HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter", "LoF_flags", "LoF_info"`
```{r}
# Consequence annotations from Ensembl VEP
# Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|
# cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|ALLELE_NUM|DISTANCE|
# STRAND|FLAGS|VARIANT_CLASS|MINIMISED|SYMBOL_SOURCE|HGNC_ID|CANONICAL|TSL|APPRIS|CCDS|ENSP|SWISSPROT|
# TREMBL|UNIPARC|GENE_PHENO|SIFT|PolyPhen|DOMAINS|HGVS_OFFSET|GMAF|AFR_MAF|AMR_MAF|EAS_MAF|EUR_MAF|
# SAS_MAF|AA_MAF|EA_MAF|ExAC_MAF|ExAC_Adj_MAF|ExAC_AFR_MAF|ExAC_AMR_MAF|ExAC_EAS_MAF|ExAC_FIN_MAF|
# ExAC_NFE_MAF|ExAC_OTH_MAF|ExAC_SAS_MAF|CLIN_SIG|SOMATIC|PHENO|PUBMED|MOTIF_NAME|MOTIF_POS|
# HIGH_INF_POS|MOTIF_SCORE_CHANGE|LoF|LoF_filter|LoF_flags|LoF_info
ens_vep_var_names <- c("Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type", "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp", "
  cDNA_position", "CDS_position", "Protein_position", "Amino_acids", "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "
  STRAND", "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID", "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "
  TREMBL", "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET", "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "
  SAS_MAF", "AA_MAF", "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF", "ExAC_EAS_MAF", "ExAC_FIN_MAF", "
  ExAC_NFE_MAF", "ExAC_OTH_MAF", "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME", "MOTIF_POS", "
  HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter", "LoF_flags", "LoF_info")

selected_variants1 <- selected_variants1 %>%
  separate(col = ens_vep, into = ens_vep_var_names,
           # paste("ens_vep", 1:100, sep=""),
           sep = "\\|", remove = F, extra = "warn", fill = "right")

head(selected_variants1 %>% select(IDs, HGVSp, "Allele": "LoF_info"))
```

## Extract p. from HGVSp
`r emo::ji("warning")` It is some protein change info unavailable.
```{r}
# VEP_extracted <- gnomad_decoded3 %>% 
#   mutate(protein_change_nomenclature = str_match(HGVSp, ":(.*?)$")[,2])
```

## Data
```{r bind cosmic p}
CH_variants_p <- inner_join(selected_variants1 %>% 
                  filter(HGVSp != "") %>%
                  select(-c(AN:t_test)),
                final_cosmic %>% 
                  filter(HGVSP != ""),
          by = c("HGVSp" = "HGVSP"))

# write_rds(CH_variants_p, "CH_variants_p.rds")
# CH_variants_p <- read_rds("/Users/colinccm/Documents/GitHub/Gillis/CH_gnomAD/CH_variants_p.rds")
```


```{r bind cosmic p head}
head(CH_variants_p %>% select(HGVSp, starts_with("IDs"), X.CHROM, chr, starts_with("POS"), starts_with("REF"), starts_with("ALT"), everything()))
```
We have `r length(unique(CH_variants_p$IDs.x))` but with at total of `r length(unique(CH_variants_p$HGVSp))` HGVSP.
<span style="color:red">**A protein change can be acquired by different mutation.**</span>

## Plots
```{r plot CH_variants_p}
CH_variants_p <- CH_variants_p %>% 
  distinct(IDs.x, .keep_all = TRUE)
```

```{r}
ids_plot <- paste0(CH_variants_p$IDs.x, collapse = "|")

ids_plot <- gnomad_decoded4 %>%
  filter(str_detect(IDs, ids_plot))

n <- 0
for(i in unique(ids_plot$IDs)) {

  All <- gnomad_decoded4 %>% 
    filter(IDs == "All individuals"
           ) %>% 
    mutate(IDs = "Reference")
  data_plot <- ids_plot %>%
    filter(IDs == i
           )
  n <- n + 1
  p <- data_plot %>%
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
                )+
    labs(title = n)+
    ylim(0, max(data_plot$sample_frequency))+
    geom_smooth(data = All, aes(x = bin_age, y= sample_frequency, color= IDs),
                alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
                )+
    geom_col(data = data_plot ,aes(x= bin_age, y= sample_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, 
                    label = paste0("t test = ", round(t_test, 3))),
      hjust   = 0,
      vjust   = -9
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, 
                    label = paste0("MannW=", round(mann_w, 3))),
      hjust   = 0,
      vjust   = -5
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals=", nbr_individuals)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$sample_frequency)
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("gene=", gene)),
      colour = "darkgreen",
      hjust   = -3,
      vjust   = 0
    )+
    theme_minimal()
  print(p)

}
```
<br>
<br>

```{r}
rm(CH_variants_p)
```


***

<br>

# VII. Extract cancer characteristics for selected variants USING `chr+POS+REF+ALT` binding
We want to extract primary site and histology of the tumor for each mutation in cosmic.  

Need to use either of those id in CosmicMutantExport: 
Sample.name = RK131_C01, ID_sample = 2120979, ID_tumour = 1995215, GENOMIC_MUTATION_ID = COSV74182898, LEGACY_MUTATION_ID = COSM or COSN, MUTATION_ID = 100426914

We could use CosmicSample which has:
phenotype_id = 32666056, sample_id = 732412, sample_name = cor-19, id_tumor = 656400, id_individual = 637439

CosmicMutationTracking has MUTATION_ID = 139363701, LEGACY_MUTATION_ID = COSM or COSN, GENOMIC_MUTATION_ID = COSV

classification only has phenotype_id

All have primary site etc...
```{r}
CosmicSample <-
  read.delim(paste0(path, "/cosmic_raw_data/example_grch38_cosmic/CosmicSample.tsv"))

inner_join(CH_variants_cpra, CosmicSample, by = c("Sample.name" = "sample_name"))



CosmicMutationTracking <-
  read.delim(paste0(path, "/cosmic_raw_data/example_grch38_cosmic/CosmicMutationTracking.tsv"))

inner_join(CH_variants_cpra, CosmicMutationTracking, by = "GENOMIC_MUTATION_ID")
```




# VIII. Extract allele count and frequency variables on the selected variants (code ready for future analysis) USING `chr+POS+REF+ALT` binding
```{r, extract allele count and frequency}
CH_variants <- CH_variants %>%
  # select(c("IDs", "ID", "INFO")) %>%
  distinct(IDs, .keep_all = TRUE) %>%
  # Generate allele count variables from the INFO var
  mutate(alt_allele_count = str_match(INFO, "AC=(.*?);")[,2]) %>%
  mutate(alt_allele_count_afr_female = str_match(INFO, "AC_female=(.*?);")[,2]) %>%
  mutate(alt_allele_count_male = str_match(INFO, "AC_male=(.*?);")[,2]) %>%

  mutate(alt_allele_count_afr = str_match(INFO, "AC_afr=(.*?);")[,2]) %>%
  mutate(alt_allele_count_afr_female = str_match(INFO, "AC_afr_female=(.*?);")[,2]) %>%
  mutate(alt_allele_count_male = str_match(INFO, "AC_afr_male=(.*?);")[,2]) %>%

  mutate(alt_allele_count_afr = str_match(INFO, "AC_amr=(.*?);")[,2]) %>%
  mutate(alt_allele_count_afr_female = str_match(INFO, "AC_amr_female=(.*?);")[,2]) %>%
  mutate(alt_allele_count_male = str_match(INFO, "AC_amr_male=(.*?);")[,2]) %>%

  # Generate allele count variables from the INFO var
  mutate(freq_allele_count = str_match(INFO, "AF=(.*?);")[,2]) %>%
  mutate(freq_allele_count_afr_female = str_match(INFO, "AF_female=(.*?);")[,2]) %>%
  mutate(freq_allele_count_male = str_match(INFO, "AF_male=(.*?);")[,2]) %>%

  mutate(freq_allele_count_afr = str_match(INFO, "AF_afr=(.*?);")[,2]) %>%
  mutate(freq_allele_count_afr_female = str_match(INFO, "AF_afr_female=(.*?);")[,2]) %>%
  mutate(freq_allele_count_male = str_match(INFO, "AF_afr_male=(.*?);")[,2]) %>%

  mutate(freq_allele_count_afr = str_match(INFO, "AF_amr=(.*?);")[,2]) %>%
  mutate(freq_allele_count_afr_female = str_match(INFO, "AF_amr_female=(.*?);")[,2]) %>%
  mutate(freq_allele_count_male = str_match(INFO, "AF_amr_male=(.*?);")[,2]) %>%

  # Depth of informative coverage for each sample
  mutate(freq_allele_count = str_match(INFO, "DP=(.*?);")[,2]) %>%
  # GQ



  # allele type
  mutate(allele_type = str_match(INFO, "allele_type=(.*?);")[,2]) %>%
  # variant type
  mutate(variant_type = str_match(INFO, "variant_type=(.*?);")[,2]) %>%
  # Number=A,Type=String,Description="Population with maximum AF in the controls subset">
  mutate(controls_popmax = str_match(INFO, "controls_popmax=(.*?);")[,2]) %>%


  ## NON CANCER
  # Generate allele count variables from the INFO var
  mutate(nc_alt_allele_count = str_match(INFO, "non_cancer_AC=(.*?);")[,2]) %>%
  mutate(nc_alt_allele_count_afr_female = str_match(INFO, "non_cancer_AC_female=(.*?);")[,2]) %>%
  mutate(nc_alt_allele_count_male = str_match(INFO, "non_cancer_AC_male=(.*?);")[,2]) %>%

  mutate(nc_alt_allele_count_afr = str_match(INFO, "non_cancer_AC_afr=(.*?);")[,2]) %>%
  mutate(nc_alt_allele_count_afr_female = str_match(INFO, "non_cancer_AC_afr_female=(.*?);")[,2]) %>%
  mutate(nc_alt_allele_count_male = str_match(INFO, "non_cancer_AC_afr_male=(.*?);")[,2]) %>%

  mutate(nc_alt_allele_count_afr = str_match(INFO, "non_cancer_AC_amr=(.*?);")[,2]) %>%
  mutate(nc_alt_allele_count_afr_female = str_match(INFO, "non_cancer_AC_amr_female=(.*?);")[,2]) %>%
  mutate(nc_alt_allele_count_male = str_match(INFO, "non_cancer_AC_amr_male=(.*?);")[,2]) %>%

  # Generate allele count variables from the INFO var
  mutate(nc_freq_allele_count = str_match(INFO, "non_cancer_AF=(.*?);")[,2]) %>%
  mutate(nc_freq_allele_count_afr_female = str_match(INFO, "non_cancer_AF_female=(.*?);")[,2]) %>%
  mutate(nc_freq_allele_count_male = str_match(INFO, "non_cancer_AF_male=(.*?);")[,2]) %>%

  mutate(nc_freq_allele_count_afr = str_match(INFO, "non_cancer_AF_afr=(.*?);")[,2]) %>%
  mutate(nc_freq_allele_count_afr_female = str_match(INFO, "non_cancer_AF_afr_female=(.*?);")[,2]) %>%
  mutate(nc_freq_allele_count_male = str_match(INFO, "non_cancer_AF_afr_male=(.*?);")[,2]) %>%

  mutate(nc_freq_allele_count_afr = str_match(INFO, "non_cancer_AF_amr=(.*?);")[,2]) %>%
  mutate(nc_freq_allele_count_afr_female = str_match(INFO, "non_cancer_AF_amr_female=(.*?);")[,2]) %>%
  mutate(nc_freq_allele_count_male = str_match(INFO, "non_cancer_AF_amr_male=(.*?);")[,2]) %>%

  # "Total number of alleles in samples in the non_cancer subset, before removing low-confidence genotypes">
  mutate(nc_freq_allele_count_male = str_match(INFO, "non_cancer_AF_raw=(.*?);")[,2])
```

```{r}
head(CH_variants)
```























