---
title: "Inverstigate chr2 pval"
author: "Christelle Colin-Leitzinger"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: no
    theme: cerulean
    highlight: pygments
    df_print: paged
editor_options:
  chunk_output_type: console
---

<style type="text/css">
h1.title {
  font-size: 25px;
}
.figure {
   margin-top: 25px;
   margin-bottom: 25px;
}

table {
    margin-top: 25px;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.align='center'
                      )
options(gtsummary.print_engine = "gt")
options(gtsummary.as_gt.addl_cmds = "gt::tab_options(table.font.size = 12, data_row.padding = gt::px(1))")
```


<style>
div.blue { background-color:#0099CC; border-radius: 5px; padding: 20px; font-size: 38px}
</style>
<div class = "blue">

<span style="color: white;">GnomAD data</span>

</div>
<br>

```{r library, include=FALSE}
library(tidyverse)
library(effsize)

theme_set(theme_classic())
```

```{r load}
path <- fs::path("", "Volumes", "Lab_Gillis", "Christelle")

gnomad <-
  read.delim(paste0(here::here(), "/chr2_CH_variants.vcf.gz"), 
             # colClasses = c("ALT"="character"),
              sep = " ")

```

```{r}
gnomad1 <- gnomad %>%
  distinct(IDs, .keep_all = TRUE)
# gnomad <- gnomad %>%
#   left_join(., gnomad_pval %>%
#               select(IDs, adjusted_pval),
#             by = "IDs")
# 
# rm(gnomad_pval)
```

<!-- # Mean and effect size -->
<!-- **CH R882H mutation is depicted in red**   -->
<!-- bh adjusted_selected in yellow    -->
<!-- effect size_selected in orange -->
<!-- ```{r plot} -->
<!-- bhadjusted_selected <- paste0(c("2-25244379-G-C",  -->
<!--                                 "2-25244494-C-T", "2-25248052-G-A", "2-25274935-C-T",  -->
<!--                                 "2-25282744-G-A"), -->
<!--                               collapse = "|") -->
<!-- effsize_selected <- paste0(c("2-25234321-G-A", "2-25234374-G-A", -->
<!--                                  "2-25328714-C-G", "2-25328736-G-A"), -->
<!--                               collapse = "|") -->

<!-- gnomad %>%  -->
<!--   select(IDs, nbr_individuals, mann_w, adjusted_pval) %>%  -->
<!--   ggplot(aes(x= adjusted_pval, y= mann_w))+ -->
<!--   geom_point(color= "grey")+ -->
<!--   geom_point(data= gnomad %>% filter(IDs=="2-25234373-C-T"), -->
<!--              aes(x= adjusted_pval, y=mann_w), color= "red")+ -->
<!--   geom_point(data= gnomad %>%  -->
<!--                filter(str_detect(IDs, bhadjusted_selected)), -->
<!--              aes(x= adjusted_pval, y=mann_w), color= "yellow")+ -->
<!--   geom_point(data= gnomad %>%  -->
<!--                filter(str_detect(IDs, effsize_selected)), -->
<!--              aes(x= adjusted_pval, y=mann_w), color= "orange")+ -->
<!--   theme(legend.position = "none") -->

<!-- gnomad %>%  -->
<!--   select(IDs, nbr_individuals, mann_w, adjusted_pval) %>%  -->
<!--   ggplot(aes(x= mann_w, y= nbr_individuals))+ -->
<!--   geom_point(color= "grey")+ -->
<!--   geom_point(data= gnomad %>% filter(IDs=="2-25234373-C-T"), -->
<!--              aes(x= mann_w, y=adjusted_pval), color= "red")+ -->
<!--   geom_point(data= gnomad %>%  -->
<!--                filter(str_detect(IDs, bhadjusted_selected)), -->
<!--              aes(x= mann_w, y=adjusted_pval), color= "yellow")+ -->
<!--   geom_point(data= gnomad %>%  -->
<!--                filter(str_detect(IDs, effsize_selected)), -->
<!--              aes(x= mann_w, y=adjusted_pval), color= "orange")+ -->
<!--   theme(legend.position = "none") -->

<!-- gnomad %>%  -->
<!--   select(IDs, nbr_individuals, mann_w, adjusted_pval) %>%  -->
<!--   ggplot(aes(x= adjusted_pval, y= nbr_individuals))+ -->
<!--   geom_point(color= "grey")+ -->
<!--   geom_point(data= gnomad %>% filter(IDs=="2-25234373-C-T"), -->
<!--              aes(x= adjusted_pval, y=nbr_individuals), color= "red")+ -->
<!--   geom_point(data= gnomad %>%  -->
<!--                filter(str_detect(IDs, bhadjusted_selected)), -->
<!--              aes(x= adjusted_pval, y=nbr_individuals), color= "yellow")+ -->
<!--   geom_point(data= gnomad %>%  -->
<!--                filter(str_detect(IDs, effsize_selected)), -->
<!--              aes(x= adjusted_pval, y=nbr_individuals), color= "orange")+ -->
<!--   theme(legend.position = "none") -->
<!-- ``` -->

```{r plot1}
# All <- gnomad %>% 
#   filter(IDs == "All individuals"
#   ) %>% 
#   # mutate(IDs = "Reference") %>% 
#   select(distribution_mean_ref = distribution_mean)
# 
# data_plot <- gnomad %>%
#   # filter(IDs == i
#   # ) %>% 
#   select("IDs", distribution_mean, adjusted_pval) %>% 
#   bind_cols(., All) %>% 
#   mutate(effect = distribution_mean - distribution_mean_ref) # need to keep sup values

# data_plot %>% 
#   slice(1:20) %>% 
#   ggplot(aes(x = IDs, y= effect, color= effect > 0
#              #, color= IDs == "All individuals"
#   ))+
#   geom_point()+
#   geom_hline(yintercept = 0)+
#   coord_flip()
```

<!-- On all the variants -->
```{r plot3}
# library(ggforce)
# data_plot %>% 
#   ggplot(aes(x= effect, y= adjusted_pval))+
#   geom_point(color= "grey")+
#   geom_point(data= data_plot %>% filter(IDs=="2-25234373-C-T"),
#              aes(x= effect, y=adjusted_pval), color= "red")+
#   geom_point(data= data_plot %>% 
#                filter(str_detect(IDs, bhadjusted_selected)),
#              aes(x= effect, y=adjusted_pval), color= "yellow")+
#   geom_point(data= data_plot %>% 
#                filter(str_detect(IDs, effsize_selected)),
#              aes(x= effect, y=adjusted_pval), color= "orange")+
#   scale_y_log10()+
#   labs(y= "BH pval (log10)")+
#   theme(legend.position = "none")
# 
# data_plot %>% 
#   ggplot(aes(x= effect, y= adjusted_pval))+
#   geom_point(color= "grey")+
#   geom_point(data= data_plot %>% filter(IDs=="2-25234373-C-T"),
#              aes(x= effect, y=adjusted_pval), color= "red")+
#   geom_point(data= data_plot %>% 
#                filter(str_detect(IDs, bhadjusted_selected)),
#              aes(x= effect, y=adjusted_pval), color= "yellow")+
#   geom_point(data= data_plot %>% 
#                filter(str_detect(IDs, effsize_selected)),
#              aes(x= effect, y=adjusted_pval), color= "orange")+
#   scale_y_continuous(trans = trans_reverser('log10'))+
#   # scale_y_continuous(trans = "log2")+
#   labs(y= "BH pval (log10)")+
#   theme(legend.position = "none")
```

<!-- On BH < 0.05 -->
```{r plot4}
# library(ggforce)
# data_plot %>% 
#   filter(adjusted_pval < 0.05) %>% 
#   ggplot(aes(x= effect, y= adjusted_pval))+
#   geom_point(color= "grey")+
#   geom_point(data= data_plot %>% filter(IDs=="2-25234373-C-T"),
#              aes(x= effect, y=adjusted_pval), color= "red")+
#   geom_point(data= data_plot %>% 
#                filter(str_detect(IDs, bhadjusted_selected)),
#              aes(x= effect, y=adjusted_pval), color= "yellow")+
#   geom_point(data= data_plot %>% 
#                filter(str_detect(IDs, effsize_selected)),
#              aes(x= effect, y=adjusted_pval), color= "orange")+
#   scale_y_log10()+
#   labs(y= "BH pval (log10)")+
#   theme(legend.position = "none")
# 
# data_plot %>% 
#   filter(adjusted_pval < 0.05) %>% 
#   ggplot(aes(x= effect, y= adjusted_pval))+
#   geom_point(color= "grey")+
#   geom_point(data= data_plot %>% filter(IDs=="2-25234373-C-T"),
#              aes(x= effect, y=adjusted_pval), color= "red")+
#   geom_point(data= data_plot %>% 
#                filter(str_detect(IDs, bhadjusted_selected)),
#              aes(x= effect, y=adjusted_pval), color= "yellow")+
#   geom_point(data= data_plot %>% 
#                filter(str_detect(IDs, effsize_selected)),
#              aes(x= effect, y=adjusted_pval), color= "orange")+
#   scale_y_continuous(trans = 'log2')+
#   # scale_y_continuous(trans = "log2")+
#   labs(y= "BH pval (log2)")+
#   theme(legend.position = "none")
```

```{r recreate age bin}
gnomad_decoded <- gnomad1 %>%
  select(-sample_frequency, -sample_count, -nbr_individuals) %>% 
  mutate(gene_in_gnomad = str_match(INFO, "(DNMT3A)(.*?)$")[,2]) %>% 
  filter(IDs == "2-25457242-C-T" | IDs == "2-25234373-C-T" | gene_in_gnomad == "DNMT3A") %>%
  # filter(INFO == "Arg882His" | gene_in_gnomad == "DNMT3A") %>% 
  # mutate(prot_change = str_match(INFO, "(|R)(.*?)|$")[,2]) %>% 
  # mutate(prot_change1 = str_match(INFO, "(|R)(.*?)|$")[,3]) %>% 
  # slice(1:50) %>% 
  # mutate(IDs = ifelse(IDs == "2-25234373-C-T", "2-25234373-C-T_R882H", IDs)) %>% 

  # "Count of age values falling below lowest histogram bin edge for heterozygous individuals">
  mutate("<30" = as.numeric(str_match(INFO, "age_hist_het_n_smaller=(.*?);")[,2])) %>%
  mutate(age_hist_het_bin_freq = str_match(INFO, "age_hist_het_bin_freq=(.*?);")[,2]) %>%
  separate(col = age_hist_het_bin_freq,
           into = c("30-35","35-40","40-45","45-50",
                    "50-55","55-60","60-65","65-70",
                    "70-75","75-80"),
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>%
    # "Count of age values falling above highest histogram bin edge for heterozygous individuals">
  mutate(">80" = as.numeric(str_match(INFO, "age_hist_het_n_larger=(.*?);")[,2])) %>%
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>% 
  
  add_row("IDs"="All individuals", "ID"="All individuals of any genotype bin",
          "<30" = 2547, "30-35" = 3423,
          "35-40" = 4546,"40-45" = 8487,
          "45-50" = 10355,"50-55" = 12693,
          "55-60" = 11933,"60-65" = 10534,
          "65-70" = 8882,"70-75" = 5991,
          "75-80" = 4136,
          ">80" = 1935) %>%
  mutate(nbr_individuals = rowSums(select(.,`<30`:`>80`), na.rm = TRUE)) %>%

  # 2. Extract count per age bin for heterozygous ind--
  # "Count of age values falling below lowest histogram bin edge for homozygous individuals">
  mutate("<30_hom" = as.numeric(str_match(INFO, "age_hist_hom_n_smaller=(.*?);")[,2])) %>%
  mutate(age_hist_hom_bin_freq = str_match(INFO, "age_hist_hom_bin_freq=(.*?);")[,2]) %>%
  separate(col = age_hist_hom_bin_freq,
           into = c("30-35_hom","35-40_hom","40-45_hom","45-50_hom",
                    "50-55_hom","55-60_hom","60-65_hom","65-70_hom",
                    "70-75_hom","75-80_hom"), 
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>% 
  # "Count of age values falling above highest histogram bin edge for homozygous individuals">
  mutate(">80_hom" = as.numeric(str_match(INFO, "age_hist_hom_n_larger=(.*?);")[,2])) %>% 
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>% 
  mutate(nbr_hom_individuals = rowSums(select(.,`<30_hom`:`>80_hom`), na.rm = TRUE)) %>%

  # 3. Combine count for Het and hom individuals
  # mutate("<30" = rowSums(select(.,`30-35_het`,`30-35_hom`), na.rm = TRUE),
  #        "30-35" = rowSums(select(.,`30-35_het`,`30-35_hom`), na.rm = TRUE),
  #        "35-40" = rowSums(select(.,`35-40_het`,`35-40_hom`), na.rm = TRUE),
  #        "40-45" = rowSums(select(.,`40-45_het`,`40-45_hom`), na.rm = TRUE),
  #        "45-50" = rowSums(select(.,`45-50_het`,`45-50_hom`), na.rm = TRUE),
  #        "50-55" = rowSums(select(.,`50-55_het`,`50-55_hom`), na.rm = TRUE),
  #        "55-60" = rowSums(select(.,`55-60_het`,`55-60_hom`), na.rm = TRUE),
  #        "60-65" = rowSums(select(.,`60-65_het`,`60-65_hom`), na.rm = TRUE),
  #        "65-70" = rowSums(select(.,`65-70_het`,`65-70_hom`), na.rm = TRUE),
  #        "70-75" = rowSums(select(.,`70-75_het`,`70-75_hom`), na.rm = TRUE),
  #        "75-80" = rowSums(select(.,`75-80_het`,`75-80_hom`), na.rm = TRUE),
  #        ">80" = rowSums(select(.,`>80_het`,`>80_hom`), na.rm = TRUE)
  #        ) %>%

  select(-c(ends_with("_hom"))) %>% 
  
  select(IDs, everything())
```

```{r recreate age bin2}
gnomad_decoded <- gnomad_decoded %>% 
  pivot_longer(cols = c(grep("^([[:digit:]]|<|>)", colnames(.))), 
               names_to = "age_bin", values_to = "sample_count") %>% 
  mutate(age_bin = factor(age_bin, 
                          levels = c("<30", "30-35","35-40","40-45","45-50","50-55",
                                     "55-60","60-65","65-70","70-75","75-80", ">80"
                                     ))) %>% 
  mutate(bin_age = case_when(
    # The average age at diagnosis is 8 overall (ages 0 to 19), 
    # 5 years old for children (aged 0 to 14), and 17 years old 
    # for adolescents (aged 15 to 19), while adults' average 
    # age for cancer diagnosis is 65
    str_detect(age_bin, "<30")              ~ 15, 
    str_detect(age_bin, "30-35")            ~ 32.5,
    str_detect(age_bin, "35-40")            ~ 37.5,
    str_detect(age_bin, "40-45")            ~ 42.5,
    str_detect(age_bin, "45-50")            ~ 47.5,
    str_detect(age_bin, "50-55")            ~ 52.5,
    str_detect(age_bin, "55-60")            ~ 57.5,
    str_detect(age_bin, "60-65")            ~ 62.5,
    str_detect(age_bin, "65-70")            ~ 67.5,
    str_detect(age_bin, "70-75")            ~ 72.5,
    str_detect(age_bin, "75-80")            ~ 77.5,
    TRUE ~ 90
  )) %>% 
  mutate(sample_frequency = sample_count / nbr_individuals)
```

<!-- <!-- Found in Cosmic data --> 

```{r load cosmic}
# cosmic <-
#   read.delim(paste0(path, "/cosmic_raw_data/cosmic_DNMT3_data.vcf.gz"),
#              header = FALSE,
#              col.names = c("Gene.name", "Accession.Number", "Gene.CDS.length", "HGNC.ID", "Sample.name",
#                            "ID_sample", "ID_tumour", "Primary.site", "Site.subtype.1", "Site.subtype.2",
#                            "Site.subtype.3", "Primary.histology", "Histology.subtype.1", "Histology.subtype.2",
#                            "Histology.subtype.3", "Genome.wide.screen", "GENOMIC_MUTATION_ID",
#                            "LEGACY_MUTATION_ID", "MUTATION_ID", "Mutation.CDS", "Mutation.AA",
#                            "Mutation.Description", "Mutation.zygosity", "LOH", "GRCh",
#                            "Mutation.genome.position", "Mutation.strand", "Resistance.Mutation",
#                            "FATHMM.prediction", "FATHMM.score", "Mutation.somatic.status", "Pubmed_PMID",
#                            "ID_STUDY", "Sample.Type", "Tumour.origin", "Age", "HGVSP", "HGVSC", "HGVSG"))

# cosmic <-
#   read.delim(paste0(path, "/cosmic_raw_data/splitted_data/filter_cosmic_withIDs/chr2_cosmic.vcf.gz"),
#               sep = " ")
```

<!-- ```{r} -->
<!-- cosmic <- -->
<!--   cosmic %>% -->
<!--   mutate(modification = str_match(HGVSG, "(delins|ins|del|dup)(.*?)$")[,2], -->
<!--          aa_inserted = case_when( -->
<!--            modification %in% c("ins", "delins")   ~ str_match(HGVSG, "(delins|ins|del|dup)(.*?)$")[,3], -->
<!--            TRUE                                   ~ NA_character_ -->
<!--          )) %>% -->

<!--   separate(HGVSG, into = c("chr", "chr_arm", "POS", "ALT"), sep = ":|\\.|>|delins|dup|del", remove = FALSE) %>% -->
<!--   mutate(REF = str_match(POS, "([0-9]*)([A-Z]*)")[,3]) %>% -->
<!--   mutate(POS = str_match(POS, "([0-9]*)([A-Z]*)")[,2], -->
<!--          POS = as.numeric(POS)) %>% -->
<!--   # separate(Mutation.genome.position, into = c("chr", "start", "end"), sep = ":|-", remove = FALSE) %>% -->
<!--   unite(IDs, c(chr, POS, REF, ALT), sep = "-", remove = FALSE) %>% #select(IDs, everything()) -->
<!--   # separate(HGVSG, into = c("ENSP", "p_prot_modif"), sep = ":", remove = FALSE) %>% -->
<!--   mutate(IDs = case_when( -->
<!--     IDs == "-NA-NA-NA"                 ~ NA_character_, -->
<!--     TRUE                               ~ IDs -->
<!--   )) %>% -->
<!--   separate(`Gene.name`, into = c("gene_name_cosmic"), sep = "_", remove = FALSE) %>% -->
<!--   # 1 tumor can have multiple sample -->
<!--   distinct(IDs, ID_tumour, .keep_all = TRUE) %>% -->
<!--   arrange(IDs) %>% -->
<!--   select(IDs, "Gene.name", gene_name_cosmic, Gene.name, "Accession.Number", "chr", "POS", REF, "ALT", HGVSG, everything()) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- cosmic <- cosmic %>% -->
<!--   group_by(IDs, gene_name_cosmic, chr, POS, REF, ALT) %>% -->
<!--   mutate(occurrence_in_cosmic = n()) -->

<!-- max_occurence = max(cosmic$occurrence_in_cosmic) -->
<!-- cosmic <- cosmic %>% ###################################### Clean up here-------------------------------- -->

<!--   # distinct() -->
<!--   group_by(IDs, gene_name_cosmic, chr, POS, REF, ALT, -->
<!--            HGNC.ID, -->
<!--            HGVSP, ###################################### Clean up here-------------------------------- -->
<!--            FATHMM.prediction, FATHMM.score, -->

<!--            occurrence_in_cosmic) %>% -->
<!--   summarise_at(vars(Accession.Number, HGVSG, -->
<!--                     Gene.CDS.length, -->
<!--     Mutation.CDS, -->
<!--           Mutation.AA, -->
<!--           Primary.site, -->
<!--           Mutation.Description, -->
<!--     Mutation.genome.position, -->
<!--           Primary.histology, -->
<!--           Histology.subtype.1, Histology.subtype.2, Histology.subtype.3, -->
<!--           HGVSC, -->
<!--           Mutation.somatic.status, Pubmed_PMID, -->
<!--           Sample.Type, Tumour.origin, Age), str_c, collapse = "; ") %>% -->
<!--   ungroup() %>% -->
<!--   separate(Age, paste("Age", 1:max(max_occurence), sep = "_"), -->
<!--            sep = "; ", remove = TRUE, -->
<!--            extra = "warn", fill = "right") %>% -->
<!--   purrr::keep(~!all(is.na(.))) %>% -->
<!--   mutate(across(starts_with("Age_"), ~ as.numeric(.))) %>% -->

<!--   mutate(average_age = rowMeans(select(., starts_with("Age")), na.rm = TRUE)) %>% -->
<!--   mutate(variant_in_cosmic = "Yes") -->
<!-- ``` -->

```{r bind cosmic Ids}
# gnomad_decoded <- left_join(gnomad_decoded %>%
#                               mutate(X.CHROM = as.character(X.CHROM)),
#                             cosmic %>%
#                               mutate(chr = as.character(chr)),
#                             by = c("IDs", "X.CHROM" = "chr", "POS", "REF", "ALT"))
```

# Plots all variants
**R882H is curve #14 aka 2-25234373-C-T.**
<div class = "row">
<div class = "col-md-6">
## Age 
```{r Plots all variants}
# mann_w <-  data.frame(matrix(nrow=1, ncol=0)) 
# effect_size <-  data.frame(matrix(nrow=1, ncol=0)) 
# effect_size_cat <-  data.frame(matrix(nrow=1, ncol=0)) 

n <- 0

set.seed(1234)

for(i in unique(gnomad_decoded$IDs)) {
  
  All <- gnomad_decoded %>% 
    filter(IDs == "All individuals"
  ) %>% 
    mutate(IDs = "Reference")
  data_plot <- gnomad_decoded %>% 
    filter(IDs == i
    ) %>% 
    bind_rows(., All) %>% 
    distinct(IDs, bin_age, .keep_all = TRUE)
  n <- n + 1
  p <- data_plot %>% 
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    # ggplot(aes(x = age_bin, y= sample_frequency, color= IDs))+
    # geom_bar(stat = "identity", aes(alpha= IDs))
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                # method.args=list(family=quasibinomial)#,
                span = 0.6
    )+
    labs(title = n)+
    geom_col(data = data_plot %>% 
               filter(IDs == i) ,
             aes(x= bin_age, y= sample_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$sample_frequency))
  
  # p_ <- layer_data(p, 1) %>% 
  # mutate(colour =  as.factor(colour)) %>% 
  # mutate_if(is.numeric, ~replace(., is.na(.), 0))
  # 
  # variant <- p_ %>% filter(group == 1)
  # variant <- sample(variant$x, 180, prob = variant$y, replace = TRUE)
  # ref <- p_ %>% filter(group == 2)
  # ref <- sample(ref$x, 180, prob = ref$y, replace = TRUE)
  
  # result1 <- wilcox.test(variant, ref, alternative = "greater")$p.value
  # mann_w <- rbind(mann_w, result1)
  
  # result2 <- cohen.d(variant, ref, hedges.correction=TRUE)$estimate
  # effect_size <- rbind(effect_size, result2)
  # result3 <- cohen.d(variant, ref,hedges.correction=TRUE)$magnitude[1] %>% 
  #   as_tibble() %>%
  #   mutate(value = case_when(
  #     value == "negligible"   ~ "negligible",
  #     value == "small"        ~ "small",
  #     value == "medium"       ~ "medium",
  #     value == "large"        ~ "large"
  #   ))
  # effect_size_cat <- rbind(effect_size_cat, result3)
  
  p <- p+
    # geom_vline(xintercept = data_plot$distribution_mean, color= "red")+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("MW pval= ", round(age_mann_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -9
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("BH= ", round(adjusted_age_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -11
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Effect size Hedges g= ", round(age_effect_size, 3))),
      hjust   = 0,
      vjust   = -17
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Hedges cat= ", age_effect_size_cat)),
      hjust   = 0,
      vjust   = -15
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals with age= ", nbr_individuals)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$sample_frequency)
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste("Is in cosmic?", variant_in_cosmic,",",Mutation.somatic.status)),
      colour = "purple",
      hjust   = 0,
      vjust   = -35
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("HGVSP= ", HGVSP)),
      colour = "purple",
      hjust   = 0,
      vjust   = -33
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("occurrence= ", occurrence_in_cosmic)),
      colour = "purple",
      hjust   = 0,
      vjust   = -31
    )
  
  print(p)

}
```
</div>

<div class = "col-md-6">
## Allele balance
```{r}
gnomad <- gnomad_decoded %>%
  filter(IDs != "All individuals") %>% 
  select(-AB_count) %>% 
  # 1. Extract count per AB bin for heterozygous ind--
  # ab_hist_alt_bin_freq,Number=A,Type=String,Description="Histogram for AB in heterozygous individuals; 
  # bin edges are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00>
  mutate(ab_hist_alt_bin_freq = str_match(INFO, "ab_hist_alt_bin_freq=(.*?);")[,2]) %>%
  separate(col = ab_hist_alt_bin_freq,
           into = c("0.00-0.05","0.05-0.10","0.10-0.15","0.15-0.20",
                    "0.20-0.25","0.25-0.30","0.30-0.35","0.35-0.40",
                    "0.40-0.45","0.45-0.50","0.50-0.55","0.55-0.60",
                    "0.60-0.65","0.65-0.70","0.70-0.75","0.75-0.80",
                    "0.80-0.85","0.85-0.90","0.90-0.95","0.95-1.00"),
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>%
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>%
  add_row("IDs"="Reference", "ID"="ASXL1 20-32435697-C-T",
          "0.00-0.05" = 0, "0.05-0.10"  = 0, "0.10-0.15" = 0, 
          "0.15-0.20" = 1, "0.20-0.25" = 2, "0.25-0.30" = 6, 
          "0.30-0.35" = 87, "0.35-0.40" = 274, "0.40-0.45" = 1220,
          "0.45-0.50" = 2286, "0.50-0.55" = 1737, "0.55-0.60" = 506,
          "0.60-0.65" = 112, "0.65-0.70" = 14, "0.70-0.75" = 4,
          "0.75-0.80" = 0, "0.80-0.85" = 1, "0.85-0.90" = 0,
          "0.90-0.95" = 0, "0.95-1.00" = 0) %>%
  mutate(total_allele_balance = rowSums(select(.,`0.00-0.05`:`0.95-1.00`), na.rm = TRUE)) %>%
  
  select(IDs, everything())

gnomad <- gnomad %>%
  pivot_longer(cols = c(grep("^[[:digit:]]", colnames(.))),
               names_to = "allele_bin", values_to = "AB_count") %>%
  mutate(allele_bin = factor(allele_bin,
                             levels = c("0.00-0.05","0.05-0.10","0.10-0.15","0.15-0.20",
                                        "0.20-0.25","0.25-0.30","0.30-0.35","0.35-0.40",
                                        "0.40-0.45","0.45-0.50","0.50-0.55","0.55-0.60",
                                        "0.60-0.65","0.65-0.70","0.70-0.75","0.75-0.80",
                                        "0.80-0.85","0.85-0.90","0.90-0.95","0.95-1.00"
                             ))) %>%
  mutate(AB_bin = case_when(
    allele_bin == "0.00-0.05"            ~ 0.025,
    allele_bin == "0.05-0.10"            ~ 0.075,
    allele_bin == "0.10-0.15"            ~ 0.125,
    allele_bin == "0.15-0.20"            ~ 0.175,
    allele_bin == "0.20-0.25"            ~ 0.225,
    allele_bin == "0.25-0.30"            ~ 0.275,
    allele_bin == "0.30-0.35"            ~ 0.325,
    allele_bin == "0.35-0.40"            ~ 0.375,
    allele_bin == "0.40-0.45"            ~ 0.425,
    allele_bin == "0.45-0.50"            ~ 0.475,
    allele_bin == "0.50-0.55"            ~ 0.525,
    allele_bin == "0.55-0.60"            ~ 0.575,
    allele_bin == "0.60-0.65"            ~ 0.625,
    allele_bin == "0.65-0.70"            ~ 0.675,
    allele_bin == "0.70-0.75"            ~ 0.725,
    allele_bin == "0.75-0.80"            ~ 0.775,
    allele_bin == "0.80-0.85"            ~ 0.825,
    allele_bin == "0.85-0.90"            ~ 0.875,
    allele_bin == "0.90-0.95"            ~ 0.925,
    allele_bin == "0.95-1.00"            ~ 0.975
  )) %>%
  mutate(AB_frequency = AB_count / total_allele_balance)


## Step 1 : Limit variants with a skewed age distribution
# allele_balance_pval <-  data.frame(matrix(nrow=length(unique(gnomad$IDs)), ncol=1,
#                                 dimnames = list(c(unique(gnomad$IDs)),
#                                                 c("value")) ))
n <- 0
set.seed(1234)
for(i in unique(gnomad$IDs)) {
  
  All <- gnomad %>%
    filter(IDs == "Reference"
    ) %>% 
    mutate(IDs = "Ref")
  data_plot <- gnomad %>%
    filter(IDs == i
    ) %>%
    bind_rows(., All) %>%
    distinct(IDs, AB_bin, .keep_all = TRUE)
  
  n <- n + 1
  p <- data_plot %>%
    ggplot(aes(x = AB_bin, y= AB_frequency, color= IDs))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
    )+
    labs(title = n)+
    geom_col(data = data_plot %>%
               filter(IDs == i) ,
             aes(x= AB_bin, y= AB_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$AB_frequency))+
    theme(legend.position = "none")
  
  p <- p+
    geom_text(
      data    = data_plot %>% 
        filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("p-AB= ", round(allele_balance_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -30
    )+
    geom_text(
      data    = data_plot %>% 
        filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("p-AB adjusted= ", round(adjusted_AB_pval, 3))),
      colour = "red",
      hjust   = 0,
      vjust   = -28
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Effect size Hedges g= ", round(AB_effect_size, 3))),
      hjust   = 0,
      vjust   = -17
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Hedges cat= ", AB_effect_size_cat)),
      hjust   = 0,
      vjust   = -15
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals with age= ", total_allele_balance)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$AB_frequency)
    )
  
  print(p)
  
}
```
</div>
</div>

```{r}
# mann_w <- mann_w %>% `colnames<-`(c("mann_w")) %>% 
#   mutate(IDs = c(unique(gnomad_decoded$IDs)))
# effect_size <- effect_size %>% `colnames<-`(c("effect_size"))%>% 
#   mutate(IDs = c(unique(gnomad_decoded$IDs)))
# effect_size_cat <- effect_size_cat %>% `colnames<-`(c("effect_size_cat"))
# 
# stat_measure <- bind_cols(effect_size, effect_size_cat) %>% 
#   left_join(gnomad_decoded, .,
#   by= "IDs")

# print(head(stat_measure))
```

<!-- ```{r} -->
<!-- gnomad_decoded %>%  -->
<!--   distinct(IDs, .keep_all = TRUE) %>%  -->
<!--   ggplot(aes(x= AB_effect_size, y= adjusted_AB_pval))+ -->
<!--   geom_point(color= "grey")+ -->
<!--   geom_point(data= gnomad_decoded %>%  -->
<!--                distinct(IDs, .keep_all = TRUE) %>%  -->
<!--                filter(IDs=="2-25234373-C-T"), -->
<!--              aes(x= AB_effect_size, y= adjusted_AB_pval), color= "red")+ -->
<!--   geom_point(data= gnomad_decoded %>%  -->
<!--                distinct(IDs, .keep_all = TRUE) %>%  -->
<!--                filter(str_detect(IDs, bhadjusted_selected)), -->
<!--              aes(x= effect_size, y=adjusted_pval), color= "yellow")+ -->
<!--   geom_point(data= gnomad_decoded %>%  -->
<!--                distinct(IDs, .keep_all = TRUE) %>%  -->
<!--                filter(str_detect(IDs, effsize_selected)), -->
<!--              aes(x= effect_size, y=adjusted_pval), color= "orange") -->

<!-- gnomad_decoded %>%  -->
<!--   distinct(IDs, .keep_all = TRUE) %>%  -->
<!--   ggplot(aes(x= AB_effect_size, y= adjusted_AB_pval))+ -->
<!--   geom_point(color= "grey")+ -->
<!--   geom_point(data= gnomad_decoded %>%  -->
<!--                distinct(IDs, .keep_all = TRUE) %>%  -->
<!--                filter(IDs=="2-25234373-C-T"), -->
<!--              aes(x= effect_size, y=adjusted_pval), color= "red")+ -->
<!--   geom_point(data= gnomad_decoded %>%  -->
<!--                distinct(IDs, .keep_all = TRUE) %>%  -->
<!--                filter(str_detect(IDs, bhadjusted_selected)), -->
<!--              aes(x= effect_size, y=adjusted_pval), color= "yellow")+ -->
<!--   geom_point(data= gnomad_decoded %>%  -->
<!--                distinct(IDs, .keep_all = TRUE) %>%  -->
<!--                filter(str_detect(IDs, effsize_selected)), -->
<!--              aes(x= effect_size, y=adjusted_pval), color= "orange")+ -->
<!--   scale_y_log10()+ -->
<!--   labs(x= "Hedges g effect size", y= "BH pval (log10)")+ -->
<!--   theme(legend.position = "none") -->

<!-- gnomad_decoded %>%  -->
<!--   distinct(IDs, .keep_all = TRUE) %>%  -->
<!--   ggplot(aes(x= effect_size, y= adjusted_pval))+ -->
<!--   geom_point(color= "grey")+ -->
<!--   geom_point(data= gnomad_decoded %>%  -->
<!--                distinct(IDs, .keep_all = TRUE) %>% -->
<!--                filter(IDs=="2-25234373-C-T"), -->
<!--              aes(x= effect_size, y=adjusted_pval), color= "red")+ -->
<!--   geom_point(data= gnomad_decoded %>%  -->
<!--                distinct(IDs, .keep_all = TRUE) %>%  -->
<!--                filter(str_detect(IDs, bhadjusted_selected)), -->
<!--              aes(x= effect_size, y=adjusted_pval), color= "yellow")+ -->
<!--   geom_point(data= gnomad_decoded %>%  -->
<!--                distinct(IDs, .keep_all = TRUE) %>%  -->
<!--                filter(str_detect(IDs, effsize_selected)), -->
<!--              aes(x= effect_size, y=adjusted_pval), color= "orange")+ -->
<!--   scale_y_continuous(trans = 'log2')+ -->
<!--   # scale_y_continuous(trans = "log2")+ -->
<!--   labs(x= "Hedges g effect size", y= "BH pval (log2)")+ -->
<!--   theme(legend.position = "none") -->
<!-- ``` -->

# Plots for BH ≤ 0.05
**R882H is curve #8 aka 2-25234373-C-T.**
<div class = "row">
<div class = "col-md-6">
## Age 
```{r}
selected_variants <- gnomad_decoded %>% filter(adjusted_AB_pval <= 0.05)
n <- 0

set.seed(1234)

for(i in unique(selected_variants$IDs)) {
  
  All <- gnomad_decoded %>% 
    filter(IDs == "All individuals"
  ) %>% 
    mutate(IDs = "Reference")
  data_plot <- selected_variants %>% 
    filter(IDs == i
    ) %>% 
    bind_rows(., All) %>% 
    distinct(IDs, bin_age, .keep_all = TRUE)
  n <- n + 1
  p <- data_plot %>% 
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    # ggplot(aes(x = age_bin, y= sample_frequency, color= IDs))+
    # geom_bar(stat = "identity", aes(alpha= IDs))
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                # method.args=list(family=quasibinomial)#,
                span = 0.6
    )+
    labs(title = n)+
    geom_col(data = data_plot %>% 
               filter(IDs == i) ,
             aes(x= bin_age, y= sample_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$sample_frequency))
  
  p <- p+
    # geom_vline(xintercept = data_plot$distribution_mean, color= "red")+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("MW pval= ", round(age_mann_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -9
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("BH= ", round(adjusted_age_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -11
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Effect size Hedges g= ", round(age_effect_size, 3))),
      hjust   = 0,
      vjust   = -17
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Hedges cat= ", age_effect_size_cat)),
      hjust   = 0,
      vjust   = -15
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste("Is in cosmic?", variant_in_cosmic,",",Mutation.somatic.status)),
      colour = "purple",
      hjust   = 0,
      vjust   = -35
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("HGVSP= ", HGVSP)),
      colour = "purple",
      hjust   = 0,
      vjust   = -33
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("occurrence= ", occurrence_in_cosmic)),
      colour = "purple",
      hjust   = 0,
      vjust   = -31
    )
  
  print(p)

}
```
</div>

<div class = "col-md-6">
## Allele balance
```{r}
gnomad <- gnomad %>% filter(adjusted_AB_pval <= 0.05)

n <- 0
set.seed(1234)
for(i in unique(gnomad$IDs)) {
  
  All <- gnomad %>%
    filter(IDs == "Reference"
    ) %>% 
    mutate(IDs = "Ref")
  data_plot <- gnomad %>%
    filter(IDs == i
    ) %>%
    bind_rows(., All) %>%
    distinct(IDs, AB_bin, .keep_all = TRUE)
  
  n <- n + 1
  p <- data_plot %>%
    ggplot(aes(x = AB_bin, y= AB_frequency, color= IDs))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
    )+
    labs(title = n)+
    geom_col(data = data_plot %>%
               filter(IDs == i) ,
             aes(x= AB_bin, y= AB_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$AB_frequency))+
    theme(legend.position = "none")
  
  p <- p+
    geom_text(
      data    = data_plot %>% 
        filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("p-AB= ", round(allele_balance_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -30
    )+
    geom_text(
      data    = data_plot %>% 
        filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("p-AB adjusted= ", round(adjusted_AB_pval, 3))),
      colour = "red",
      hjust   = 0,
      vjust   = -28
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Effect size Hedges g= ", round(AB_effect_size, 3))),
      hjust   = 0,
      vjust   = -17
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Hedges cat= ", AB_effect_size_cat)),
      hjust   = 0,
      vjust   = -15
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals with age= ", total_allele_balance)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$AB_frequency)
    )
  
  print(p)
  
}
```
</div>
</div>

<br>

# Plots Hedges >= small
**R882H is curve #8 aka 2-25234373-C-T.**
<div class = "row">
<div class = "col-md-6">
## Age 
```{r}
selected_variants <- selected_variants %>% filter(AB_effect_size_cat != "negligible")
n <- 0

set.seed(1234)

for(i in unique(selected_variants$IDs)) {
  
  All <- gnomad_decoded %>% 
    filter(IDs == "All individuals"
  ) %>% 
    mutate(IDs = "Reference")
  data_plot <- selected_variants %>% 
    filter(IDs == i
    ) %>% 
    bind_rows(., All) %>% 
    distinct(IDs, bin_age, .keep_all = TRUE)
  n <- n + 1
  p <- data_plot %>% 
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    # ggplot(aes(x = age_bin, y= sample_frequency, color= IDs))+
    # geom_bar(stat = "identity", aes(alpha= IDs))
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                # method.args=list(family=quasibinomial)#,
                span = 0.6
    )+
    labs(title = n)+
    geom_col(data = data_plot %>% 
               filter(IDs == i) ,
             aes(x= bin_age, y= sample_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$sample_frequency))
  
  p <- p+
    # geom_vline(xintercept = data_plot$distribution_mean, color= "red")+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("MW pval= ", round(age_mann_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -9
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("BH= ", round(adjusted_age_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -11
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Effect size Hedges g= ", round(age_effect_size, 3))),
      hjust   = 0,
      vjust   = -17
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Hedges cat= ", age_effect_size_cat)),
      hjust   = 0,
      vjust   = -15
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste("Is in cosmic?", variant_in_cosmic,",",Mutation.somatic.status)),
      colour = "purple",
      hjust   = 0,
      vjust   = -35
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("HGVSP= ", HGVSP)),
      colour = "purple",
      hjust   = 0,
      vjust   = -33
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("occurrence= ", occurrence_in_cosmic)),
      colour = "purple",
      hjust   = 0,
      vjust   = -31
    )
  
  print(p)

}
```
</div>

<div class = "col-md-6">
## Allele balance
```{r}
gnomad <- gnomad %>% filter(AB_effect_size_cat != "negligible")

n <- 0
set.seed(1234)
for(i in unique(gnomad$IDs)) {
  
  All <- gnomad %>%
    filter(IDs == "Reference"
    ) %>% 
    mutate(IDs = "Ref")
  data_plot <- gnomad %>%
    filter(IDs == i
    ) %>%
    bind_rows(., All) %>%
    distinct(IDs, AB_bin, .keep_all = TRUE)
  
  n <- n + 1
  p <- data_plot %>%
    ggplot(aes(x = AB_bin, y= AB_frequency, color= IDs))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
    )+
    labs(title = n)+
    geom_col(data = data_plot %>%
               filter(IDs == i) ,
             aes(x= AB_bin, y= AB_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$AB_frequency))+
    theme(legend.position = "none")
  
  p <- p+
    geom_text(
      data    = data_plot %>% 
        filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("p-AB= ", round(allele_balance_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -30
    )+
    geom_text(
      data    = data_plot %>% 
        filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("p-AB adjusted= ", round(adjusted_AB_pval, 3))),
      colour = "red",
      hjust   = 0,
      vjust   = -28
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Effect size Hedges g= ", round(AB_effect_size, 3))),
      hjust   = 0,
      vjust   = -17
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Hedges cat= ", AB_effect_size_cat)),
      hjust   = 0,
      vjust   = -15
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals with age= ", total_allele_balance)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$AB_frequency)
    )
  
  print(p)
  
}
```
</div>
</div>

<br>











