---
title: "Inverstigate chr2 pval"
author: "Christelle Colin-Leitzinger"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: no
    theme: cerulean
    highlight: pygments
    df_print: paged
editor_options:
  chunk_output_type: console
---

<style type="text/css">
h1.title {
  font-size: 25px;
}
.figure {
   margin-top: 25px;
   margin-bottom: 25px;
}

table {
    margin-top: 25px;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.align='center'
                      )
options(gtsummary.print_engine = "gt")
options(gtsummary.as_gt.addl_cmds = "gt::tab_options(table.font.size = 12, data_row.padding = gt::px(1))")
```


<style>
div.blue { background-color:#0099CC; border-radius: 5px; padding: 20px; font-size: 38px}
</style>
<div class = "blue">

<span style="color: white;">GnomAD data</span>

</div>
<br>

```{r library, include=FALSE}
library(tidyverse)
library(effsize)

theme_set(theme_classic())
```

```{r load}
path <- fs::path("", "Volumes", "Lab_Gillis", "Christelle")

gnomad <-
  read.delim(paste0(here::here(), "/chr2_CH_variants.vcf.gz"), 
             colClasses = c("ALT"="character"),
              sep = " ")
cosmic <-
  read.delim(paste0(path, "/cosmic_raw_data/cosmic_IDH1.vcf.gz"),
             header = FALSE,
             col.names = c("#CHROM", "Gene.name", "Accession.Number", "Gene.CDS.length", "HGNC.ID", "Sample.name", 
                                     "ID_sample", "ID_tumour", "Primary.site", "Site.subtype.1", "Site.subtype.2", 
                                     "Site.subtype.3", "Primary.histology", "Histology.subtype.1", "Histology.subtype.2", 
                                     "Histology.subtype.3", "Genome.wide.screen", "GENOMIC_MUTATION_ID", 
                                     "LEGACY_MUTATION_ID", "MUTATION_ID", "Mutation.CDS", "Mutation.AA", 
                                     "Mutation.Description", "Mutation.zygosity", "LOH", "GRCh", 
                                     "Mutation.genome.position", "Mutation.strand", "Resistance.Mutation", 
                                     "FATHMM.prediction", "FATHMM.score", "Mutation.somatic.status", "Pubmed_PMID", 
                                     "ID_STUDY", "Sample.Type", "Tumour.origin", "Age", "HGVSP", "HGVSC", "HGVSG"))
```

```{r cosmic}
cosmic <-
  cosmic %>%
  mutate(modification = str_match(HGVSG, "(delins|ins|del|dup)(.*?)$")[,2],
         aa_inserted = case_when(
           modification %in% c("ins", "delins")   ~ str_match(HGVSG, "(delins|ins|del|dup)(.*?)$")[,3],
           TRUE                                   ~ NA_character_
         )) %>%

  separate(HGVSG, into = c("chr", "chr_arm", "POS", "ALT"), sep = ":|\\.|>|delins|dup|del", remove = FALSE) %>%
  mutate(REF = str_match(POS, "([0-9]*)([A-Z]*)")[,3]) %>%
  mutate(POS = str_match(POS, "([0-9]*)([A-Z]*)")[,2],
         POS = as.numeric(POS)) %>%
  # separate(Mutation.genome.position, into = c("chr", "start", "end"), sep = ":|-", remove = FALSE) %>%
  unite(IDs, c(chr, POS, REF, ALT), sep = "-", remove = FALSE) %>% #select(IDs, everything())
  # separate(HGVSG, into = c("ENSP", "p_prot_modif"), sep = ":", remove = FALSE) %>%
  mutate(IDs = case_when(
    IDs == "-NA-NA-NA"                 ~ NA_character_,
    TRUE                               ~ IDs
  )) %>%
  separate(`Gene.name`, into = c("gene_name_cosmic"), sep = "_", remove = FALSE) %>%
  # 1 tumor can have multiple sample
  distinct(IDs, ID_tumour, .keep_all = TRUE) %>%
  arrange(IDs) %>%
  select(IDs, "Gene.name", gene_name_cosmic, Gene.name, "Accession.Number", "chr", "POS", REF, "ALT", HGVSG, everything())

cosmic <- cosmic %>%
    group_by(IDs, gene_name_cosmic, chr, POS, REF, ALT) %>%
    mutate(occurrence_in_cosmic = n()) %>%
    filter(occurrence_in_cosmic > 1) %>% 
    mutate(variant_in_cosmic = "Yes")
```

```{r}
gnomad_decoded <- gnomad %>%
  select(-sample_frequency, -sample_count, -nbr_individuals) %>% 
  mutate(gene_in_gnomad = str_match(INFO, "(IDH1)(.*?)$")[,2]) %>% 
  filter(gene_in_gnomad == "IDH1")
```

```{r bind cosmic Ids}
IDH1_gnomad_cosmic <- inner_join(gnomad_decoded %>%
                                  distinct(IDs, .keep_all = TRUE) %>% 
                              mutate(X.CHROM = as.character(X.CHROM)),
                            cosmic %>%
                              mutate(chr = as.character(chr)),
                            by = c("IDs", "X.CHROM" = "chr", "POS", "REF", "ALT"
                                   ))
```

From 130 IDH1 variants in GnomAD, 10 are present in Cosmic.
```{r recreate age bin}
gnomad_decoded <- IDH1_gnomad_cosmic %>%
  # filter(INFO == "Arg882His" | gene_in_gnomad == "DNMT3A") %>% 
  # mutate(prot_change = str_match(INFO, "(|R)(.*?)|$")[,2]) %>% 
  # mutate(prot_change1 = str_match(INFO, "(|R)(.*?)|$")[,3]) %>% 
  # slice(1:50) %>% 
  # mutate(IDs = ifelse(IDs == "2-25234373-C-T", "2-25234373-C-T_R882H", IDs)) %>% 

  # "Count of age values falling below lowest histogram bin edge for heterozygous individuals">
  mutate("<30" = as.numeric(str_match(INFO, "age_hist_het_n_smaller=(.*?);")[,2])) %>%
  mutate(age_hist_het_bin_freq = str_match(INFO, "age_hist_het_bin_freq=(.*?);")[,2]) %>%
  separate(col = age_hist_het_bin_freq,
           into = c("30-35","35-40","40-45","45-50",
                    "50-55","55-60","60-65","65-70",
                    "70-75","75-80"),
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>%
    # "Count of age values falling above highest histogram bin edge for heterozygous individuals">
  mutate(">80" = as.numeric(str_match(INFO, "age_hist_het_n_larger=(.*?);")[,2])) %>%
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>% 
  
  add_row("IDs"="All individuals", "ID"="All individuals of any genotype bin",
          "<30" = 2547, "30-35" = 3423,
          "35-40" = 4546,"40-45" = 8487,
          "45-50" = 10355,"50-55" = 12693,
          "55-60" = 11933,"60-65" = 10534,
          "65-70" = 8882,"70-75" = 5991,
          "75-80" = 4136,
          ">80" = 1935) %>%
  mutate(nbr_individuals = rowSums(select(.,`<30`:`>80`), na.rm = TRUE)) %>%

  # 2. Extract count per age bin for heterozygous ind--
  # "Count of age values falling below lowest histogram bin edge for homozygous individuals">
  mutate("<30_hom" = as.numeric(str_match(INFO, "age_hist_hom_n_smaller=(.*?);")[,2])) %>%
  mutate(age_hist_hom_bin_freq = str_match(INFO, "age_hist_hom_bin_freq=(.*?);")[,2]) %>%
  separate(col = age_hist_hom_bin_freq,
           into = c("30-35_hom","35-40_hom","40-45_hom","45-50_hom",
                    "50-55_hom","55-60_hom","60-65_hom","65-70_hom",
                    "70-75_hom","75-80_hom"), 
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>% 
  # "Count of age values falling above highest histogram bin edge for homozygous individuals">
  mutate(">80_hom" = as.numeric(str_match(INFO, "age_hist_hom_n_larger=(.*?);")[,2])) %>% 
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>% 
  mutate(nbr_hom_individuals = rowSums(select(.,`<30_hom`:`>80_hom`), na.rm = TRUE)) %>%

  # 3. Combine count for Het and hom individuals
  # mutate("<30" = rowSums(select(.,`30-35_het`,`30-35_hom`), na.rm = TRUE),
  #        "30-35" = rowSums(select(.,`30-35_het`,`30-35_hom`), na.rm = TRUE),
  #        "35-40" = rowSums(select(.,`35-40_het`,`35-40_hom`), na.rm = TRUE),
  #        "40-45" = rowSums(select(.,`40-45_het`,`40-45_hom`), na.rm = TRUE),
  #        "45-50" = rowSums(select(.,`45-50_het`,`45-50_hom`), na.rm = TRUE),
  #        "50-55" = rowSums(select(.,`50-55_het`,`50-55_hom`), na.rm = TRUE),
  #        "55-60" = rowSums(select(.,`55-60_het`,`55-60_hom`), na.rm = TRUE),
  #        "60-65" = rowSums(select(.,`60-65_het`,`60-65_hom`), na.rm = TRUE),
  #        "65-70" = rowSums(select(.,`65-70_het`,`65-70_hom`), na.rm = TRUE),
  #        "70-75" = rowSums(select(.,`70-75_het`,`70-75_hom`), na.rm = TRUE),
  #        "75-80" = rowSums(select(.,`75-80_het`,`75-80_hom`), na.rm = TRUE),
  #        ">80" = rowSums(select(.,`>80_het`,`>80_hom`), na.rm = TRUE)
  #        ) %>%

  select(-c(ends_with("_hom"))) %>% 
  
  select(IDs, everything())
```

```{r recreate age bin2}
gnomad_decoded <- gnomad_decoded %>% 
  pivot_longer(cols = c(grep("^([[:digit:]]|<|>)", colnames(.))), 
               names_to = "age_bin", values_to = "sample_count") %>% 
  mutate(age_bin = factor(age_bin, 
                          levels = c("<30", "30-35","35-40","40-45","45-50","50-55",
                                     "55-60","60-65","65-70","70-75","75-80", ">80"
                                     ))) %>% 
  mutate(bin_age = case_when(
    # The average age at diagnosis is 8 overall (ages 0 to 19), 
    # 5 years old for children (aged 0 to 14), and 17 years old 
    # for adolescents (aged 15 to 19), while adults' average 
    # age for cancer diagnosis is 65
    str_detect(age_bin, "<30")              ~ 15, 
    str_detect(age_bin, "30-35")            ~ 32.5,
    str_detect(age_bin, "35-40")            ~ 37.5,
    str_detect(age_bin, "40-45")            ~ 42.5,
    str_detect(age_bin, "45-50")            ~ 47.5,
    str_detect(age_bin, "50-55")            ~ 52.5,
    str_detect(age_bin, "55-60")            ~ 57.5,
    str_detect(age_bin, "60-65")            ~ 62.5,
    str_detect(age_bin, "65-70")            ~ 67.5,
    str_detect(age_bin, "70-75")            ~ 72.5,
    str_detect(age_bin, "75-80")            ~ 77.5,
    TRUE ~ 90
  )) %>% 
  mutate(sample_frequency = sample_count / nbr_individuals)
```


# Plots all variants
```{r Plots all variants}
n <- 0

set.seed(1234)

for(i in unique(gnomad_decoded$IDs)) {
  
  All <- gnomad_decoded %>% 
    filter(IDs == "All individuals"
  ) %>% 
    mutate(IDs = "Reference")
  data_plot <- gnomad_decoded %>% 
    filter(IDs == i
    ) %>% 
    bind_rows(., All) %>%
    distinct(IDs, bin_age, .keep_all = TRUE)
  
  n <- n + 1
  p <- data_plot %>% 
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    # ggplot(aes(x = age_bin, y= sample_frequency, color= IDs))+
    # geom_bar(stat = "identity", aes(alpha= IDs))
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                # method.args=list(family=quasibinomial)#,
                span = 0.6
    )+
    labs(title = n)+
    geom_col(data = data_plot %>% 
               filter(IDs == i) ,
             aes(x= bin_age, y= sample_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$sample_frequency))
  
  # p_ <- layer_data(p, 1) %>% 
  # mutate(colour =  as.factor(colour)) %>% 
  # mutate_if(is.numeric, ~replace(., is.na(.), 0))
  # 
  # variant <- p_ %>% filter(group == 1)
  # variant <- sample(variant$x, 180, prob = variant$y, replace = TRUE)
  # ref <- p_ %>% filter(group == 2)
  # ref <- sample(ref$x, 180, prob = ref$y, replace = TRUE)
  # 
  # # result1 <- wilcox.test(variant, ref, alternative = "greater")$p.value
  # # mann_w <- rbind(mann_w, result1)
  # 
  # result2 <- cohen.d(variant, ref, hedges.correction=TRUE)$estimate
  # effect_size <- rbind(effect_size, result2)
  # result3 <- cohen.d(variant, ref,hedges.correction=TRUE)$magnitude[1] %>% 
  #   as_tibble() %>%
  #   mutate(value = case_when(
  #     value == "negligible"   ~ "negligible",
  #     value == "small"        ~ "small",
  #     value == "medium"       ~ "medium",
  #     value == "large"        ~ "large"
  #   ))
  # effect_size_cat <- rbind(effect_size_cat, result3)
  
  p <- p+
    # geom_vline(xintercept = data_plot$distribution_mean, color= "red")+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("MannW= ", round(mann_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -9
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("BH= ", round(adjusted_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -11
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Effect size Hedges g= ", round(effect_size, 3))),
      hjust   = 0,
      vjust   = -17
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Hedges cat= ", effect_size_cat)),
      hjust   = 0,
      vjust   = -15
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals= ", nbr_individuals)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$sample_frequency)
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste("Is in cosmic?", variant_in_cosmic.x,",",Mutation.somatic.status.x)),
      colour = "purple",
      hjust   = 0,
      vjust   = -35
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("HGVSP= ", HGVSP.x)),
      colour = "purple",
      hjust   = 0,
      vjust   = -33
    )+
    geom_text(
      data    = data_plot %>%
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("occurrence= ", occurrence_in_cosmic.x)),
      colour = "purple",
      hjust   = 0,
      vjust   = -31
    )
  
  print(p)

}
```
