---
title: "Inverstigate chr2 pval"
author: "Christelle Colin-Leitzinger"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: no
    theme: cerulean
    highlight: pygments
    df_print: paged
editor_options:
  chunk_output_type: console
---

<style type="text/css">
h1.title {
  font-size: 25px;
}
.figure {
   margin-top: 25px;
   margin-bottom: 25px;
}

table {
    margin-top: 25px;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.align='center'
                      )
options(gtsummary.print_engine = "gt")
options(gtsummary.as_gt.addl_cmds = "gt::tab_options(table.font.size = 12, data_row.padding = gt::px(1))")
```


<style>
div.blue { background-color:#0099CC; border-radius: 5px; padding: 20px; font-size: 38px}
</style>
<div class = "blue">

<span style="color: white;">GnomAD data</span>

</div>
<br>

```{r library, include=FALSE}
library(tidyverse)
library(effsize)

theme_set(theme_classic())
```

```{r load}
path <- fs::path("", "Volumes", "Lab_Gillis", "Christelle")
gnomad_pval <-
  read.delim(paste0(path, "/gnomAD_raw_data/chr2_test/padjust_age_selected_variants/chr2_padjusted.vcf.gz"),
              sep = " ")

gnomad <-
  read.delim(paste0(path, "/gnomAD_raw_data/chr2_test/chr2_gnomad_age_mannw.vcf.gz"),
              sep = " ")

# gnomad_pval <-
#   read.delim("/Users/colinccm/Documents/Labs/Gillis/chr2_padjusted.vcf.gz",
#               sep = " ")
```

```{r}
gnomad <- gnomad %>% 
  left_join(., gnomad_pval %>% 
              select(IDs, adjusted_pval),
            by = "IDs")

rm(gnomad_pval)
```

# Mean and effect size
**CH R882H mutation is depicted in red**
```{r}
gnomad %>% 
  select(IDs, nbr_individuals, mann_w, adjusted_pval) %>% 
  mutate(order= ifelse(IDs=="2-25234373-C-T", 2, 1)) %>% 
  arrange(order) %>% 
  ggplot(aes(x= adjusted_pval, y= mann_w, 
             color= IDs == "2-25234373-C-T",
             order= order))+
  geom_point()+
  scale_color_manual(breaks=c('FALSE', 'TRUE'), values = c("grey", "red"))+
  theme(legend.position = "none")

gnomad %>% 
  select(IDs, nbr_individuals, mann_w, adjusted_pval) %>% 
  mutate(order= ifelse(IDs=="2-25234373-C-T", 2, 1)) %>% 
  arrange(order) %>% 
  ggplot(aes(x= mann_w, y= nbr_individuals, 
             color= IDs == "2-25234373-C-T",
             order= order))+
  geom_point()+
  scale_color_manual(breaks=c('FALSE', 'TRUE'), values = c("grey", "red"))+
  theme(legend.position = "none")

gnomad %>% 
  select(IDs, nbr_individuals, mann_w, adjusted_pval) %>% 
  mutate(order= ifelse(IDs=="2-25234373-C-T", 2, 1)) %>% 
  arrange(order) %>% 
  ggplot(aes(x= adjusted_pval, y= nbr_individuals, 
             color= IDs == "2-25234373-C-T",
             order= order))+
  geom_point()+
  scale_color_manual(breaks=c('FALSE', 'TRUE'), values = c("grey", "red"))+
  theme(legend.position = "none")
```

```{r}
All <- gnomad %>% 
  filter(IDs == "All individuals"
  ) %>% 
  # mutate(IDs = "Reference") %>% 
  select(distribution_mean_ref = distribution_mean)

data_plot <- gnomad %>%
  # filter(IDs == i
  # ) %>% 
  select("IDs", distribution_mean, adjusted_pval) %>% 
  bind_cols(., All) %>% 
  mutate(effect = distribution_mean - distribution_mean_ref) # need to keep sup values

data_plot %>% 
  slice(1:20) %>% 
  ggplot(aes(x = IDs, y= effect, color= effect > 0
             #, color= IDs == "All individuals"
  ))+
  geom_point()+
  geom_hline(yintercept = 0)+
  coord_flip()
```

On all the variants
```{r}
library(ggforce)
data_plot %>% 
  mutate(order= ifelse(IDs=="2-25234373-C-T", 2, 1)) %>% 
  arrange(order) %>% 
  ggplot(aes(x= effect, y= adjusted_pval, 
             color= IDs == "2-25234373-C-T",
             order= order))+
  geom_point()+
  scale_y_log10()+
  labs(y= "BH pval (log10)")+
  scale_color_manual(breaks=c('FALSE', 'TRUE'), values = c("grey", "red"))+
  theme(legend.position = "none")

data_plot %>% 
  mutate(order= ifelse(IDs=="2-25234373-C-T", 2, 1)) %>% 
  arrange(order) %>% 
  ggplot(aes(x= effect, y= adjusted_pval, 
             color= IDs == "2-25234373-C-T",
             order= order))+
  geom_point()+
  scale_y_continuous(trans = trans_reverser('log10'))+
  # scale_y_continuous(trans = "log2")+
  labs(y= "BH pval (log10)")+
  scale_color_manual(breaks=c('FALSE', 'TRUE'), values = c("grey", "red"))+
  theme(legend.position = "none")
```

On BH < 0.05
```{r}
library(ggforce)
data_plot %>% 
  filter(adjusted_pval < 0.05) %>% 
  mutate(order= ifelse(IDs=="2-25234373-C-T", 2, 1)) %>% 
  arrange(order) %>% 
  ggplot(aes(x= effect, y= adjusted_pval, 
             color= IDs == "2-25234373-C-T",
             order= order))+
  geom_point()+
  scale_y_log10()+
  labs(y= "BH pval (log10)")+
  scale_color_manual(breaks=c('FALSE', 'TRUE'), values = c("grey", "red"))+
  theme(legend.position = "none")

data_plot %>% 
  filter(adjusted_pval < 0.05) %>% 
  mutate(order= ifelse(IDs=="2-25234373-C-T", 2, 1)) %>% 
  arrange(order) %>% 
  ggplot(aes(x= effect, y= adjusted_pval, 
             color= IDs == "2-25234373-C-T",
             order= order))+
  geom_point()+
  scale_y_continuous(trans = trans_reverser('log10'))+
  # scale_y_continuous(trans = "log2")+
  labs(y= "BH pval (log10)")+
  scale_color_manual(breaks=c('FALSE', 'TRUE'), values = c("grey", "red"))+
  theme(legend.position = "none")
```

```{r}
gnomad_decoded <- gnomad %>%
  select(-age_bin, -sample_frequency, -sample_count, -nbr_individuals) %>% 
  mutate(gene_in_gnomad = str_match(INFO, "(DNMT3A)(.*?)$")[,2]) %>% 
  filter(IDs == "2-25457242-C-T" | IDs == "2-25234373-C-T" | gene_in_gnomad == "DNMT3A") %>%
  # filter(INFO == "Arg882His" | gene_in_gnomad == "DNMT3A") %>% 
  # mutate(prot_change = str_match(INFO, "(|R)(.*?)|$")[,2]) %>% 
  # mutate(prot_change1 = str_match(INFO, "(|R)(.*?)|$")[,3]) %>% 
  # slice(1:50) %>% 
  mutate(IDs = ifelse(IDs == "2-25234373-C-T", "2-25234373-C-T_R882H", IDs)) %>% 

  mutate(age_hist_het_bin_freq = str_match(INFO, "age_hist_het_bin_freq=(.*?);")[,2]) %>%
  # "Count of age values falling below lowest histogram bin edge for heterozygous individuals">
  mutate("<30_het" = as.numeric(str_match(INFO, "age_hist_het_n_smaller=(.*?);")[,2])) %>%
  # "Count of age values falling above highest histogram bin edge for heterozygous individuals">
  mutate(">80_het" = as.numeric(str_match(INFO, "age_hist_het_n_larger=(.*?);")[,2])) %>% 
  separate(col = age_hist_het_bin_freq,
           into = c("30-35_het","35-40_het","40-45_het","45-50_het",
                    "50-55_het","55-60_het","60-65_het","65-70_het",
                    "70-75_het","75-80_het"), 
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>% 
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>% 

  # 2. Extract count per age bin for heterozygous ind--
  mutate(age_hist_hom_bin_freq = str_match(INFO, "age_hist_hom_bin_freq=(.*?);")[,2]) %>%
  # "Count of age values falling below lowest histogram bin edge for homozygous individuals">
  mutate("<30_hom" = as.numeric(str_match(INFO, "age_hist_hom_n_smaller=(.*?);")[,2])) %>%
  # "Count of age values falling above highest histogram bin edge for homozygous individuals">
  mutate(">80_hom" = as.numeric(str_match(INFO, "age_hist_hom_n_larger=(.*?);")[,2])) %>% 
  separate(col = age_hist_hom_bin_freq,
           into = c("30-35_hom","35-40_hom","40-45_hom","45-50_hom",
                    "50-55_hom","55-60_hom","60-65_hom","65-70_hom",
                    "70-75_hom","75-80_hom"), 
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>% 
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>% 
  
  # 3. Combine count for Het and hom individuals
  mutate("<30" = rowSums(select(.,`30-35_het`,`30-35_hom`), na.rm = TRUE),
         "30-35" = rowSums(select(.,`30-35_het`,`30-35_hom`), na.rm = TRUE),
         "35-40" = rowSums(select(.,`35-40_het`,`35-40_hom`), na.rm = TRUE),
         "40-45" = rowSums(select(.,`40-45_het`,`40-45_hom`), na.rm = TRUE),
         "45-50" = rowSums(select(.,`45-50_het`,`45-50_hom`), na.rm = TRUE),
         "50-55" = rowSums(select(.,`50-55_het`,`50-55_hom`), na.rm = TRUE),
         "55-60" = rowSums(select(.,`55-60_het`,`55-60_hom`), na.rm = TRUE),
         "60-65" = rowSums(select(.,`60-65_het`,`60-65_hom`), na.rm = TRUE),
         "65-70" = rowSums(select(.,`65-70_het`,`65-70_hom`), na.rm = TRUE),
         "70-75" = rowSums(select(.,`70-75_het`,`70-75_hom`), na.rm = TRUE),
         "75-80" = rowSums(select(.,`75-80_het`,`75-80_hom`), na.rm = TRUE),
         ">80" = rowSums(select(.,`>80_het`,`>80_hom`), na.rm = TRUE)
         ) %>%
  add_row("IDs"="All individuals", "ID"="All individuals of any genotype bin",
          "<30" = 2547, "30-35" = 3423,
          "35-40" = 4546,"40-45" = 8487,
          "45-50" = 10355,"50-55" = 12693,
          "55-60" = 11933,"60-65" = 10534,
          "65-70" = 8882,"70-75" = 5991,
          "75-80" = 4136,
          ">80" = 1935) %>%
  mutate(nbr_individuals = rowSums(select(.,`<30`:`>80`), na.rm = TRUE)) %>%
  
  select(-c(ends_with("_het"), ends_with("_hom"))) %>% 
  
  select(IDs, everything())
```

```{r}
gnomad_decoded <- gnomad_decoded %>% 
  pivot_longer(cols = c(grep("[[:digit:]]", colnames(.))), 
               names_to = "age_bin", values_to = "sample_count") %>% 
  mutate(age_bin = factor(age_bin, 
                          levels = c("<30", "30-35","35-40","40-45","45-50","50-55",
                                     "55-60","60-65","65-70","70-75","75-80", ">80"
                                     ))) %>% 
  mutate(bin_age = case_when(
    # The average age at diagnosis is 8 overall (ages 0 to 19), 
    # 5 years old for children (aged 0 to 14), and 17 years old 
    # for adolescents (aged 15 to 19), while adults' average 
    # age for cancer diagnosis is 65
    str_detect(age_bin, "<30")              ~ 15, 
    str_detect(age_bin, "30-35")            ~ 32.5,
    str_detect(age_bin, "35-40")            ~ 37.5,
    str_detect(age_bin, "40-45")            ~ 42.5,
    str_detect(age_bin, "45-50")            ~ 47.5,
    str_detect(age_bin, "50-55")            ~ 52.5,
    str_detect(age_bin, "55-60")            ~ 57.5,
    str_detect(age_bin, "60-65")            ~ 62.5,
    str_detect(age_bin, "65-70")            ~ 67.5,
    str_detect(age_bin, "70-75")            ~ 72.5,
    str_detect(age_bin, "75-80")            ~ 77.5,
    TRUE ~ 90
  )) %>% 
  mutate(sample_frequency = sample_count / nbr_individuals)
```

# Plots all variants
**R882H is curve #6.**
```{r}
mann_w <-  data.frame(matrix(nrow=1, ncol=0)) 
effect_size <-  data.frame(matrix(nrow=1, ncol=0)) 
effect_size_cat <-  data.frame(matrix(nrow=1, ncol=0)) 

n <- 0

set.seed(1234)

for(i in unique(gnomad_decoded$IDs)) {
  
  All <- gnomad_decoded %>% 
    filter(IDs == "All individuals"
  ) %>% 
    mutate(IDs = "Reference")
  data_plot <- gnomad_decoded %>% 
    filter(IDs == i
    ) %>% 
    bind_rows(., All) %>% 
    select("IDs", "age_bin", sample_frequency, bin_age, distribution_mean,
           nbr_individuals, #prot_change, 
           adjusted_pval)
  n <- n + 1
  p <- data_plot %>% 
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    # ggplot(aes(x = age_bin, y= sample_frequency, color= IDs))+
    # geom_bar(stat = "identity", aes(alpha= IDs))
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                # method.args=list(family=quasibinomial)#,
                span = 0.6
    )+
    labs(title = n)+
    geom_col(data = data_plot %>% 
               filter(IDs == i) ,
             aes(x= bin_age, y= sample_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$sample_frequency))
  
  p_ <- layer_data(p, 1) %>% 
  mutate(colour =  as.factor(colour)) %>% 
  mutate_if(is.numeric, ~replace(., is.na(.), 0))

  variant <- p_ %>% filter(group == 1)
  variant <- sample(variant$x, 180, prob = variant$y, replace = TRUE)
  ref <- p_ %>% filter(group == 2)
  ref <- sample(ref$x, 180, prob = ref$y, replace = TRUE)
  
  result1 <- wilcox.test(variant, ref, alternative = "greater")$p.value
  mann_w <- rbind(mann_w, result1)
  
  result2 <- cohen.d(variant, ref, hedges.correction=TRUE)$estimate
  effect_size <- rbind(effect_size, result2)
  result3 <- cohen.d(variant, ref,hedges.correction=TRUE)$magnitude %>% 
    as_tibble() %>% 
    mutate(value = case_when(
      value == "negligible"   ~ "negligible",
      value == "small"        ~ "small",
      value == "medium"       ~ "medium",
      value == "large"        ~ "large"
    ))
  effect_size_cat <- rbind(effect_size_cat, result3)
  
  p <- p+
    geom_vline(xintercept = data_plot$distribution_mean, color= "red")+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("MannW= ", round(result1, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -12
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("BH= ", round(adjusted_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -14
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Effect size Hedges g= ", round(result2, 3))),
      hjust   = 0,
      vjust   = -25
    )+
    geom_text(
      data    = data_plot,
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Hedges cat= ", result3)),
      hjust   = 0,
      vjust   = -23
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals= ", nbr_individuals)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$sample_frequency)
    )#+
    # geom_text(
    #   data    = data_plot %>% 
    #            filter(IDs == i),
    #   mapping = aes(x = -Inf, y = -Inf, label = paste0("prot change=", prot_change)),
    #   colour = "darkblue",
    #   hjust   = -1.5,
    #   vjust   = -min(data_plot$sample_frequency)
    # )
  
  print(p)

}

# mann_w <- mann_w %>% `colnames<-`(c("mann_w")) %>% 
#   mutate(IDs = c(unique(gnomad_decoded$IDs)))
effect_size <- effect_size %>% `colnames<-`(c("effect_size"))%>% 
  mutate(IDs = c(unique(gnomad_decoded$IDs)))
effect_size_cat <- effect_size_cat %>% `colnames<-`(c("effect_size_cat"))

stat_measure <- bind_cols(effect_size, effect_size_cat) %>% 
  left_join(gnomad_decoded, .,
  by= "IDs")

# print(head(stat_measure))
```

# Plots for BH < 0.05
```{r}
selected_variants <- stat_measure %>% filter(adjusted_pval < 0.05)
n <- 0

for(i in unique(selected_variants$IDs)) {
  
  All <- gnomad_decoded %>% 
    filter(IDs == "All individuals"
  ) %>% 
    mutate(IDs = "Reference")
  data_plot <- selected_variants %>% 
    filter(IDs == i
    ) %>% 
    bind_rows(., All) %>% 
    select("IDs", "age_bin", sample_frequency, bin_age, distribution_mean,
           nbr_individuals, mann_w, #prot_change, 
           adjusted_pval, effect_size, effect_size_cat)
  n <- n + 1
  p <- data_plot %>% 
    ggplot(aes(x = bin_age, y= sample_frequency, color= IDs))+
    # ggplot(aes(x = age_bin, y= sample_frequency, color= IDs))+
    # geom_bar(stat = "identity", aes(alpha= IDs))
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                # method.args=list(family=quasibinomial)#,
                span = 0.6
    )+
    labs(title = n)+
    geom_col(data = data_plot %>% 
               filter(IDs == i) ,
             aes(x= bin_age, y= sample_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$sample_frequency))
  
  p <- p+
    geom_vline(xintercept = data_plot$distribution_mean, color= "red")+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("MannW= ", round(mann_w, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -12
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("BH= ", round(adjusted_pval, 3))),
      colour = "black",
      hjust   = 0,
      vjust   = -14
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Effect size Hedges g= ", round(effect_size, 3))),
      hjust   = 0,
      vjust   = -25
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("Hedges cat= ", effect_size_cat)),
      hjust   = 0,
      vjust   = -23
    )+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = -Inf, y = -Inf, label = paste0("nbr_individuals= ", nbr_individuals)),
      colour = "darkblue",
      hjust   = 0,
      vjust   = -min(data_plot$sample_frequency)
    )#+
    # geom_text(
    #   data    = data_plot %>% 
    #            filter(IDs == i),
    #   mapping = aes(x = -Inf, y = -Inf, label = paste0("prot change=", prot_change)),
    #   colour = "darkblue",
    #   hjust   = -1.5,
    #   vjust   = -min(data_plot$sample_frequency)
    # )
  
  print(p)

}
```












