---
title: "Manuscript CH in GnomAD"
author: "Christelle Colin-Leitzinger"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: no
    theme: cerulean
    highlight: pygments
    df_print: paged
editor_options:
  chunk_output_type: console
---

<style type="text/css">
h1.title {
  font-size: 25px;
}
.figure {
   margin-top: 25px;
   margin-bottom: 25px;
}

table {
    margin-top: 25px;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.align='center'
                      )
```


<style>
div.blue { background-color:#0099CC; border-radius: 5px; padding: 20px; font-size: 38px}
</style>
<div class = "blue">

<span style="color: white;">Peripheral blood somatic mosaicism and clonal hematopoiesis inferred from 125,748 human exomes</span>

</div>
<br>

```{r library, include=FALSE}
library(tidyverse)
library(GenomicRanges)
library(gtsummary)
library(ggconsort)
library(ggbreak)
library(ggforce)
library(viridis)
library(ggridges)

theme_gtsummary_compact()
theme_set(theme_classic())
```

```{r load}
gnomad_variants <-
  read.delim(paste0(here::here(), "/chromosome data/chr2_CH_variants.vcf.gz"),
             colClasses = c("ALT"="character"),
              sep = " ")

vep_bick_genes <-
  readxl::read_xlsx(
    paste0(here::here(), 
           "/gnomad_clonal_mosaicism_variants_list_w_all_annotation_L_CHIP_09.12.2023.xlsx"), 
    na = c(".", "NA"))

considered_CH_variants <- 
  readxl::read_xlsx(paste0(here::here(), "/gene_ids.xlsx")) %>% 
  select(ids, Gene, Variant) %>% 
  tidyr::unite(col = "considered_variants", c(Gene : Variant), sep = " ")
```

```{r clean vep}
vep_bick_genes <- vep_bick_genes %>% 
  mutate(mutations_in_BickWHO = case_when(
    In_Bick_WHO == "TRUE"              ~ "Yes",
    TRUE                               ~ NA_character_
  )) %>% 
  mutate(Polyphen2_HDIV_pred = case_when(
    Polyphen2_HDIV_pred == "D"     ~ "Probably damaging (>=0.957)",
    Polyphen2_HDIV_pred == "P"     ~ "Possibly damaging (0.453<=pp2_hdiv<=0.956)",
    Polyphen2_HDIV_pred == "B"     ~ "Benign (pp2_hdiv<=0.452)"
  )) %>% 
  mutate(SIFT_pred = case_when(
    SIFT_pred == "D"               ~ "Deleterious",
    SIFT_pred == "T"               ~ "Tolerated"
  )) %>% 
  mutate(FATHMM_pred = case_when(
    FATHMM_pred == "D"             ~ "Deleterious",
    FATHMM_pred == "T"             ~ "Tolerated"
  )) %>% 
  mutate(CADD_phred_cat = case_when(
    CADD_phred >= 20               ~ "Likely pathogenic",
    CADD_phred < 20                ~ "< 20"
  )) %>% 
  mutate(new_HGVSc_2 = str_match(AAChange.refGene, "c.(.)([:alnum:]*)(.):p.")[,2]) %>% 
  mutate(new_HGVSc_4 = str_match(AAChange.refGene, "c.(.)([:alnum:]*)(.):p.")[,4]) %>% 
  unite(new_HGVSc, c(new_HGVSc_2, new_HGVSc_4), sep = ">", na.rm = TRUE) %>%
  mutate(new_HGVSc = case_when(
    new_HGVSc == ""                ~ NA_character_,
    TRUE                           ~ new_HGVSc
  )) %>% 
  select(-c(In_Bick_WHO, X.CHROM))
```


```{r merge vep considered variants with data}
gnomad_variants <- gnomad_variants %>% 
  # Create filter for removing noise in AB
  # mutate(ab_hist_alt_bin_freq = str_match(INFO, "ab_hist_alt_bin_freq=(.*?);")[,2]) %>%
  # separate(col = ab_hist_alt_bin_freq,
  #          into = c("0.00-0.05"),
  #          sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>%
  # mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>%
  # mutate(contaminated_AB = case_when(
  #   `0.00-0.05` == total_allele_balance               ~ "sequencing noise",
  #   TRUE                                              ~ "clean sequencing"
  # )) %>% 
  # select(-c("0.00-0.05")) %>% 
  # Add VEP and Bick&WHO
  left_join(., vep_bick_genes,
            by= "IDs") %>% 
  mutate(mutations_in_BickWHO = case_when(
    mutations_in_BickWHO == "Yes"              ~ "Yes",
    TRUE                                       ~ "No"
  ), mutations_in_BickWHO = factor(mutations_in_BickWHO, levels = c("No", "Yes")))

gnomad_variants <- gnomad_variants %>% 
  left_join(., considered_CH_variants,
            by = c("IDs" = "ids"))

# cleaning
rm(considered_CH_variants,
   vep_bick_genes)
```

We want to look at skewness of age distribution OR AB in GnomAD.  
GnomAD has data available for :  
- v2 release is composed of 125,748 exomes and 15,708 genomes (GRCh37)  
- gnomAD structural variant (SV) v2.1 represents 10,847 genomes (GRCh37)  
- v3.1 spans 76,156 genomes (GRCh38)  

The gnomAD v2.1 data set contains data from 125,748 exomes and 15,708 whole genomes, all mapped to the GRCh37/hg19 reference sequence. The gnomAD v3.1 data set contains 76,156 whole genomes (and no exomes), all mapped to the GRCh38 reference sequence. Most of the genomes from v2 are included in v3.1.

download page : https://gnomad.broadinstitute.org/downloads  
Is ASXL1 real?
https://gnomad.broadinstitute.org/variant/20-31022441-A-AG?dataset=gnomad_r2_1

<br>
<br>

***
<br>

# I. Filtering steps for potential CH variants selection
## Here are the selection steps with few figures whicg will not be added to the paper but they are for sanity check.

The number of variants in chromosome `r as.character(gnomad_variants %>% distinct(X.CHROM))` is `r nrow(gnomad_variants %>% distinct(IDs))`.
```{r removing variants with AB only <5%}
clean_AB_variants <- gnomad_variants %>% 
  filter(contaminated_AB == "clean sequencing") %>% 
  filter(black_list == 0) %>% 
  filter(centromeres == 0) %>% 
  filter(segmental_duplication == 0) %>% 
  filter(wm_sdust == 0)
```
In chromosome `r as.character(clean_AB_variants %>% distinct(X.CHROM))`, the number of "non-contaminated" variants (aka variants showing AB with more than 0.00-0.05) is `r nrow(clean_AB_variants %>% distinct(IDs))`.

```{r}
allele_depth <- 10
depth_variants <- clean_AB_variants %>% 
  filter(dp_het_median >= allele_depth)
```
We have `r nrow(depth_variants %>% distinct(IDs))` with an allele depth greater or equal to `r allele_depth`.

```{r protein coding}
prot_coding <- depth_variants %>%
  filter(is_gencode_protein_coding_gene == "Yes")
```
In chromosome `r as.character(prot_coding %>% distinct(X.CHROM))`, the number of variants which are on a protein coding gene is `r nrow(prot_coding %>% distinct(IDs))`.

```{r AB BH}
pval_selection <- 0.05
AB_distribution_variants <- prot_coding %>% 
  filter(adjusted_AB_pval < pval_selection)
```
In chromosome `r as.character(AB_distribution_variants %>% distinct(X.CHROM))`, the number Clonal mosaicism of variants with BH on AB distribution < `r pval_selection` is `r nrow(AB_distribution_variants %>% distinct(IDs))`.

```{r AB Hedges g}
AB_hedges_selection <- "large"
clonal_mosaicism <- AB_distribution_variants %>%
  # filter(effect_size_cat != "negligible")
  filter(AB_effect_size_cat == AB_hedges_selection)
```
In chromosome `r as.character(clonal_mosaicism %>% distinct(X.CHROM))`, the number of variants with `r AB_hedges_selection` Hedges g effect size is `r nrow(clonal_mosaicism %>% distinct(IDs))`.

```{r}
cosmic_var <- clonal_mosaicism %>% 
  filter(variant_in_cosmic == "Yes")
```
Variant in cosmic : `r nrow(cosmic_var %>% distinct(IDs))`.


```{r age BH}
age_distribution_variants <- cosmic_var %>% 
  filter(adjusted_age_pval < pval_selection)
```
In chromosome `r as.character(age_distribution_variants %>% distinct(X.CHROM))`, the number of variants with BH on Age distribution < `r pval_selection` is `r nrow(age_distribution_variants %>% distinct(IDs))`.

```{r age Hedges g}
age_hedges_selection <- "negligible|small"
CH_variants <- age_distribution_variants %>%
  filter(!str_detect(age_effect_size_cat, age_hedges_selection))
```
In chromosome `r as.character(CH_variants %>% distinct(X.CHROM))`, the number of variants with Hedges g effect size better than `r age_hedges_selection %>% str_replace(., "\\|", " or ")` is `r nrow(CH_variants %>% distinct(IDs))`. These variants are located on `r nrow(CH_variants %>% distinct(IDs, .keep_all = TRUE) %>% distinct(SYMBOL))` genes.

# Final number of variants in chromosome `r as.character(CH_variants %>% distinct(X.CHROM))` : `r nrow(CH_variants %>% distinct(IDs))`

<br>
<br>

***
<br>

# **Figure 1** GnomAD variants description / description AB filter
## **Figure 1A** Consort diagram
```{r consort, fig.width=12, out.width = '75%'}
# study_cohorts <-
#   gnomad_variants %>%
#   distinct(IDs, .keep_all = TRUE) %>%
#   cohort_start("**Variants considered**") %>%
#   # Define cohorts using named expressions
#   # Notice that you can use previously defined cohorts in subsequent steps
#   cohort_define(
# 
#     clean_seq = .full %>%
#       filter(contaminated_AB == "clean sequencing"),
#     black_clean = clean_seq %>%
#       filter(black_list == 0),
#     centro_clean = black_clean %>%
#       filter(centromeres == 0),
#     dup_clean = centro_clean %>%
#       filter(segmental_duplication == 0),
#     wm_sdust_clean = dup_clean %>%
#       filter(wm_sdust == 0),
#     clean_AB_variants = wm_sdust_clean %>%
#       filter(dp_het_median >= allele_depth),
# 
#     protcoding_variants = clean_AB_variants  %>%
#       filter(is_gencode_protein_coding_gene == "Yes"),
# 
#     AB_BH = protcoding_variants %>%
#       filter(adjusted_AB_pval <= pval_selection),
#     clonal_mosaicism = AB_BH %>%
#       filter(AB_effect_size_cat == AB_hedges_selection),
# 
#     cosmic_variants = clonal_mosaicism  %>%
#       filter(variant_in_cosmic == "Yes"),
# 
#     age_BH = cosmic_variants  %>%
#       filter(adjusted_age_pval <= pval_selection),
#     CH_variants = age_BH %>%
#       filter(!str_detect(age_effect_size_cat, age_hedges_selection)),
# 
#     CH_variants_end = CH_variants,
# 
#     # anti_join is useful for counting exclusions
#     excluded2 = anti_join(.full, clean_AB_variants, by = "IDs"),
#     excluded2_1 = anti_join(.full, clean_seq, by = "IDs"),
#     excluded2_2 = anti_join(clean_seq, black_clean, by = "IDs"),
#     excluded2_3 = anti_join(black_clean, centro_clean, by = "IDs"),
#     excluded2_4 = anti_join(centro_clean, dup_clean, by = "IDs"),
#     excluded2_5 = anti_join(dup_clean, wm_sdust_clean, by = "IDs"),
#     excluded2_6 = anti_join(wm_sdust_clean, clean_AB_variants, by = "IDs"),
# 
#     excluded3 = anti_join(clean_AB_variants, protcoding_variants, by = "IDs"),
# 
#     excluded1 = anti_join(protcoding_variants, clonal_mosaicism, by = "IDs"),
#     excluded1_1 = anti_join(protcoding_variants, AB_BH, by = "IDs"),
#     excluded1_2 = anti_join(AB_BH, clonal_mosaicism, by = "IDs"),
#     
#     excluded5 = anti_join(clonal_mosaicism, cosmic_variants, by = "IDs"),
# 
#     excluded4 = anti_join(cosmic_variants, CH_variants, by = "IDs"),
#     excluded4_1 = anti_join(cosmic_variants, age_BH, by = "IDs"),
#     excluded4_2 = anti_join(age_BH, CH_variants, by = "IDs")
#   ) %>%
#   # Provide text labels for cohorts
#   cohort_label(
#     excluded2 = "<span style='color:darkblue'>Exclude variants with sequencing artifacts</span>",
#     excluded2_1 = "Exclude variants with sequencing artifacts",
#     excluded2_2 = "Exclude 'curated “blacklist” developed by ENCODE (73), and excluded mutations in the blacklisted regions'",
#     excluded2_3 = "Exclude 'Centromeres (“Centromeres” in the “Mapping and Sequencing” group)'",
#     excluded2_4 = "Exclude 'Segmental duplications (“segmental dups”)'",
#     excluded2_5 = "Exclude 'Low complexity regions (“WM + SDust”) track'",
#     excluded2_6 = "Exclude variants with depth inferieur to 10",
#     excluded3 = "<span style='color:darkblue'>Select for protein coding mutations</span>",
#     excluded1 = "<span style='color:darkblue'>Exclude germline variants</span>",
#     excluded1_1 = "BH pval < 0.05",
#     excluded1_2 = "Hedges g == 'large'",
#     excluded5 = "<span style='color:darkblue'>Select variants present in Cosmic</span>",
#     excluded4 = "<span style='color:darkblue'>Exclude non-age-skewed mutations</span>",
#     excluded4_1 = "BH pval < 0.05",
#     excluded4_2 = "Hedges g == 'large' or 'medium'",
#     clean_AB_variants = "**Variants after quality control filtering**",
#     protcoding_variants = "**Protein-coding somatic mutations**",
#     clonal_mosaicism = "**Peripheral blood somatic mutations (clonal_mosaicism)**",
#     cosmic_variants = "**Candidate variants in COSMIC**",
#     CH_variants_end = "**Age-dependent peripheral blood somatic mutations**"
#   )
# study_consort <- study_cohorts %>%
#   consort_box_add(
#     # top eligibility box at 40 height
#     "full", 0.05, 50, cohort_count_adorn(study_cohorts, .full)
#   ) %>%
#     # first top excluded box at 0.2 (right from 0.05 eligibility box)
#   consort_box_add(
#     "exc_contamination", 0.2, 45, glue::glue(
#       "{cohort_count_adorn(study_cohorts, excluded2)}<br>
#       • {cohort_count_adorn(study_cohorts, excluded2_1)}<br>
#       • {cohort_count_adorn(study_cohorts, excluded2_2)}<br>
#       • {cohort_count_adorn(study_cohorts, excluded2_3)}<br>
#       • {cohort_count_adorn(study_cohorts, excluded2_4)}<br>
#       • {cohort_count_adorn(study_cohorts, excluded2_5)}<br>
#       • {cohort_count_adorn(study_cohorts, excluded2_6)}
#       ")
#   ) %>%
#   consort_box_add(
#     "clean_AB_variants", 0.05, 40, cohort_count_adorn(study_cohorts, clean_AB_variants)
#   ) %>%
#   consort_box_add(
#     "exc_non_protein_coding", 0.2, 35, glue::glue(
#       "{cohort_count_adorn(study_cohorts, excluded3)}<br>
#       • include variants present in a protein coding gene
#       ")
#   ) %>%
#   consort_box_add(
#     "protcoding_variants", 0.05, 30, cohort_count_adorn(study_cohorts, protcoding_variants)
#   ) %>%
#     consort_box_add(
#     "exc_germline", 0.2, 25, glue::glue(
#       "{cohort_count_adorn(study_cohorts, excluded1)}<br>
#       • {cohort_count_adorn(study_cohorts, excluded1_1)}<br>
#       • {cohort_count_adorn(study_cohorts, excluded1_2)}
#       ")
#   ) %>%
#   consort_box_add(
#     "clonal_mosaicism", 0.05, 20, cohort_count_adorn(study_cohorts, clonal_mosaicism)
#   ) %>%
#   consort_box_add(
#     "exc_non_cosmic", 0.2, 15, glue::glue(
#       "{cohort_count_adorn(study_cohorts, excluded5)}<br>
#       • Select variants present in Cosmic
#       ")
#   ) %>%
#   consort_box_add(
#     "cosmic_variants", 0.05, 10, cohort_count_adorn(study_cohorts, cosmic_variants)
#   ) %>%
#   consort_box_add(
#     "exc_non_age_skew", 0.2, 5, glue::glue(
#       "{cohort_count_adorn(study_cohorts, excluded4)}<br>
#       • {cohort_count_adorn(study_cohorts, excluded4_1)}<br>
#       • {cohort_count_adorn(study_cohorts, excluded4_2)}
#       ")
#   ) %>%
#   # Add bottom box
#   consort_box_add(
#     "CH_variants", 0.05, 0, cohort_count_adorn(study_cohorts, CH_variants_end)
#   ) %>%
#   # Add exclusion arrows
#   consort_arrow_add(
#     end = "exc_contamination", end_side = "left", start_x = 0.05, start_y = 45
#   ) %>%
#   consort_arrow_add(
#     end = "exc_non_protein_coding", end_side = "left", start_x = 0.05, start_y = 35
#   ) %>%
#   consort_arrow_add(
#     end = "exc_germline", end_side = "left", start_x = 0.05, start_y = 25
#   ) %>%
#   consort_arrow_add(
#     end = "exc_non_cosmic", end_side = "left", start_x = 0.05, start_y = 15
#   ) %>%
#   consort_arrow_add(
#     end = "exc_non_age_skew", end_side = "left", start_x = 0.05, start_y = 5
#   ) %>%
#   # Add top to bottom arrow
#   consort_arrow_add(
#     end = "CH_variants", end_side = "top", start_x = 0.05, start_y = 50
#   )
# jpeg("consort diagram.jpeg", width = 950, height = 650)
# study_consort %>%
#   ggplot() +
#   geom_consort() + #theme_classic()
#   theme_consort(margin_h = c(1,12), # bottom, left
#                 margin_v = c(1,40)) # top, right
# dev.off()

knitr::include_graphics('/Users/colinccm/Documents/GitHub/Gillis/CH_gnomAD/consort diagram.jpeg')
```
<br>

## **Figure 1B**. Representation of filters on AB distributuion of overall variants
```{r, fig.width=12}
library(VennDiagram)
# venn.diagram(
#   x = list(gnomad_variants$IDs,
#            prot_coding$IDs,
#            clonal_mosaicism$IDs,
#            cosmic_var$IDs,
#            CH_variants$IDs
#            ),
#   category.names = c("Study Population",
#                      "Clean protein \ncoding variants",
#                      "Clonal mosaicism",
#                      "Cosmic \nvariants",
#                      "CH variants"),
#   filename = 'VennDiagram.png',
#   output=TRUE,
#   disable.logging = TRUE,
# 
#   # Output features
#   imagetype="png" ,
#   height = 1000 ,
#   width = 1000 ,
#   resolution = 300,
#   compression = "lzw",
# 
#   # Circles
#   lwd = 2,
#   lty = 'blank',
#   fill = c(viridis::magma(5)),
#   margin = 0.2,
# 
#   # Numbers
#   cex = .6,
#   fontface = "bold",
#   fontfamily = "sans",
#   # cat.pos = c(-90, 0, 90), # effect, Population , BH
#   # cat.dist = c(0.09, 0.1, 0.1), # Population, BH, effect
#   ext.percent = 2
#   #ext.percent = 5
# )
# knitr::include_graphics('/Users/colinccm/Documents/GitHub/Gillis/CH_gnomAD/Rmd/VennDiagram.png')

library(ggraph)
library(igraph)

cat_to_cat <- tibble(from= c("gnomad_variants", "prot_coding", "clonal_mosaicism", "cosmic_var", 
                             "gnomad_variants", "prot_coding", "clonal_mosaicism", "cosmic_var"),
to= c("prot_coding", "clonal_mosaicism", "cosmic_var", "CH_variants", 
      "Lost_incleaning", "lost_inAB", "lost_inCosmic", "lost_inAge"))

size_plot <- tibble(name= c("gnomad_variants", "prot_coding", "clonal_mosaicism", "cosmic_var", "CH_variants", 
                            "Lost_incleaning", "lost_inAB", "lost_inCosmic", "lost_inAge"),
size= c(265806, 182382, 30578, 1566, 257,
        83424, 151804, 29012, 1309),
shortName= c("gnomad_variants", "prot_coding", "clonal_mosaicism", "cosmic_var", "CH_variants", 
             "Lost_incleaning", "lost_inAB", "lost_inCosmic", "lost_inAge"))
mygraph1 <- graph_from_data_frame(cat_to_cat, vertices=size_plot)

ggraph(mygraph1, layout = 'circlepack', weight=size) + 
  geom_node_circle(data= . %>% filter(shortName %in% c("gnomad_variants", "prot_coding", 
                                                       "clonal_mosaicism", "cosmic_var",
                                                       "CH_variants")),
                   aes(fill= factor(shortName,
                                    levels= c("gnomad_variants", "prot_coding", 
                                              "clonal_mosaicism", "cosmic_var",
                                              "CH_variants"
                                    )))) +
  scale_fill_manual(values = c("#440154FF",  "#3B528BFF",  "#21908CFF",  "#5DC863FF",  "#FDE725FF"),
                    name = NULL)


library(ggalluvial)









fig1_b <- gnomad_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, AB_distribution_mean) %>% 
  mutate(data = "Variants considered") %>% 

  bind_rows(., prot_coding %>%
              distinct(IDs, .keep_all = TRUE) %>% 
              select(IDs, AB_distribution_mean) %>% 
              mutate(data = "Clean Protein-coding somatic mutations")) %>% 
  bind_rows(., clonal_mosaicism %>%
              distinct(IDs, .keep_all = TRUE) %>% 
              select(IDs, AB_distribution_mean) %>% 
              mutate(data = "Peripheral blood somatic mutations")) %>% 
  bind_rows(., cosmic_var %>%
              distinct(IDs, .keep_all = TRUE) %>% 
              select(IDs, AB_distribution_mean) %>% 
              mutate(data = "Candidate variants in COSMIC")) %>% 
  bind_rows(., CH_variants %>%
              distinct(IDs, .keep_all = TRUE) %>% 
              select(IDs, AB_distribution_mean) %>% 
              mutate(data = "Age-dependent peripheral blood somatic mutations")) %>% 
  
  mutate(AB_distribution_mean = round(AB_distribution_mean, 2)) %>% 
  filter(!is.na(AB_distribution_mean)) %>% 
  group_by(AB_distribution_mean, data) %>% 
  summarise(count = n()) %>% 
  ungroup() %>% 
  mutate(data = factor(data, levels= c("Variants considered",
                                       "Clean Protein-coding somatic mutations",
                                       "Peripheral blood somatic mutations", 
                                       "Candidate variants in COSMIC",
                                       "Age-dependent peripheral blood somatic mutations"
                                         ))) 

fig1_b %>%
  ggplot(aes(x= AB_distribution_mean, y=count, fill= data
             ))+
  geom_area(position = "identity") +
  scale_fill_viridis(discrete = TRUE, name = NULL)+
  labs(x= paste("Allele Balance distribution for variants in chromosome",
                as.character(gnomad_variants %>% distinct(X.CHROM)), "`(mean)"),
       y= "Number of variants"
  )



fig1_b %>%
  ggplot(aes(x= AB_distribution_mean, y=count, fill= data
             ))+
  geom_area(position = "identity") +
  scale_fill_viridis(discrete = TRUE, name = NULL)+
  labs(x= paste("Allele Balance distribution for variants in chromosome",
                as.character(gnomad_variants %>% distinct(X.CHROM)), "`(mean)"),
       y= "Number of variants"
  )+
  coord_cartesian(ylim = c(0,5000))


fig1_b %>%
  ggplot(aes(x= AB_distribution_mean, y=count, fill= data
             ))+
  geom_area(position = "identity") +
  scale_fill_viridis(discrete = TRUE, name = NULL)+
  labs(x= paste("Allele Balance distribution for variants in chromosome",
                as.character(gnomad_variants %>% distinct(X.CHROM)), "`(mean)"),
       y= "Number of variants"
  )+
  facet_zoom(ylim = c(0, 5000))
```

```{r, fig.width=12}
fig1_b <- gnomad_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, AB_distribution_mean) %>% 
  mutate(data = "Variants considered") %>% 

  bind_rows(., prot_coding %>%
              distinct(IDs, .keep_all = TRUE) %>% 
              select(IDs, AB_distribution_mean) %>% 
              mutate(data = "Clean Protein-coding somatic mutations")) %>% 
  bind_rows(., clonal_mosaicism %>%
              distinct(IDs, .keep_all = TRUE) %>% 
              select(IDs, AB_distribution_mean) %>% 
              mutate(data = "Peripheral blood somatic mutations")) %>% 
  
  mutate(AB_distribution_mean = round(AB_distribution_mean, 2)) %>% 
  filter(!is.na(AB_distribution_mean)) %>% 
  group_by(AB_distribution_mean, data) %>% 
  summarise(count = n()) %>% 
  ungroup() %>% 
  mutate(data = factor(data, levels= c("Variants considered",
                                       "Clean Protein-coding somatic mutations",
                                       "Peripheral blood somatic mutations"
                                         ))) 

fig1_b %>%
  ggplot(aes(x= AB_distribution_mean, y=count, fill= data
             ))+
  geom_area(position = "identity") +
  scale_fill_viridis(discrete = TRUE, name = NULL, option = "A")+
  labs(x= paste("Allele Balance distribution for variants in chromosome",
                as.character(gnomad_variants %>% distinct(X.CHROM)), "`(mean)"),
       y= "Number of variants"
  )
```

```{r, fig.width=12}
fig1_b <- clonal_mosaicism %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, AB_distribution_mean) %>% 
  mutate(data = "Peripheral blood somatic mutations") %>% 
  bind_rows(., cosmic_var %>%
              distinct(IDs, .keep_all = TRUE) %>% 
              select(IDs, AB_distribution_mean) %>% 
              mutate(data = "Candidate variants in COSMIC")) %>% 
  bind_rows(., CH_variants %>%
              distinct(IDs, .keep_all = TRUE) %>% 
              select(IDs, AB_distribution_mean) %>% 
              mutate(data = "Age-dependent peripheral blood somatic mutations")) %>% 
  
  mutate(AB_distribution_mean = round(AB_distribution_mean, 2)) %>% 
  filter(!is.na(AB_distribution_mean)) %>% 
  group_by(AB_distribution_mean, data) %>% 
  summarise(count = n()) %>% 
  ungroup() %>% 
  mutate(data = factor(data, levels= c("Peripheral blood somatic mutations", 
                                       "Candidate variants in COSMIC",
                                       "Age-dependent peripheral blood somatic mutations"
                                         ))) 

fig1_b %>%
  ggplot(aes(x= AB_distribution_mean, y=count, fill= data
             ))+
  geom_area(position = "identity") +
  scale_fill_viridis(discrete = TRUE, name = NULL, option = "E")+
  labs(x= paste("Allele Balance distribution for variants in chromosome",
                as.character(gnomad_variants %>% distinct(X.CHROM)), "`(mean)"),
       y= "Number of variants"
  )
```
<br>
<br>

# **Figure 2** Description of the CH variants - Only focus on CH variants after filtering
## **Figure 2A**  Prevalence plot - How many individuals who have mutation in 1 gene
`r emo::ji("pencil")` <span style="color: red;">Removed prevalence plot</span>
```{r}
# CH_variants %>% 
#   distinct(IDs, .keep_all = TRUE) %>% 
#   select(IDs, total_allele_balance, SYMBOL) %>%
#   group_by(SYMBOL) %>%
#   summarise(sum_individuals_with_mutations = sum(total_allele_balance)) %>%
#   ggplot(aes(x= fct_reorder(SYMBOL, desc(sum_individuals_with_mutations)), 
#              y= sum_individuals_with_mutations))+
# 
#   geom_bar(stat = "identity")+
#   ggtitle("Prevalence of mutations per gene in chromosome 2")+ 
#   labs(x= "genes", y="Number of individuals")+
#   coord_flip()
```

<!-- Those genes are in our selection - DNMT3A would be in page 10 so about 100 genes have higher prevalence -->
```{r}
# CH_variants %>% 
#   distinct(IDs, .keep_all = TRUE) %>% 
#   select(IDs, total_allele_balance, SYMBOL, mutations_in_BickWHO) %>%
#   group_by(SYMBOL, mutations_in_BickWHO) %>%
#   summarise(sum_individuals_with_mutations = sum(total_allele_balance),
#             number_of_variants = n()) %>%
#   arrange(desc(mutations_in_BickWHO), desc(sum_individuals_with_mutations))

# CH_variants %>% 
#   distinct(IDs, .keep_all = TRUE) %>% 
#   select(IDs, total_allele_balance, SYMBOL, mutations_in_BickWHO) %>%
#   group_by(SYMBOL, mutations_in_BickWHO) %>%
#   summarise(sum_individuals_with_mutations = sum(total_allele_balance),
#             number_of_variants = n()) %>%
#   arrange(desc(mutations_in_BickWHO), desc(sum_individuals_with_mutations)) %>%
#   filter(SYMBOL == "SF3B1" | SYMBOL == "DNMT3A")
```

<!-- check nbr of hom -->
<!-- ```{r} -->
<!-- CH_variants %>%  -->
<!--   distinct(IDs, .keep_all = TRUE) %>%  -->
<!--   mutate(ab_hist_alt_bin_freq = str_match(INFO, "ab_hist_alt_bin_freq=(.*?);")[,2]) %>%  -->
<!--   mutate(nbr_hom_control_subsetpop = str_match(INFO, "controls_nhomalt=(.*?);")[,2]) %>%  -->
<!--   mutate(nbr_hom_individuals_in_samples = str_match(INFO, ";nhomalt=(.*?);")[,2]) %>%  -->
<!--   select(IDs, total_allele_balance, SYMBOL, ab_hist_alt_bin_freq,  -->
<!--          nbr_hom_individuals, nbr_hom_control_subsetpop, nbr_hom_individuals_in_samples) %>% -->
<!--   filter(SYMBOL == "TTN") -->
<!-- ``` -->

<!-- Look at exon length -->
<!-- ```{r} -->
<!-- CH_variants %>%  -->
<!--   distinct(IDs, .keep_all = TRUE) %>%  -->
<!--   select(IDs, total_allele_balance, SYMBOL, mutations_in_BickWHO, exon_effective_length) %>% -->
<!--   group_by(SYMBOL, mutations_in_BickWHO, exon_effective_length) %>% -->
<!--   summarise(sum_individuals_with_mutations = sum(total_allele_balance), -->
<!--             number_of_variants = n()) %>% -->
<!--   arrange(desc(mutations_in_BickWHO), desc(sum_individuals_with_mutations)) -->
<!-- ``` -->
<!-- AN? -->
<!-- ```{r} -->
<!-- CH_variants %>%  -->
<!--   distinct(IDs, .keep_all = TRUE) %>%  -->
<!--   select(IDs, total_allele_balance, SYMBOL, mutations_in_BickWHO, INFO) %>% -->
<!--   arrange(desc(mutations_in_BickWHO), SYMBOL, desc(total_allele_balance)) %>%  -->
<!--   mutate(AN = str_match(INFO, "AN=(.*?);")[,2]) %>%  -->
<!--   select(-INFO) %>%  -->
<!--   filter(SYMBOL %in% c("DNMT3A", "SF3B1")) -->
<!-- CH_variants %>%  -->
<!--   distinct(IDs, .keep_all = TRUE) %>%  -->
<!--   select(IDs, total_allele_balance, SYMBOL, mutations_in_BickWHO, INFO) %>% -->
<!--   arrange(desc(mutations_in_BickWHO), SYMBOL, desc(total_allele_balance)) %>%  -->
<!--   mutate(AN = str_match(INFO, "AN=(.*?);")[,2]) %>%  -->
<!--   select(-INFO) %>%  -->
<!--   filter(SYMBOL %in% c("GALM")) -->
<!-- ``` -->

## **Figure 2A** Compare top 10-20 genes for CH vs CM
```{r Compare top 10-20 genes for CH vs CM}
CH_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, SYMBOL) %>% 
  mutate(data = "CH variants") %>% 
  bind_rows(., clonal_mosaicism %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs, SYMBOL) %>% 
  mutate(data = "CM variants")) %>% 
  select(SYMBOL, data) %>% 
  tbl_summary(by = data,
              sort = list(everything() ~ "frequency")) %>% 
  bold_labels() %>% 
  modify_header(
    label = "**Compare top 10-20 genes for CH vs CM**"
  )
```



## **Figure 2B**  VEP
Updated with Yi-Han's file - Yi-Han will use maftools
```{r vep from Yi-Han}
CH_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(Func.refGene, ExonicFunc.refGene, 
         new_HGVSc, SIFT_pred, Polyphen2_HDIV_pred, 
         FATHMM_pred, CADD_phred, CADD_phred_cat, CLNSIG,
         REF_dbGaP_PopFreq, ALT_dbGaP_PopFreq) %>% 
  tbl_summary(sort = list(everything() ~ "frequency")) %>% 
  bold_labels() %>% 
  modify_header(
    label = "**VEPs extracted from Yi-Han**"
  )
```

```{r extract vep}
vep_in_gnomAD <- CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  mutate(ens_vep = str_match(INFO, ";vep=(.*?)$")[,2])
vep_in_gnomAD <- vep_in_gnomAD %>%
  separate(col = ens_vep, paste("ens_vep", 1:25, sep=""),
           sep = ",", remove = T, extra = "warn", fill = "right") %>%
  keep(~!all(is.na(.))) %>%
  pivot_longer(cols = starts_with("ens_vep"), names_to = NULL, values_to = "ens_vep") %>%
  drop_na(ens_vep)

ens_vep_var_names <- c("Allele", "Consequence", "IMPACT", "SYMBOL", "Gene", "Feature_type",
                       "Feature", "BIOTYPE", "EXON", "INTRON", "HGVSc", "HGVSp",
                       "cDNA_position", "CDS_position", "Protein_position", "Amino_acids",
                       "Codons", "Existing_variation", "ALLELE_NUM", "DISTANCE", "STRAND",
                       "FLAGS", "VARIANT_CLASS", "MINIMISED", "SYMBOL_SOURCE", "HGNC_ID",
                       "CANONICAL", "TSL", "APPRIS", "CCDS", "ENSP", "SWISSPROT", "TREMBL",
                       "UNIPARC", "GENE_PHENO", "SIFT", "PolyPhen", "DOMAINS", "HGVS_OFFSET",
                       "GMAF", "AFR_MAF", "AMR_MAF", "EAS_MAF", "EUR_MAF", "SAS_MAF", "AA_MAF",
                       "EA_MAF", "ExAC_MAF", "ExAC_Adj_MAF", "ExAC_AFR_MAF", "ExAC_AMR_MAF",
                       "ExAC_EAS_MAF", "ExAC_FIN_MAF", "ExAC_NFE_MAF", "ExAC_OTH_MAF",
                       "ExAC_SAS_MAF", "CLIN_SIG", "SOMATIC", "PHENO", "PUBMED", "MOTIF_NAME",
                       "MOTIF_POS", "HIGH_INF_POS", "MOTIF_SCORE_CHANGE", "LoF", "LoF_filter",
                       "LoF_flags", "LoF_info")

vep_in_gnomAD <- vep_in_gnomAD %>%
  separate(col = ens_vep, into = ens_vep_var_names,
           sep = "\\|", remove = F, extra = "warn", fill = "right")

vep_in_gnomAD <- vep_in_gnomAD %>%
  mutate(SOMATIC_1 = case_when(
    str_detect(SOMATIC, "1")      ~ 1,
    TRUE                          ~ 0
  )) %>% 
  group_by(IDs) %>% 
  mutate(SOMATIC = max(SOMATIC_1),
         SOMATIC = case_when(
    SOMATIC == 1                  ~ "Yes",
    TRUE                          ~ "No"
  )) %>% 
  ungroup() %>% 
  distinct(IDs, .keep_all = TRUE)
```

```{r}
vep_in_gnomAD %>% 
  select(
         SOMATIC) %>% 
  tbl_summary(type = list(c(SOMATIC) ~ "categorical"),
              sort = list(everything() ~ "frequency")) %>% 
  bold_labels() %>% 
  modify_header(
    label = "**VEP in gmomAD**"
  ) # %>%
  # modify_spanning_header(all_stat_cols() ~ "**Variants**")
```
<br>
<br>

***
<br>

Other data
```{r}
CH_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  mutate(popmax = str_match(INFO, ";popmax=(.*?);")[,2]) %>% 
  select(dp_het_median, allele_type, 
         variant_type, controls_popmax, popmax) %>% 
  tbl_summary(sort = list(everything() ~ "frequency")) %>% 
  bold_labels()
```
<br>
<br>

***
<br>

# **Figure 3**  AB / Age distribution
<span style="color: red;">CH variants noted to be considered by Nancy from other chromosomes will be added automatically in final figure.</span>  
<div class = "row">
<div class = "col-md-6">
#### AB distribution of known CH variant
#### A.
```{r}
extract_AB <- CH_variants %>% 
  filter(!is.na(considered_variants)) %>%
  # filter(IDs %in% c("2-25234373-C-T", "2-25235726-A-G",
  #                   "2-197402110-T-C")) %>%
  distinct(IDs, .keep_all = TRUE)

gnomad <- extract_AB %>%
  filter(IDs != "All individuals") %>% 
  select(-AB_count) %>% 
  # 1. Extract count per AB bin for heterozygous ind--
  # ab_hist_alt_bin_freq,Number=A,Type=String,Description="Histogram for AB in heterozygous individuals; 
  # bin edges are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00>
  mutate(ab_hist_alt_bin_freq = str_match(INFO, "ab_hist_alt_bin_freq=(.*?);")[,2]) %>%
  separate(col = ab_hist_alt_bin_freq,
           into = c("0.00-0.05","0.05-0.10","0.10-0.15","0.15-0.20",
                    "0.20-0.25","0.25-0.30","0.30-0.35","0.35-0.40",
                    "0.40-0.45","0.45-0.50","0.50-0.55","0.55-0.60",
                    "0.60-0.65","0.65-0.70","0.70-0.75","0.75-0.80",
                    "0.80-0.85","0.85-0.90","0.90-0.95","0.95-1.00"),
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>%
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>%
  add_row("IDs"="Reference", "ID"="ASXL1 20-32435697-C-T",
          "0.00-0.05" = 0, "0.05-0.10"  = 0, "0.10-0.15" = 0, 
          "0.15-0.20" = 1, "0.20-0.25" = 2, "0.25-0.30" = 6, 
          "0.30-0.35" = 87, "0.35-0.40" = 274, "0.40-0.45" = 1220,
          "0.45-0.50" = 2286, "0.50-0.55" = 1737, "0.55-0.60" = 506,
          "0.60-0.65" = 112, "0.65-0.70" = 14, "0.70-0.75" = 4,
          "0.75-0.80" = 0, "0.80-0.85" = 1, "0.85-0.90" = 0,
          "0.90-0.95" = 0, "0.95-1.00" = 0, .before = 1
          ) %>%
  mutate(total_allele_balance = rowSums(select(.,`0.00-0.05`:`0.95-1.00`), na.rm = TRUE)) %>%
  
  select(IDs, everything())

gnomad <- gnomad %>%
  pivot_longer(cols = c(grep("^[[:digit:]]", colnames(.))),
               names_to = "allele_bin", values_to = "AB_count") %>%
  mutate(allele_bin = factor(allele_bin,
                             levels = c("0.00-0.05","0.05-0.10","0.10-0.15","0.15-0.20",
                                        "0.20-0.25","0.25-0.30","0.30-0.35","0.35-0.40",
                                        "0.40-0.45","0.45-0.50","0.50-0.55","0.55-0.60",
                                        "0.60-0.65","0.65-0.70","0.70-0.75","0.75-0.80",
                                        "0.80-0.85","0.85-0.90","0.90-0.95","0.95-1.00"
                             ))) %>%
  mutate(AB_bin = case_when(
    allele_bin == "0.00-0.05"            ~ 0.025,
    allele_bin == "0.05-0.10"            ~ 0.075,
    allele_bin == "0.10-0.15"            ~ 0.125,
    allele_bin == "0.15-0.20"            ~ 0.175,
    allele_bin == "0.20-0.25"            ~ 0.225,
    allele_bin == "0.25-0.30"            ~ 0.275,
    allele_bin == "0.30-0.35"            ~ 0.325,
    allele_bin == "0.35-0.40"            ~ 0.375,
    allele_bin == "0.40-0.45"            ~ 0.425,
    allele_bin == "0.45-0.50"            ~ 0.475,
    allele_bin == "0.50-0.55"            ~ 0.525,
    allele_bin == "0.55-0.60"            ~ 0.575,
    allele_bin == "0.60-0.65"            ~ 0.625,
    allele_bin == "0.65-0.70"            ~ 0.675,
    allele_bin == "0.70-0.75"            ~ 0.725,
    allele_bin == "0.75-0.80"            ~ 0.775,
    allele_bin == "0.80-0.85"            ~ 0.825,
    allele_bin == "0.85-0.90"            ~ 0.875,
    allele_bin == "0.90-0.95"            ~ 0.925,
    allele_bin == "0.95-1.00"            ~ 0.975
  )) %>%
  mutate(AB_frequency = AB_count / total_allele_balance)

All <- gnomad %>%
    filter(IDs == "Reference"
    ) %>% 
    mutate(IDs = "Ref")
All %>%
    ggplot(aes(x = AB_bin, y= AB_frequency))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6, color= "#3333FF"
    )+
    labs(title = All$ID)+
    geom_col(data = All ,
             aes(x= AB_bin, y= AB_frequency),
             position = "identity",
             fill = "#6699FF",
             color= "#3333FF",
             alpha = 0.1)+
    ylim(0, max(All$AB_frequency))

# n <- 0
set.seed(1234)
for(i in unique(gnomad$IDs)) {
  
  if (i != "Reference") {
  # All <- gnomad %>%
  #   filter(IDs == "Reference"
  #   ) %>% 
  #   mutate(IDs = "Ref")
  data_plot <- gnomad %>%
    filter(IDs == i
    ) %>%
    bind_rows(., All) %>%
    distinct(IDs, AB_bin, .keep_all = TRUE)
  
  plot_title <- data_plot %>% 
    filter(IDs == i) %>% 
    select(considered_variants) %>% distinct() %>% as.character()
  # n <- n + 1
  p <- data_plot %>%
    ggplot(aes(x = AB_bin, y= AB_frequency, color= IDs))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
    )+
    scale_color_manual(values = c("#F8766D", "#3333FF"))+
    labs(title = plot_title)+
    geom_col(data = data_plot %>%
               filter(IDs == i) ,
             aes(x= AB_bin, y= AB_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$AB_frequency))+
    theme(legend.position = "none")
  
  p <- p+
    geom_text(
      data = data_plot %>% 
        filter(IDs == i),
      mapping = aes(x = AB_distribution_mean + 0.2, y = -Inf, 
                    label = paste0("p-AB adjusted= ", round(adjusted_AB_pval, 3))),
      colour = "black",
      # hjust   = -0.5,
      vjust   = -28
    )
  
  print(p)
  }
}
```
</div>

<div class = "col-md-6">
#### Age distribution of known CH variant
#### B.
```{r}
extract_Age <- CH_variants %>% 
  filter(!is.na(considered_variants)) %>%
  # filter(IDs %in% c("2-25234373-C-T", "2-25235726-A-G",
  #                   "2-197402110-T-C")) %>%
  distinct(IDs, .keep_all = TRUE)

extract_Age <- extract_Age %>% 

  select(-sample_frequency, -sample_count, -nbr_individuals) %>% 

  # "Count of age values falling below lowest histogram bin edge for heterozygous individuals">
  mutate("<30" = as.numeric(str_match(INFO, "age_hist_het_n_smaller=(.*?);")[,2])) %>%
  mutate(age_hist_het_bin_freq = str_match(INFO, "age_hist_het_bin_freq=(.*?);")[,2]) %>%
  separate(col = age_hist_het_bin_freq,
           into = c("30-35","35-40","40-45","45-50",
                    "50-55","55-60","60-65","65-70",
                    "70-75","75-80"), 
           sep = "\\|", remove = TRUE, extra = "warn", fill = "right") %>% 
  # "Count of age values falling above highest histogram bin edge for heterozygous individuals">
  mutate(">80" = as.numeric(str_match(INFO, "age_hist_het_n_larger=(.*?);")[,2])) %>% 
  mutate(across(grep("^[[:digit:]]", colnames(.)), ~ as.numeric(.))) %>% 

  add_row("IDs"="All individuals", "ID"="All individuals of any genotype bin",
          "<30" = 2547, "30-35" = 3423,
          "35-40" = 4546,"40-45" = 8487,
          "45-50" = 10355,"50-55" = 12693,
          "55-60" = 11933,"60-65" = 10534,
          "65-70" = 8882,"70-75" = 5991,
          "75-80" = 4136,
          ">80" = 1935, .before = 1
          ) %>%
  mutate(nbr_individuals = rowSums(select(.,`<30`:`>80`), na.rm = TRUE))

extract_Age <- extract_Age %>% 
  pivot_longer(cols = c(grep("^([[:digit:]]|<|>)", colnames(.))), 
               names_to = "age_bin", values_to = "sample_count") %>% 
  mutate(age_bin = factor(age_bin, 
                          levels = c("<30", "30-35","35-40","40-45","45-50","50-55",
                                     "55-60","60-65","65-70","70-75","75-80", ">80"
                                     ))) %>% 
  mutate(bin_age = case_when(
    # The average age at diagnosis is 8 overall (ages 0 to 19), 
    # 5 years old for children (aged 0 to 14), and 17 years old 
    # for adolescents (aged 15 to 19), while adults' average 
    # age for cancer diagnosis is 65
    str_detect(age_bin, "<30")              ~ 15, 
    str_detect(age_bin, "30-35")            ~ 32.5,
    str_detect(age_bin, "35-40")            ~ 37.5,
    str_detect(age_bin, "40-45")            ~ 42.5,
    str_detect(age_bin, "45-50")            ~ 47.5,
    str_detect(age_bin, "50-55")            ~ 52.5,
    str_detect(age_bin, "55-60")            ~ 57.5,
    str_detect(age_bin, "60-65")            ~ 62.5,
    str_detect(age_bin, "65-70")            ~ 67.5,
    str_detect(age_bin, "70-75")            ~ 72.5,
    str_detect(age_bin, "75-80")            ~ 77.5,
    TRUE ~ 90
  )) %>% 
  mutate(age_frequency = sample_count / nbr_individuals)

All <- extract_Age %>%
    filter(IDs == "All individuals"
    )
All %>%
    ggplot(aes(x = bin_age, y= age_frequency))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6, color= "#3333FF"
    )+
    labs(title = All$ID)+
    geom_col(data = All ,
             aes(x= bin_age, y= age_frequency),
             position = "identity",
             fill = "#6699FF",
             color= "#3333FF",
             alpha = 0.1)+
    ylim(0, max(All$age_frequency))

# n <- 0

set.seed(1234)

for(i in unique(extract_Age$IDs)) {
  
  if (i != "All individuals") {
  # All <- extract_Age %>% 
  #   filter(IDs == "All individuals"
  # ) %>% 
  #   mutate(IDs = paste0("Reference= ", ID))
  data_plot <- extract_Age %>% 
    filter(IDs == i
    ) %>% 
    bind_rows(., All) %>%
    distinct(IDs, bin_age, .keep_all = TRUE)
  
  plot_title <- data_plot %>% 
    filter(IDs == i) %>% 
    select(considered_variants) %>% distinct() %>% as.character()
  # n <- n + 1
  p <- data_plot %>%
    ggplot(aes(x = bin_age, y= age_frequency, color= IDs))+
    geom_smooth(alpha= 0.5, se = FALSE, method = "loess",
                span = 0.6
    )+
    scale_color_manual(values = c("#F8766D", "#3333FF"))+
    labs(title = plot_title)+
    geom_col(data = data_plot %>%
               filter(IDs == i) ,
             aes(x= bin_age, y= age_frequency),
             position = "identity",
             fill = "red",
             alpha = 0.1)+
    ylim(0, max(data_plot$age_frequency))+
    theme(legend.position = "none")
  
  p <- p+
    geom_text(
      data    = data_plot %>% 
               filter(IDs == i),
      mapping = aes(x = age_distribution_mean - 15, y = -Inf, 
                    label = paste0("BH= ", round(adjusted_age_pval, 3))),
      colour = "black",
      # hjust   = 0,
      vjust   = -28
    )
  
  print(p)
  }

}
```
</div>
</div>

# **Figure3C** Number of variants per Genes
```{r}
CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  select(SYMBOL, X.CHROM, POS) %>% 
  group_by(X.CHROM, SYMBOL) %>% 
  mutate(position = round(POS, -5)) %>% 
  
  group_by(SYMBOL) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  
  distinct(SYMBOL, .keep_all = TRUE) %>% 
  ggplot(aes(x= position, y= n))+
  geom_segment(aes(x = position, xend = position, y = 0, yend = n),
               color = "gray", lwd = 1) +
  geom_point(size = 7.5, pch = 21, bg = 4, aes(color = SYMBOL)) +
  geom_text(aes(label = n), color = "white", size = 3)+ 
  geom_text(aes(x= position, y= n+1, label = SYMBOL), color = "black", size = 3, )+ 
  guides(color = FALSE)
```





<br>
<br>

***
<br>

# **Figure4** Disparities in CH variants (VAF **non-cancer pop**)
nc_freq_allele_count = str_match(INFO, "non_cancer_AF=(.*?);")[,2])  
non_cancer_AF, Alternate allele frequency in samples in the non_cancer subset  
```{r}
# CH_variants %>%
#   distinct(IDs, .keep_all = TRUE) %>%
#   filter(!is.na(considered_variants)) %>%
#   # filter(IDs %in% c("2-25234373-C-T", "2-25235726-A-G",
#   #                   "2-197402110-T-C")) %>%
#   distinct(IDs, .keep_all = TRUE) %>%
#   mutate(popmax = str_match(INFO, ";popmax=(.*?);")[,2]) %>%
#   mutate(AN_sas = str_match(INFO, "AN_sas=(.*?);")[,2]) %>%
#   mutate(AN_eas = str_match(INFO, "AN_eas=(.*?);")[,2]) %>%
#   select(IDs, popmax, controls_popmax,
#          AN_sas, AN_eas)
```

```{r figure 4}
CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  unite(IDs, c(IDs, SYMBOL)) %>% 
  select(IDs,
         # nc_freq_allele_count, nc_freq_allele_count_female, nc_freq_allele_count_male,
         nc_freq_allele_count_afr, nc_freq_allele_count_afr_female, nc_freq_allele_count_afr_male,
         nc_freq_allele_count_amr, nc_freq_allele_count_amr_female, nc_freq_allele_count_amr_male,
         nc_freq_allele_count_eas, nc_freq_allele_count_eas_female, nc_freq_allele_count_eas_male,
         # nc_freq_allele_count_sas, nc_freq_allele_count_sas_female, nc_freq_allele_count_sas_male,
         nc_freq_allele_count_nfe, nc_freq_allele_count_nfe_female, nc_freq_allele_count_nfe_male 
         ) %>%
  pivot_longer(cols = -IDs) %>%
  mutate(value = as.numeric(value)) %>%
  mutate(name = str_remove(name, "nc_freq_allele_count_"),
         # name = str_replace(name, "nc_freq_allele_count", "overall"),
         name = factor(name, levels = c(
    # "male", "female", "overall",
    "nfe_male", "nfe_female", "nfe",
    "eas_male", "eas_female", "eas", 
    # "sas_male", "sas_female", "sas", 
    "amr_male", "amr_female", "amr", 
    "afr_male", "afr_female", "afr")
    )) %>% 
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>%
  mutate(Ethnicity = str_extract(name, "afr|amr|nfe_nwe|nfe|eas"), Ethnicity = case_when(
    is.na(Ethnicity)       ~ "overall",
    TRUE                   ~ Ethnicity
  )) %>%
  select(name, value,
         Ethnicity, Sex) %>% 
  # mutate(mutations_in_BickWHO = factor(mutations_in_BickWHO, levels= c("Yes", "No"))) %>% 
  
  ggplot(aes(x=value, y=name, fill=Ethnicity, linetype= Sex)) +
  geom_density_ridges(alpha=0.5)+
  ggtitle("Popultaion Allele Frequency Density Distribution")+
  labs(x= "Popultaion Allele Frequency", y= NULL)

CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  unite(IDs, c(IDs, SYMBOL)) %>% 
  select(IDs,
         # nc_freq_allele_count, nc_freq_allele_count_female, nc_freq_allele_count_male,
         nc_freq_allele_count_afr, nc_freq_allele_count_afr_female, nc_freq_allele_count_afr_male,
         nc_freq_allele_count_amr, nc_freq_allele_count_amr_female, nc_freq_allele_count_amr_male,
         nc_freq_allele_count_eas, nc_freq_allele_count_eas_female, nc_freq_allele_count_eas_male,
         # nc_freq_allele_count_sas, nc_freq_allele_count_sas_female, nc_freq_allele_count_sas_male,
         nc_freq_allele_count_nfe, nc_freq_allele_count_nfe_female, nc_freq_allele_count_nfe_male 
         ) %>%
  pivot_longer(cols = -IDs) %>%
  mutate(value = as.numeric(value)) %>%
  mutate(name = str_remove(name, "nc_freq_allele_count_"),
         # name = str_replace(name, "nc_freq_allele_count", "overall"),
         name = factor(name, levels = c(
    # "male", "female", "overall",
    "nfe_male", "nfe_female", "nfe",
    "eas_male", "eas_female", "eas", 
    # "sas_male", "sas_female", "sas", 
    "amr_male", "amr_female", "amr", 
    "afr_male", "afr_female", "afr")
    )) %>% 
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>%
  mutate(Ethnicity = str_extract(name, "afr|amr|nfe_nwe|nfe|eas"), Ethnicity = case_when(
    is.na(Ethnicity)       ~ "overall",
    TRUE                   ~ Ethnicity
  )) %>%
  select(name, value,
         Ethnicity, Sex) %>% 
  # mutate(mutations_in_BickWHO = factor(mutations_in_BickWHO, levels= c("Yes", "No"))) %>% 
  
  ggplot(aes(x=value, y=name, fill=Ethnicity, linetype= Sex)) +
  geom_density_ridges(alpha=0.2, rel_min_height = 0.01, scale = 0.8)+
  ggtitle("Popultaion Allele Frequency Density Distribution - xlim(0,0.001)")+
  labs(x= "Popultaion Allele Frequency", y= NULL)+
  xlim(0,0.001)

CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  unite(IDs, c(IDs, SYMBOL)) %>% 
  select(IDs,
         # nc_freq_allele_count, nc_freq_allele_count_female, nc_freq_allele_count_male,
         nc_freq_allele_count_afr, nc_freq_allele_count_afr_female, nc_freq_allele_count_afr_male,
         nc_freq_allele_count_amr, nc_freq_allele_count_amr_female, nc_freq_allele_count_amr_male,
         nc_freq_allele_count_eas, nc_freq_allele_count_eas_female, nc_freq_allele_count_eas_male,
         # nc_freq_allele_count_sas, nc_freq_allele_count_sas_female, nc_freq_allele_count_sas_male,
         nc_freq_allele_count_nfe, nc_freq_allele_count_nfe_female, nc_freq_allele_count_nfe_male 
         ) %>%
  pivot_longer(cols = -IDs) %>%
  mutate(value = as.numeric(value)) %>%
  mutate(name = str_remove(name, "nc_freq_allele_count_"),
         # name = str_replace(name, "nc_freq_allele_count", "overall"),
         name = factor(name, levels = c(
    # "male", "female", "overall",
    "nfe_male", "nfe_female", "nfe",
    "eas_male", "eas_female", "eas", 
    # "sas_male", "sas_female", "sas", 
    "amr_male", "amr_female", "amr", 
    "afr_male", "afr_female", "afr")
    )) %>% 
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>%
  mutate(Ethnicity = str_extract(name, "afr|amr|nfe_nwe|nfe|eas"), Ethnicity = case_when(
    is.na(Ethnicity)       ~ "overall",
    TRUE                   ~ Ethnicity
  )) %>%
  select(name, value,
         Ethnicity, Sex) %>% 
  # mutate(mutations_in_BickWHO = factor(mutations_in_BickWHO, levels= c("Yes", "No"))) %>% 
  
  ggplot(aes(x=value, y=name, fill=Ethnicity, linetype= Sex)) +
  geom_density_ridges(alpha=0.5, stat="binline", bins = 100, scale = 0.95, draw_baseline = FALSE)+
  ggtitle("Popultaion Allele Frequency Distribution - xlim(-0.00001,0.001)")+
  labs(x= "Popultaion Allele Frequency", y= NULL)+
  xlim(-0.00001,0.001)
```

```{r table AF disparities values}
CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs,
         # nc_freq_allele_count, nc_freq_allele_count_female, nc_freq_allele_count_male,
         nc_freq_allele_count_afr, nc_freq_allele_count_afr_female, nc_freq_allele_count_afr_male,
         nc_freq_allele_count_amr, nc_freq_allele_count_amr_female, nc_freq_allele_count_amr_male,
         nc_freq_allele_count_eas, nc_freq_allele_count_eas_female, nc_freq_allele_count_eas_male,
         # nc_freq_allele_count_sas, nc_freq_allele_count_sas_female, nc_freq_allele_count_sas_male,
         nc_freq_allele_count_nfe, nc_freq_allele_count_nfe_female, nc_freq_allele_count_nfe_male 
         ) %>%
  pivot_longer(cols = -IDs) %>%
  mutate(value = as.numeric(value)) %>%
  mutate(name = str_remove(name, "nc_freq_allele_count_"),
         # name = str_replace(name, "nc_freq_allele_count", "overall"),
         name = factor(name, levels = c(
    # "male", "female", "overall", 
    "nfe_male", "nfe_female", "nfe",
    "eas_male", "eas_female", "eas", 
    # "sas_male", "sas_female", "sas", 
    "amr_male", "amr_female", "amr", 
    "afr_male", "afr_female", "afr")
    )) %>% 
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>%
  mutate(Ethnicity = str_extract(name, "afr|amr|nfe_nwe|nfe|eas"), Ethnicity = case_when(
    is.na(Ethnicity)       ~ "overall",
    TRUE                   ~ Ethnicity
  )) %>% 
  filter(Ethnicity != "overall") %>% 
  filter(Sex == "overall") %>% 
  select(`Population Allele Frequency` = value, Ethnicity) %>% 
  tbl_summary(by= Ethnicity,
              digits = all_continuous() ~ function(x) format(x, digits = 3, scientific = TRUE)
              ) %>% 
  bold_labels() %>% 
  add_p() %>% bold_p(t=.05) %>% add_overall() %>% add_q()

CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs,
         nc_freq_allele_count_female, nc_freq_allele_count_male
         ) %>%
  pivot_longer(cols = -IDs) %>%
  mutate(value = as.numeric(value)) %>%
  mutate(name = str_remove(name, "nc_freq_allele_count_"),
         name = str_replace(name, "nc_freq_allele_count", "overall"),
         name = factor(name, levels = c(
    "male", "female")
    )) %>% 
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>%
  select(`Population Allele Frequency` = value, Sex) %>% 
  tbl_summary(by= Sex,
              digits = all_continuous() ~ function(x) format(x, digits = 3, scientific = TRUE)
              ) %>% 
  bold_labels() %>% 
  add_p() %>% bold_p(t=.05) %>% add_overall()

CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  select(IDs,
         nc_freq_allele_count_nfe_female, nc_freq_allele_count_nfe_male
         ) %>%
  pivot_longer(cols = -IDs) %>%
  mutate(value = as.numeric(value)) %>%
  mutate(name = str_remove(name, "nc_freq_allele_count_"),
         name = str_replace(name, "nc_freq_allele_count", "overall")#,
    #      name = factor(name, levels = c(
    # "male", "female")
    # )
   ) %>% 
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>%
  select(`Population Allele Frequency` = value, Sex) %>% 
  tbl_summary(by= Sex,
              digits = all_continuous() ~ function(x) format(x, digits = 3, scientific = TRUE)
              ) %>% 
  bold_labels() %>% 
  add_p() %>% bold_p(t=.05) %>% add_overall() %>% 
  modify_header(
    label = "**Caucasian AF female vs male**"
  )
```
<br>
<br>

## Potential supplement figure for few variants...
```{r figure S4, fig.width=12, fig.height=7}
CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  filter(!is.na(considered_variants)) %>%
  # filter(IDs %in% c("2-25234373-C-T", "2-25235726-A-G",
  #                   "2-197402110-T-C")) %>%
  unite(IDs, c(IDs, SYMBOL)) %>% 
  select(IDs, considered_variants,
         # nc_freq_allele_count, nc_freq_allele_count_female, nc_freq_allele_count_male,
         nc_freq_allele_count_afr, nc_freq_allele_count_afr_female, nc_freq_allele_count_afr_male,
         nc_freq_allele_count_amr, nc_freq_allele_count_amr_female, nc_freq_allele_count_amr_male,
         nc_freq_allele_count_eas, nc_freq_allele_count_eas_female, nc_freq_allele_count_eas_male,
         # nc_freq_allele_count_sas, nc_freq_allele_count_sas_female, nc_freq_allele_count_sas_male,
         nc_freq_allele_count_nfe, nc_freq_allele_count_nfe_female, nc_freq_allele_count_nfe_male 
         ) %>%
  pivot_longer(cols = -c(IDs, considered_variants)) %>%
  mutate(value = as.numeric(value)) %>%
  mutate(name = str_remove(name, "nc_freq_allele_count_"),
         # name = str_replace(name, "nc_freq_allele_count", "overall"),
         name = factor(name, levels = c(
    # "male", "female", "overall", 
    "nfe_male", "nfe_female", "nfe",
    "eas_male", "eas_female", "eas", 
    # "sas_male", "sas_female", "sas", 
    "amr_male", "amr_female", "amr", 
    "afr_male", "afr_female", "afr")
    )) %>% 
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>%
  mutate(Ethnicity = str_extract(name, "afr|amr|nfe_nwe|nfe|eas"), Ethnicity = case_when(
    is.na(Ethnicity)       ~ "overall",
    TRUE                   ~ Ethnicity
  )) %>% 
  
  ggplot(aes(x=name, y= value, shape= Sex, color= Ethnicity))+
  geom_point(size= 3)+
  labs(title = "Variant Allele Frequency in Non-Cancer population", 
       x=NULL, y=NULL)+
  theme_classic(base_size = 16)+
  # theme(legend.position = "none")+
  facet_wrap(. ~ considered_variants, scales = "free", ncol = 2)+
  coord_flip()
```
<br>
<br>

***
<br>

# **Figure5** CH variants in Cancer base study population (Cosmic)
`r emo::ji("pencil")` <span style="color: red;">1 variant can be present in multiple patients and 1 patient can have multiple variants.</span>
```{r}
CH_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>%
  select(variant_in_cosmic) %>%
  tbl_summary(type = list(variant_in_cosmic ~ "categorical"),
              sort = everything() ~ "frequency") %>% 
  bold_labels() %>% 
  modify_spanning_header(all_stat_cols() ~ "Number of unique variants")
```

```{r}
variants_in_cosmic <- CH_variants %>% 
  filter(variant_in_cosmic == "Yes")
```
In chromosome `r as.character(variants_in_cosmic %>% distinct(X.CHROM))`,  
- individuals with at least 1 CH variants in Cosmic is `r nrow(variants_in_cosmic %>% distinct(Sample.name))` (1 individual can have multiple variants);  
- the number of variants present in Cosmic is `r nrow(variants_in_cosmic %>% distinct(IDs))` (1 variant is found in multiple individuals).


# **Figure** Genes?
```{r}
variants_in_cosmic %>%
  distinct(Sample.name, .keep_all = TRUE) %>% 
  select(SYMBOL, X.CHROM, POS) %>% 
  group_by(X.CHROM, SYMBOL) %>% 
  mutate(position = round(POS, -5)) %>% 
  
  group_by(SYMBOL) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  
  distinct(SYMBOL, .keep_all = TRUE) %>% 
  arrange(desc(n))
variants_in_cosmic %>%
  distinct(Sample.name, .keep_all = TRUE) %>% 
  select(SYMBOL, X.CHROM, POS) %>% 
  group_by(X.CHROM, SYMBOL) %>% 
  mutate(position = round(POS, -5)) %>% 
  
  group_by(SYMBOL) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  
  distinct(SYMBOL, .keep_all = TRUE) %>% 
  ggplot(aes(x= position, y= n))+
  ggtitle("Nbr of Cosmic individuals per gene")+
  geom_segment(aes(x = position, xend = position, y = 0, yend = n),
               color = "gray", lwd = 1) +
  geom_point(size = 7.5, pch = 21, bg = 4, aes(color = SYMBOL)) +
  geom_text(aes(label = n), color = "white", size = 3)+ 
  guides(color = FALSE)
```

# **Figure 5A** Age in COSMIC
```{r}
a <- CH_variants %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  select(Age, Sample.name) %>%
  mutate(data = "selected variants")

gnomad_variants %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  filter(!Sample.name %in% c(a$Sample.name)) %>% 
  select(Age) %>%
  mutate(data = "not selected") %>% 
  bind_rows(., a
  ) %>% 
  group_by(data) %>% 
  mutate(median_ = median(Age, na.rm = TRUE)) %>% 
  ggplot(aes(x=Age, fill= data))+
  geom_histogram(binwidth = 1)+
  scale_fill_manual(values = c("lightgrey", "purple"), name="")+
  geom_vline(aes(xintercept= median_, color= data))+
  scale_color_manual(values = c("darkgrey", "purple"))+ 
  guides(color = FALSE)

gnomad_variants %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  filter(!Sample.name %in% c(a$Sample.name)) %>% 
  select(Age) %>%
  mutate(data = "not selected") %>% 
  bind_rows(., a
  ) %>% 
  select(Age, data) %>% 
  tbl_summary(by=data) %>% 
  bold_labels() %>% add_overall() %>% 
  add_p() %>% bold_p(t=.05) %>% 
  modify_header(
    label = "**Cosmic Patients Characteristics**"
  )

```
# **Figure 5A2** Age in COSMIC
```{r}
CH_variants %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  select(Age) %>%
  mutate(data = "CH variants") %>% 
  bind_rows(., clonal_mosaicism %>% 
              distinct(Sample.name, .keep_all = TRUE) %>%
              select(Age) %>%
              mutate(data = "CM variants")) %>% 
  select(Age, data) %>% 
  tbl_summary(by=data) %>% 
  bold_labels() %>% add_overall() %>% 
  add_p() %>% bold_p(t=.05) %>% 
  modify_header(
    label = "**Age in Cosmic CH vs CM**"
  )
```


# **Figure 5B** Cancers in COSMIC
`r emo::ji("pencil")` Update primary site plot.  
- I changed the axis proportion (between before and after the break) as it took me just a minute.  
- We want to make the gap of the break bigger and tail longer (tail is already longer since I changed the axis proportion - maybe wait and see after increasing the gap).  
- Try inverting haematopoietic_neoplasm and lymphoid_neoplasm to have haematopoietic_neoplasm at the bottom (just need to change the Primary.histology factor levels)

```{r, fig.width=12}
variants_in_cosmic %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  select(Primary.site, Primary.histology) %>% 
  mutate(Primary.histology = case_when(
    Primary.site == "haematopoietic_and_lymphoid_tissue"    ~ Primary.histology,
    TRUE    ~ NA_character_
  )) %>% 
  group_by(Primary.site, Primary.histology) %>%
  summarise(sum_individuals_by_primary = n()) %>%
  ggplot(aes(x= fct_reorder(Primary.site, sum_individuals_by_primary), 
             y=sum_individuals_by_primary, fill= Primary.histology))+
  geom_bar(stat = "identity")+ 
  scale_fill_manual(breaks = c("haematopoietic_neoplasm", "lymphoid_neoplasm"), 
                    values = c("purple", "green")) +
  labs(x= "Primary Site")+
  scale_y_break(c(510, 1790), 
                ticklabels=c(1800, 2000),
                expand = FALSE,
                scales="fixed") +
  coord_flip()


variants_in_cosmic %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  select(Primary.site) %>% 
  tbl_summary(
              sort = everything() ~ "frequency") %>% 
  bold_labels() %>% 
  modify_header(
    label = "**Cosmic Patients/Cell Line Characteristics**"
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Samples, careful if 1 sample shows multiple gene-mutations**")

variants_in_cosmic %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  filter(Primary.site == "haematopoietic_and_lymphoid_tissue") %>% 
  select(Primary.histology) %>% 
  tbl_summary(
              sort = everything() ~ "frequency") %>% 
  bold_labels() %>% 
  modify_header(
    label = "**Cosmic Patients/Cell Line Characteristics <br>for Primary.site == 'haematopoietic_and_lymphoid_tissue'**"
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Samples, careful if 1 sample shows multiple gene-mutations**")

variants_in_cosmic %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  select(Mutation.Description,
         FATHMM.prediction, Mutation.somatic.status) %>% 
  mutate(Mutation.somatic.status = case_when(
    str_detect(Mutation.somatic.status, "somatic")        ~ "reported or confirmed somatic",
    TRUE                                                  ~ Mutation.somatic.status
  )) %>% 
  tbl_summary(
              sort = everything() ~ "frequency") %>% 
  bold_labels() %>% 
  modify_header(
    label = "**Cosmic Patients/Cell Line Characteristics**"
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**Samples, careful if 1 sample shows multiple gene-mutations**")
```

```{r}
# variants_in_cosmic %>% 
#   distinct(IDs, .keep_all = TRUE) %>%
#   select(Mutation.somatic.status) %>% 
#   mutate(Mutation.somatic.status = case_when(
#     str_detect(Mutation.somatic.status, "somatic")        ~ "reported or confirmed somatic",
#     TRUE                                                  ~ Mutation.somatic.status
#   )) %>% 
#   ggplot(aes(x=Mutation.somatic.status))+
#   geom_bar()+
#   ggtitle("Number of unique selected variants reported or confirmed somatic")+
#   coord_flip()
```

# **Figure 5C** Table cosmic by cancers: Focus on the top5 cancers
```{r}
variants_in_cosmic %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  filter(Primary.site %in% 
           c("haematopoietic_and_lymphoid_tissue", 
             "large_intestine", "breast", "skin", "lung", "endometrium")) %>% 
  select(Primary.site, Age, Mutation.somatic.status, SYMBOL) %>% 
  mutate(Mutation.somatic.status = case_when(
    str_detect(Mutation.somatic.status, "somatic")        ~ "reported or confirmed somatic",
    TRUE                                                  ~ Mutation.somatic.status
  )) %>% 
  tbl_summary(by= Primary.site,
              sort = everything() ~ "frequency") %>% 
  bold_labels() %>% 
  add_p()
```

# **Figure** Focus on HEM
```{r}
a <- CH_variants %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  filter(Primary.site == "haematopoietic_and_lymphoid_tissue") %>% 
  select(Age, Sample.name) %>%
  mutate(data = "selected variants")

gnomad_variants %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  filter(Primary.site == "haematopoietic_and_lymphoid_tissue") %>% 
  filter(!Sample.name %in% c(a$Sample.name)) %>% 
  select(Age) %>%
  mutate(data = paste("not selected in chromosome", clean_AB_variants %>% distinct(X.CHROM))
         ) %>% 
  bind_rows(., a
  ) %>% 
  group_by(data) %>% 
  mutate(median_ = median(Age, na.rm = TRUE)) %>% 
  ggplot(aes(x=Age, fill= data))+
  geom_histogram(binwidth = 1)+
  scale_fill_manual(values = c("lightgrey", "purple"), name="")+
  geom_vline(aes(xintercept= median_, color= data))+
  scale_color_manual(values = c("#999999", "#990066"))+ 
  guides(color = FALSE)

gnomad_variants %>% 
  distinct(Sample.name, .keep_all = TRUE) %>%
  filter(Primary.site == "haematopoietic_and_lymphoid_tissue") %>% 
  filter(!Sample.name %in% c(a$Sample.name)) %>% 
  select(Age) %>%
  mutate(data = paste("not selected in chromosome", clean_AB_variants %>% distinct(X.CHROM))
         ) %>% 
  bind_rows(., a
  ) %>% 
  select(Age, data) %>% 
  tbl_summary(by=data) %>% 
  bold_labels() %>% add_overall() %>% 
  add_p() %>% bold_p(t=.05) %>% 
  modify_header(
    label = "**Cosmic HEM Patients Characteristics**"
  )
```

<br>
<br>

***
<br>

# **Figure6** Bick+WHO - validation? Are we detecting all Bick+WHO variants?
## Bick+WHO in gnomAD

```{r, fig.height=2}
CH_variants %>% 
  distinct(IDs, .keep_all = TRUE) %>% 
  select(AB_distribution_mean, age_distribution_mean,
         mutations_in_BickWHO, variant_in_cosmic) %>% 
  mutate(mutations_in_BickWHO = factor(mutations_in_BickWHO, levels= c("Yes", "No"))) %>% 
  
  ggplot(aes(x=AB_distribution_mean, y=mutations_in_BickWHO, fill=mutations_in_BickWHO)) +
    geom_density_ridges(alpha=0.5)+
  ggtitle("Average Allele Balance Density Distribution")

# CH_variants %>% 
#   distinct(IDs, .keep_all = TRUE) %>% 
#   mutate(AB_distribution_mean = round(AB_distribution_mean, 2)) %>% 
#   select(AB_distribution_mean, age_distribution_mean,
#          mutations_in_BickWHO, variant_in_cosmic) %>% 
#   group_by(AB_distribution_mean, mutations_in_BickWHO) %>% 
#   summarize(sumu= n()) %>% 
#   ggplot(aes(x=AB_distribution_mean, y=sumu, fill=mutations_in_BickWHO))+
#   geom_area()

# CH_variants %>% 
#   distinct(IDs, .keep_all = TRUE) %>% 
#   select(AB_distribution_mean, age_distribution_mean,
#          mutations_in_BickWHO, variant_in_cosmic) %>% 
#   mutate(mutations_in_BickWHO = factor(mutations_in_BickWHO, levels= c("Yes", "No"))) %>% 
# 
#   ggplot(aes(x=AB_distribution_mean, y=mutations_in_BickWHO, fill=mutations_in_BickWHO)) +
#   geom_density_ridges(alpha=0.5) +
#   
#   facet_wrap(~ variant_in_cosmic, nrow = 1)
  # geom_density_ridges2()+
  # theme_ridges()
  # geom_ridgeline()
```

## VAF in Bick+WHO
```{r}
# CH_variants %>%
#   distinct(IDs, .keep_all = TRUE) %>%
#   filter(mutations_in_BickWHO == "Yes") %>% 
#   summarise(mean_of_nc_freq_allele_count = mean(nc_freq_allele_count, na.rm = TRUE),
#             median_of_nc_freq_allele_count = median(nc_freq_allele_count),
#             min_of_nc_freq_allele_count = min(nc_freq_allele_count),
#             max_of_nc_freq_allele_count = max(nc_freq_allele_count),)


# CH_variants %>%
#   distinct(IDs, .keep_all = TRUE) %>% 
#   filter(mutations_in_BickWHO == "Yes") %>% 
#   unite(IDs, c(IDs, SYMBOL)) %>% 
#   select(IDs,
#          # nc_freq_allele_count, nc_freq_allele_count_female, nc_freq_allele_count_male,
#          nc_freq_allele_count_afr, nc_freq_allele_count_afr_female, nc_freq_allele_count_afr_male,
#          nc_freq_allele_count_amr, nc_freq_allele_count_amr_female, nc_freq_allele_count_amr_male,
#          nc_freq_allele_count_eas, nc_freq_allele_count_eas_female, nc_freq_allele_count_eas_male,
#          # nc_freq_allele_count_sas, nc_freq_allele_count_sas_female, nc_freq_allele_count_sas_male,
#          nc_freq_allele_count_nfe, nc_freq_allele_count_nfe_female, nc_freq_allele_count_nfe_male 
#          ) %>%
#   pivot_longer(cols = -IDs) %>%
#   mutate(value = as.numeric(value)) %>%
#   mutate(name = str_remove(name, "nc_freq_allele_count_"),
#          # name = str_replace(name, "nc_freq_allele_count", "overall"),
#          name = factor(name, levels = c(
#     # "male", "female", "overall",
#     "nfe_male", "nfe_female", "nfe",
#     "eas_male", "eas_female", "eas", 
#     # "sas_male", "sas_female", "sas", 
#     "amr_male", "amr_female", "amr", 
#     "afr_male", "afr_female", "afr")
#     )) %>% 
#   mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
#     is.na(Sex)             ~ "overall",
#     TRUE                   ~ Sex
#   )) %>%
#   mutate(Ethnicity = str_extract(name, "afr|amr|nfe_nwe|nfe|eas"), Ethnicity = case_when(
#     is.na(Ethnicity)       ~ "overall",
#     TRUE                   ~ Ethnicity
#   )) %>%
#   ggplot(aes(x=name, y= value, color= Ethnicity))+
#   geom_boxplot()+
#   scale_y_log10()+
#   labs(title = "Population Allele Frequency in Non-Cancer population & \nin Bick+WHO", 
#        x="Populations", y="log10 Alternate Allele Frequency")+
#   coord_flip()
# 




CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>% 
  filter(mutations_in_BickWHO == "Yes") %>% 
  unite(IDs, c(IDs, SYMBOL)) %>% 
  select(IDs,
         # nc_freq_allele_count, nc_freq_allele_count_female, nc_freq_allele_count_male,
         nc_freq_allele_count_afr, nc_freq_allele_count_afr_female, nc_freq_allele_count_afr_male,
         nc_freq_allele_count_amr, nc_freq_allele_count_amr_female, nc_freq_allele_count_amr_male,
         nc_freq_allele_count_eas, nc_freq_allele_count_eas_female, nc_freq_allele_count_eas_male,
         # nc_freq_allele_count_sas, nc_freq_allele_count_sas_female, nc_freq_allele_count_sas_male,
         nc_freq_allele_count_nfe, nc_freq_allele_count_nfe_female, nc_freq_allele_count_nfe_male 
         ) %>%
  pivot_longer(cols = -IDs) %>%
  mutate(value = as.numeric(value)) %>%
  mutate(name = str_remove(name, "nc_freq_allele_count_"),
         # name = str_replace(name, "nc_freq_allele_count", "overall"),
         name = factor(name, levels = c(
    # "male", "female", "overall",
    "nfe_male", "nfe_female", "nfe",
    "eas_male", "eas_female", "eas", 
    # "sas_male", "sas_female", "sas", 
    "amr_male", "amr_female", "amr", 
    "afr_male", "afr_female", "afr")
    )) %>% 
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>%
  mutate(Ethnicity = str_extract(name, "afr|amr|nfe_nwe|nfe|eas"), Ethnicity = case_when(
    is.na(Ethnicity)       ~ "overall",
    TRUE                   ~ Ethnicity
  )) %>%
  select(name, value,
         Ethnicity, Sex) %>% 
  # ggplot(aes(x=value, y=name, fill=Ethnicity, linetype= Sex)) +
  # geom_density_ridges(alpha=0.5)+
  ggplot(aes(x=value, y=name, fill=Ethnicity, linetype= Sex, height = ..density..)) +
  geom_density_ridges(alpha=0.5, stat = "density")+
  ggtitle("Popultaion Allele Frequency Density Distribution")+
  labs(x= "Popultaion Allele Frequency", y= NULL)

CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>%
  select(nc_freq_allele_count_afr,
         nc_freq_allele_count_amr, nc_freq_allele_count_nfe,
         nc_freq_allele_count_eas,
         nc_freq_allele_count_female, nc_freq_allele_count_male,
         mutations_in_BickWHO) %>% 
  tbl_summary(by= mutations_in_BickWHO,
              digits = all_continuous() ~ function(x) format(x, digits = 3, scientific = TRUE)
              ) %>% 
  bold_labels() %>% 
  add_p() %>% bold_p(0.05) %>% add_overall() %>% 
  modify_spanning_header(c(stat_1, stat_2) ~ "**In Bick+WHO?**")

CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>%
  filter(mutations_in_BickWHO == "Yes") %>%
  # filter(SYMBOL %in% c("DNMT3A", "SF3B1")) %>%
  select(IDs, SYMBOL,
         # nc_freq_allele_count, nc_freq_allele_count_female, nc_freq_allele_count_male,
         nc_freq_allele_count_afr, nc_freq_allele_count_afr_female, nc_freq_allele_count_afr_male,
         nc_freq_allele_count_amr, nc_freq_allele_count_amr_female, nc_freq_allele_count_amr_male,
         nc_freq_allele_count_eas, nc_freq_allele_count_eas_female, nc_freq_allele_count_eas_male,
         # nc_freq_allele_count_sas, nc_freq_allele_count_sas_female, nc_freq_allele_count_sas_male,
         nc_freq_allele_count_nfe, nc_freq_allele_count_nfe_female, nc_freq_allele_count_nfe_male 
         ) %>%
  pivot_longer(cols = -c(IDs, SYMBOL)) %>%
  mutate(value = as.numeric(value)) %>%
  mutate(name = str_remove(name, "nc_freq_allele_count_"),
         # name = str_replace(name, "nc_freq_allele_count", "overall"),
         name = factor(name, levels = c(
    # "male", "female", "overall",
    "nfe_male", "nfe_female", "nfe",
    "eas_male", "eas_female", "eas", 
    # "sas_male", "sas_female", "sas", 
    "amr_male", "amr_female", "amr", 
    "afr_male", "afr_female", "afr")
    )) %>% 
  mutate(Sex = str_extract(name, "female|male"), Sex = case_when(
    is.na(Sex)             ~ "overall",
    TRUE                   ~ Sex
  )) %>%
  mutate(Ethnicity = str_extract(name, "afr|amr|nfe_nwe|nfe|eas"), Ethnicity = case_when(
    is.na(Ethnicity)       ~ "overall",
    TRUE                   ~ Ethnicity
  )) %>%
  # ggplot(aes(x=value, y=name, fill=Ethnicity, linetype= Sex)) +
  # geom_density_ridges(alpha=0.5)+
  ggplot(aes(x=value, y=name, fill=Ethnicity, linetype= Sex, height = ..density..)) +
  geom_density_ridges(alpha=0.5, stat = "density")+
  ggtitle("Popultaion Allele Frequency Density Distribution")+
  labs(x= "Popultaion Allele Frequency", y= NULL)+
  facet_wrap(. ~ SYMBOL, scales = "free_x")

CH_variants %>%
  distinct(IDs, .keep_all = TRUE) %>%
  filter(mutations_in_BickWHO == "Yes") %>% 
  # filter(SYMBOL %in% c("DNMT3A", "SF3B1")) %>% 
  select(nc_freq_allele_count_afr,
         nc_freq_allele_count_amr, nc_freq_allele_count_nfe,
         nc_freq_allele_count_eas,
         nc_freq_allele_count_female, nc_freq_allele_count_male,
         SYMBOL) %>% 
  tbl_summary(by = SYMBOL,
              type = list(everything() ~ "continuous"),
              digits = all_continuous() ~ function(x) format(x, digits = 3, scientific = TRUE)
              ) %>% 
  bold_labels() %>% 
  add_p() %>% bold_p(0.05) %>% add_overall() %>% 
  modify_spanning_header(c(stat_1, stat_2) ~ "**Gene in Bick+WHO**")
```



## Bick+WHO in Cancer pop

```{r}
variants_in_cosmic %>% 
  filter(mutations_in_BickWHO == "Yes") %>% 
  distinct(Sample.name, SYMBOL, .keep_all = TRUE) %>% 
  ggplot(aes(x= fct_infreq(SYMBOL), fill= SYMBOL))+
  geom_bar()+
  ggtitle("Number of overall COSMIC samples per gene In Bick+WHO",
          subtitle = "Warning patient counted multiple time if had both gene")+
  scale_fill_discrete(name = NULL)+
  labs(x= NULL, y="Number of samples")+
  coord_flip()
```

```{r}
variants_in_cosmic %>% 
  arrange(desc(mutations_in_BickWHO)) %>%
  distinct(Sample.name, .keep_all = TRUE) %>% 
  select(Age, mutations_in_BickWHO) %>% 
  tbl_summary(by= mutations_in_BickWHO,
    statistic = list(all_continuous() ~ "{mean} ({sd})"),
    digits = all_continuous() ~ 2
  ) %>% 
  bold_labels() %>%
  add_p() %>% bold_p(t=.05) %>% add_overall() %>% 
  modify_header(
    label = "**Cosmic Patients Characteristics**"
  ) %>%
  modify_spanning_header(all_stat_cols() ~ "**In Bick+WHO**")
```

<br>
<br>

***
<br>



# Definition

Mutation somatic status :  
Information on whether the sample was reported to be Confirmed Somatic, Previously Reported or Variant of unknown origin  
- Previously observed = when the mutation has been reported as somatic previously but not in current paper.  
- Confirmed Somatic = if the mutation has been confimed to be somatic in the experiment by sequencing both the tumour and a matched normal from the same patient.  
- Variant of unknown origin = when the mutation is known to be somatic but the tumour was sequenced without a matched normal.


INFO=<ID=AC,Number=A,Type=Integer,Description="Alternate allele count for samples  
AF, Alternate allele frequency in samples  
AN, Total number of alleles in samples  
non_cancer_AC, Alternate allele count for samples in the non_cancer subset  
non_cancer_AF, Alternate allele frequency in samples in the non_cancer subset  
afr, African-American/African ancestry
amr, Latino ancestry
asj, Ashkenazi Jewish ancestry
eas, East Asian ancestry  
sas, South Asian ancestry
fin, Finnish ancestry  
nfe, Non-Finnish European ancestry  
nfe_bgr, Bulgarian (Eastern European) ancestry  
nfe_est, Estonian ancestry  
nfe_nwe, North-Western European ancestry  
nfe_onf, Other Non-Finnish European ancestry  
nfe_seu, Southern European ancestry  
nfe_swe, Swedish ancestry  
oth, Other ancestry  

https://docs.gdc.cancer.gov/Data/PDF/Data_UG.pdf

GMAF Non-reference allele and frequency of existing variant in 1000 Genomes   
ExAC_MAF Frequency of existing variant in ExAC combined population  
ExAC_NFE_MAF Frequency of existing variant in ExAC Non-Finnish European population   
ExAC_OTH_MAF Frequency of existing variant in ExAC combined other combined populations   
ExAC_SAS_MAF Frequency of existing variant in ExAC South Asian population   
CLIN_SIG Clinical significance of variant from dbSNP Somatic status of existing variant(s)  
SOMATIC Somatic status of existing variant(s)  
[88- SOMATIC Somatic status of each ID reported under Existing_variation (0, 1, or null).]  
PHENO Indicates if existing variant is associated with a phenotype, disease or trait  
[99-PHENO  Indicates if existing variant is associated with a phenotype, disease or trait (0, 1, or null)]  

SOMATIC - Somatic status of existing variant(s); multiple values correspond to multiple values in the Existing_variation field  
PHENO - Indicates if existing variant is associated with a phenotype, disease or trait; multiple values correspond to multiple values in the Existing_variation field

73- SIFT The SIFT prediction and/or score, with both given as prediction (score)  
[SIFT predicts whether an amino acid substitution affects protein function based on sequence homology and the physical properties of amino acids. SIFT can be applied to naturally occurring nonsynonymous polymorphisms and laboratory-induced missense mutations.]  
74- PolyPhen The PolyPhen prediction and/or score  
[PolyPhen-2 (Polymorphism Phenotyping v2) is a tool which predicts possible impact of an amino acid substitution on the structure and function of a human protein using straightforward physical and comparative considerations]  

# Questions/Answers

* Julie Dutil - ancestry  
  + I talked with Zhihua Chen.  
  + He said it depends what we want to study.  
  + We could use nfe (Non-Finnish European ancestry) which contain nfe_nwe (North-Western European ancestry) + nfe_seu (Southern European ancestry).  
  + But if we want to be more precise we can use nfe_nwe which is the closest to Great Britain aka Utah individuals. Not using the southern European population is better when comparing with Latino population as they are shown to overlap.  
  + For Asian population, he recommend talking about both as Eastern and Southern Asian are really different but if we have to pick one for our analysis (comparison with Caucasian) we should use Eastern Asian in the main figure. That is because Southern Asian ancestry shares more ancestry with European. Maybe talk about Southern Asian in supplement...
  
* IDH1 in Cosmic  
  + I checked, IDH1 is present in Cosmic on chr 2. The selected IDH1 variants are not present in Cosmic.  

